%\documentclass[handout]{beamer} % \pause disabled
\documentclass[t]{beamer} % \pause enabled
\input{macros_arya}
\usepackage{datetime}
\def\vecbold#1{{\boldsymbol#1}}

\newdateformat{monthyeardate}{%
  \monthname[\THEMONTH], \THEYEAR}

%\usepackage[linesnumbered,ruled]{algorithm2e}
\usetheme{Madrid}
\setbeamertemplate{footline}
{
  \leavevmode%
  \hbox{%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{author in head/foot}%
    \usebeamerfont{author in head/foot}\insertshortauthor
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{title in head/foot}%
    \usebeamerfont{title in head/foot}\insertshorttitle
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,right]{date in head/foot}%
    \usebeamerfont{date in head/foot}\insertshortdate{}\hspace*{2em}
    \insertframenumber{} / \inserttotalframenumber\hspace*{2ex} 
  \end{beamercolorbox}}%
  \vskip0pt%
}
\def\dHAF{\text{-HAF}}
\makeatother
\title{Detecting Selection in Experimental Evolution Experiment}
\subtitle{}
\author[Arya Iranmehr]
{%
  \texorpdfstring{
      \centering
      Arya Iranmehr\\
      \href{mailto:airanmehr@ucsd.edu}{airanmehr@ucsd.edu}
  }
  {Arya Iranmehr}
}
%\email{}
\institute{
Bafna Lab\\
University of California, San Diego}
\date{
\monthyeardate\today}
\subject{Optimization}
\usefonttheme{serif} % default family is serif
%\AtBeginSection[]
%  {
%     \begin{frame}<beamer>
%     \frametitle{Agenda}
%     \tableofcontents[currentsection,currentsubsection]
%     \end{frame}
%  }
%\AtBeginSubsection[]
%{
%  \begin{frame}<beamer>{Outline}
%    \tableofcontents[currentsection,currentsubsection]
%  \end{frame}
%}

\begin{document}
\begin{frame}
  \titlepage
\end{frame}


\begin{frame}{Introduction}
\begin{itemize}
\item We are interested in identifying genetic adaptations in different organisms in find genes (or alleles) that are beneficial w.r.t. a  \emphr{selective pressure}.
\item Examples of interesting selective pressures: harsh environmental conditions, antibiotics, chemotherapy, etc.
\end{itemize}
\end{frame}



\begin{frame}{History}
\begin{itemize}
\item Traditionally, detecting/locating selection is done by analysing Allele 
Frequency Spectrum(AFS) or Haplotypes of a region (say 50Kbp).
\item {\bf Now}
\begin{enumerate}
\item We can sequence the population at different generations
\item We can't wait until fixation. 
\end{enumerate}
\end{itemize}
\end{frame}

\begin{frame}
Experimental Evolution
\begin{figure}
\includegraphics[trim={0 2.1in 0 0},clip, scale=0.6]{eg.jpg} 
\end{figure}
\end{frame}

\begin{frame}{Goals}
\begin{itemize}
\item Given the genome time series data (allele frequencies) we are interested in
\begin{enumerate}[I]
\item Detecting selection 
\item Locating selection
\begin{enumerate}[(i)]
 \item Identify gene under selection
 \item Identify the mutation 
\end{enumerate}
\item Estimating model parameters
\end{enumerate}
\end{itemize}
\end{frame}

\begin{frame}{Simulations}
\begin{enumerate}[(i)]
\item For each simulation, population of $F=200$ founder lines is created in  \texttt{msms} program with parameters 
\begin{itemize}
\item window size $L=50$Kbp.
\item mutation rate $\theta=200$.
\end{itemize}

\item Using $F=200$ founder lines a population diploid is created.
\item Using forward simulator population is evolved and AF is sampled every 10 generation for 50 generations. 
\item This procedure is repeated 100 times for each $s=\{0.1, 0.05, 0.02, 0.01, 0\}$.
\end{enumerate}
\end{frame}


%\begin{frame}{Hard Sweep}
%\begin{itemize}
%\item With this setting around 20\% of sites are at Minimum Allele Frequency 
%=1/200  (at time 0).
%\end{itemize}
%\hspace{-0in}\includegraphics[scale=0.17]{afs}
%\end{frame}

\begin{frame}{Single Locus Logistic Model: Carrier Frequency}
Dynamics of sweep extremely depend on $s$ as well as initial carrier frequency.Sampling times are very important prediction performance!

\begin{figure}
\centering
\hspace{-0in}\includegraphics[trim={2in 0.5in 1.5in 0in},clip,page=2,width=0.5\textwidth]{sigmoidSoft}
\hspace{-0in}\includegraphics[trim={2in 0.5in 1.9in 0in},clip,page=2,width=0.5\textwidth]{sigmoidHard}
\end{figure}
%{\tiny
%\begin{enumerate}
%\item Left) Terhorst et al. considered soft sweep. The time-window which  they 
%sampled is shaded. 
%\item Right) With the same sampling scheme, it is difficult to make 
%predictions base on observations under 50 generations.
%\end{enumerate}
%}
\end{frame}

\begin{frame}{Bottleneck}
\begin{itemize}
	\item side-effect of not tracking new mutations and restricting population 
	to founder-lines.
%\item When restricting observations to a fixed set of sites, at each 
%generation some sites get lost (due to drift), but no new mutation happens.
%\item Pr(Absorption of site $i$) = Frequency of site $i$.
\end{itemize}
\begin{figure}
\centering
\includegraphics[trim=2in 0 2in 0, clip,width=\textwidth]{bottleneck}
\end{figure}

\end{frame}



\begin{figure}
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  \caption{Mean and 95\% CI of 1000 simulations for strong selection.}
\end{figure}

\begin{figure}
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=4,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  \caption{Mean and 95\% CI of 1000 simulations for weak selection.}
\end{figure}

\begin{frame}[allowframebreaks]{Single-Locus Least-Squares Method}
\begin{figure}
\begin{enumerate}
\item Model is consistent with observations
\begin{figure}
  \centering
\hspace{-0.5in}\includegraphics[trim={2in 9.3in 2in 0in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  		   \end{figure}
  		   \newpage
 \item strong linkage in a window
\begin{figure}
\centering
\hspace{-0in}\includegraphics[scale=0.2]{spaceTimeLD}
\end{figure}
 \item Estimating $s$ at each site
\begin{figure}
\centering
\hspace{-0in}\includegraphics[scale=0.2]{ls01}
\end{figure}

 \item spectrum of $\hat{s}$ for a window
\begin{figure}
\centering
\hspace{-0in}\includegraphics[scale=0.15]{sspectrum}
\end{figure}
\item We considered avg of top $\hat{s}$ as a predictor of selection for a 
window. 
\end{enumerate}

\end{figure}
\end{frame}

\begin{frame}{Tajima's D and Fay Wu's H}
\begin{enumerate}
\item After correcting observations for bottleneck effect the model look like observations, but it has a higher variance.
\begin{figure}
\hspace{-0.5in} \includegraphics[trim={1.8in 0.25in 2in 9.25in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  		   \end{figure}
\begin{figure}
\hspace{-0.5in} \includegraphics[trim={1.8in 7.75in 2in 2.75in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  		   \end{figure}

  \item The parametric models assumes hard-sweep, so the model become ivalid in soft sweep.
\end{enumerate}

\end{frame}


\begin{frame}{SFSelect and Gaussian Process}
\begin{enumerate}
\item SFSelect process is a monotone process, so we summed all the values in the time series to get a predictor for selection.
\begin{figure}
\hspace{-0.5in} \includegraphics[trim={1.8in 4.5in 2in 5.95in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
  		   \end{figure}

  \item GP is the state-of-the art model, proposed by Terhorst et al, which fits a Gaussian process on the time series data.
\end{enumerate}

\end{frame}

\begin{frame}{Experiments}
\begin{itemize}
\item for each setting we performs 200 simulations, (100 neutral and 100 selection), and predictive performance of all the methods in detecting selection.
\item We computed ROC curve and defined power of a method as area under ROC curve when False-Positive rate is less than 0.1.
\end{itemize}
\end{frame}

\begin{figure}
  \centering \includegraphics[width=\textwidth]{{power}}
\end{figure}



\begin{frame}{Summary}
\begin{itemize}
\item Behaviour of different population statistics in time is studied.
\item A method based on single locus AF is proposed and shown to have superior performance than multi-locus and traditional methods.
\item Analysis of real data and locating genes in the genome is our next step.
\end{itemize}

\end{frame}


%\end{frame}


%
%\begin{frame}{Computational Performance}
%\begin{figure}[H]
%  \centering
%    \includegraphics[width=0.6\textwidth]{times}
%\end{figure}
%\end{frame}





\begin{frame}{\ }
\vspace{1	in}
\begin{center}
	\huge{Thanks!}\\
\end{center}

\end{frame}

\begin{frame}{Single Locus Model: Carrier Frequency}
\begin{itemize}
\item By differentiating update equations ($x_{t+1}=x_t+\frac{sx_t(1-x_t)}{2+2sx_t}$ w.r.t. $t$ and solving differential equation, we have
\beqq
\nu_t=\sigma\left(st/2+\eta(\nu_0)\right)
\eeqq
where $\sigma(.)$ is logistic function and $\eta=\sigma^{-1}$ is the logit function and $\nu_t$ is the frequency of the carrier at time $t$.
\end{itemize}
\hspace{-0.6in}\includegraphics[scale=0.2]{sigmoid}
\end{frame}
\begin{frame}{Multi Locus Model: Average Haplotype Allele Frequency}

\begin{equation*}
\resizebox{.99\hsize}{!}{
$\mathbb{E}[1\dHAF(t)]= \|\mathbf{x}_t\|^2\approx \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
 \theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)$
}
\end{equation*}
where $ \mathbf{x}_t$ is vector of AF at time $t$ and $\nu_t=\sigma\left(st/2+\eta(\nu_0)\right)$.

\hspace{-0.1in}\includegraphics[scale=0.2]{haf}
\end{frame}
\begin{frame}{Critics to GP}
\begin{itemize}
\item Its too complex, likely to overfit with small number of iid replicates.
\item Although the likelihood model is based on different parameters, in practice, it can learn only one parameter at a time.
\item not tractable, its time complexity is quartic!
\item worse, each iteration requires \texttt{maxGeneration} recursion which makes it very hard to analyse late epochs of sweeps.
\item In addition to PoolSeq data, it requires initial population haplotypes.
\item Despite its elegant theory, it has not compared with classical methods.
\item In practice, single locus scan is performed and multi-locus (with 3-7 seg. sites.) model is fitted at regions of interests.
\end{itemize}
\end{frame}
\end{document}


\begin{frame}{Simulations}
\begin{enumerate}[(i)]
\item For each simulation, population of $F=200$ founder lines is created in  \texttt{msms} program with parameters 
\begin{itemize}
\item Length of genome $L=10^6$Kbp.
\item Population size $N_e=10^6$.
\item Recombination rate $r=2\times10^{-8}$.
\item Mutation rate $\mu=10^{-9}$ which give $\theta = 200$ and $m\approx 1200$ .
\end{itemize}

\item Using $F=200$ founder lines a population of $N=1000$ diploid homozygote individuals are created.
\item Using \texttt{SimuPop}, the initial population is forward simulated for 10 replicates and AF is recorded every 10 generation for a complete hard sweep. 
\item I.e., a site with minimum allele frequency of $1/F$ is always under selection.
\item for each $s=\{0.1, 0.05, 0.02, 0.01, 0\}$ 100 simulations are done.
\end{enumerate}
\end{frame}
\begin{frame}{Multi-Locus Methods}
\begin{itemize}
\item In hard sweep, site under selection has perfect LD with all the other sites at time 0.\item With hard-sweep, it is difficult to aim locating mutation in less than 2K generations.
\item 50Kb window seems to be a reasonable resolution.
\end{itemize}
\begin{figure}
\centering
\hspace{-0in}\includegraphics[scale=0.17]{spaceTimeLD}
\caption{Decay of LD in space through time under neutral evolution, when selected site is at position 500K. 50Kb window centered at the site is shaded.}

\end{figure}
\end{frame}
