\documentclass[11pt]{article}
\input{macros_vineet}
\input{macros_arya}
\title{A Hidden Markov Model for Analyzing Adaptive Experimental Evolutions with Pooled Sequencing Data}
\author[1]{Arya Iranmehr}
\author[1]{Ali Akbari}
\author[2]{Christian Schl\"{o}tterer}
\author[3]{Vineet Bafna}
\affil[1]{\footnotesize Electrical and Computer Engineering, University of California, San Diego, La Jolla, CA 92093, USA.}
\affil[2]{\footnotesize Institut f\"{u}r Populationsgenetik, Vetmeduni, Vienna, Austria.}
\affil[3]{\footnotesize Computer Science \& Engineering, University of California, San Diego, La Jolla, CA 92093, USA}
\date{}
\def\comale{\text{{\sc Comale}}}
\usepackage{booktabs}
\begin{document}
\maketitle
\begin{abstract}
  Experimental evolution (EE) studies are powerful tools for observing
  molecular evolution ``in-action'' in wild and controlled
  environments. This paradigm of experiment was infeasible until
  recently when the whole-genome and whole-population was made
  possible by next-generation sequencing technologies.  However, one
  of the primary constraints of the EE studies is the limited time for
  the experiment, which primarily depend on the organism's generation
  time. This constraint impedes adaptation and optimization
  (evolvability) studies, where the population can only evolved and
  re-sequenced in a small number of generations, relative to the
  number of generations required for fixation of adaptive allele.
  Although a powerful library of tests-of-selection has already been
  developed, they are mainly designed for static data to to identify
  adaptation when the sample is taken close enough (before/after) to
  the fixation of adaptive allele.  In this article, we study the
  problem of identifying selective sweep in short-term experimental
  evolution of sexual organisms and propose Composite Of MArkovian
  Likelihoods for Experimental evolution (\comale) statistic which
  computes its score by averaging likelihood ratios of polymorphisms
  for a genomic region.  The likelihood of null (neutral) and
  alternative (selection) hypotheses calculated using the
  Wright-Fisher Markov chain model for each variant. Extensive
  simulation study shows that \comale\ achieves higher detection power
  methods on both soft and hard sweep simulations for various
  selection strengths.  Finally, we apply the \comale\ statistic to
  the controlled experimental evolution of D. melanogaster to detect
  adaptive genes/alleles under alternating cold and hot temperatures.
\end{abstract}



\section{Introduction}

Natural selection is the key force in evolution, and a mechanism by
which populations can adapt to external `selection'
constraints. Examples of adaptation abound in the natural world,
including for example, classic examples like lactose tolerance in
Northern Europeans~\cite{bersaglieri2004genetic}, human adaptation to high
altitudes~\cite{yi2010sequencing,simonson2010genetic}, but also drug resistance 
in
pests~\cite{daborn2001ddt}, HIV~\cite{Feder2016More},
cancer~\cite{gottesman2002mechanisms,zahreddine2013mechanisms},
malarial parasite~\cite{ariey2014molecular,nair2007recurrent}, and
other antibiotic resistance~\cite{spellberg2008epidemic}. In each of
these examples, understanding the genetic basis of adaptation can
provide actionable information, underscoring the importance of the
problem.

Recent advances in whole genome sequencing have enabled us to sequence
populations at a reasonable cost not only for natural populations, but
also time-series samples from \emph{longitudinal studies} of
populations, allowing us to study the dynamics of evolution. Modern
experimental evolution refers to the study of the evolutionary
processes of a model organism at genomic level in a controlled
\cite{hegreness2006equivalence,lang2013pervasive,orozco2012adaptation,
  lang2011genetic,barrick2009genome,bollback2007clonal,oz2014strength}
or natural
\cite{maldarelli2013hiv,reid2011new,denef2012situ,winters2012development,
  daniels2013genetic,barrett2008natural,bergland2014genomic}
environment.  Although constraints such as small population sizes,
limited timescales and oversimplified laboratory environments limits
interpreting experimental evolution results, they can be used to test
different hypotheses~\cite{kawecki2012experimental} regarding mutation
rate, inbreeding, environmental variability, sexual selection \&
conflict, kin selection and cooperation life history and sex
allocation, sexual reproduction and mating systems, behavior and
cognition, host–parasite interactions, speciation repeatability of
evolution and make more accurate inferences than static data analysis
\cite{boyko2008assessing,desai2008polymorphism,sawyer1992population}. In
addition, dynamic data has been used to estimate model parameters
including population size
\cite{williamson1999using,wang2001pseudo,pollak1983new,waples1989generalized,
  Terhorst2015Multi} strength of selection
\cite{mathieson2013estimating,illingworth2011distinguishing,Terhorst2015Multi,
  bollback2008estimation,illingworth2012quantifying,malaspinas2012estimating,
  Steinrücken2014a}, allele age \cite{malaspinas2012estimating}
recombination rate~\cite{Terhorst2015Multi}, mutation
rate~\cite{Barrick2013Genome, Terhorst2015Multi}, quantitative trait
 loci~\cite{baldwin2014power} and test neutrality
hypotheses
\cite{feder2014Identifying,Terhorst2015Multi,burke2010genome,bergland2014genomic}.


Among different types of evolution experiments
\cite{Barrick2013Genome,schlotterer2015combining} in this paper we
only focus on adaptive evolution of multicellular sexual organisms
with continuous culture, fixed population size, and for the most part,
positive single locus selection (only one favored mutation).  For this
setting, \emph{Drosophila} is often the model organism of choice. It
has been used to identify adaptive genes in longevity and aging
~\cite{burke2010genome,remolina2012genomic} (600 and generations
respectively), courtship song \cite{turner2011population} (100
generations), hypoxia tolerance \cite{zhou2011experimental} (200
generations), adaptation to new
temperatures\cite{orozco2012adaptation,tobler2014massive} (59
generations), egg size \cite{jha2015whole} (40 generations), C virus
resistance \cite{martins2014host} (20 generations), and
dark-fly~\cite{izutsu2015dynamics} (49 generations) experiments. We
also note that our methods can be generalized to other sexual
populations, but are separate from the analysis of asexual
populations. In sexual populations, the mutation responding to the
selection constraint is linked to neighboring mutations, and this
provides a strengthened signal. Therefore, methods for identifying
genomic regions under selection in sexual populations focus on genomic
regions, rather than single sites.

The task of identification of a selection event in natural or
experimental evolution can refer to one of many different
processes. At the coarsest level, identification can be done by
determining whether a region (e.g. a small region with no or low
recombination) on genome is under selection or not.  In the following,
we refer to this task as \emph{detection}. In contrast, the task of
\emph{site-identification} corresponds to the process of finding the
favored mutation/allele itself. Finally, estimating model parameters
such as strength of selection and overdominance at the site fully
describes the selective sweep.

A wide range of computational methods~\cite{vitti2013detecting} have
been developed to identify regions under positive selection. Majority of the 
existing
methods have devoted to \emph{static data} analysis, statistical test on single 
sample of the
population at a specific time, either during the sweep, or subsequent
to fixation of the favored allele. For instance, reduction in genetic
diversity\cite{tajima1989statistical,fay2000hitchhiking,ronen2013learning}
in allele-frequency data, prevalence of long haplotypes
\cite{sabeti2006positive,vitti2013detecting} in haplotype (phased)
data, population differentiation
\cite{holsinger2009genetics,burke2010genome} in multiple-population
data and others. An important component of many methods is the
analysis of the Site Frequency Spectrum (SFS) to identify departure
from neutrality. Classical examples including Tajima's \emph{D}
~\cite{tajima1989statistical}, Fay and Wu's
\emph{H}~\cite{fay2000hitchhiking}, Composite Likelihood
Ratio~\cite{nielsen2005genomic}, were all shown to be weighted linear
combination of the SFS values~\cite{achaz2009frequency}. However, the
SFS of a sample under selection changes with time since onset of
selection, the frequency of the favored allele at the onset of
selection (High initial frequency is also referred to as standing
variation) and other parameters. Therefore, different weights used by
different tests influence their power under different selection
regimes. This fact was exploited to learn the right weights for each
regime in SFSelect~\cite{ronen2013learning}.

While successful, these methods can miss true regions under
selection. It has been argued that ``many, if not most, cases of
adaptation are yet to be
discovered''~\cite{messer2013population}. Besides these false
negatives, the methods are also prone to false-discoveries due to
other confounding factors such as demography, including bottleneck and
population expansions, and ascertainment bias ~\cite{ptak2002evidence,
  ramos2002statistical,akey2009constructing,
  nielsen2003correcting,messer2013population}. Nevertheless, SFS-based
tests are simple and inexpensive and continue to be used, often in
combination with other
tests~\cite{akey2009constructing,vitti2013detecting}. It remains an
open question if the analysis of \emph{dynamic} time-series data can
improve the power of these methods.


\ignore{

rapid increase in allele frequencies
\cite{bergland2014genomic} in the dynamic data are different
signatures of selective sweep in the polymorphism data.




In practice, (positive and sum-to-one) weighted 
linear combinations scaled SFS distribution \cite{achaz2009frequency}
is used as different estimators of $\theta$ and pairwise discrepancy between 
them is used as a test statistic for detecting selection.
Under neutrality, the discrepancies should be distributed around zero, and a 
simple $t$ test can provide p-value for rejecting neutrality. 
For example, 
test statistics for Tajima's $D$\cite{tajima1989statistical}, 
Fay Wu's $H$~\cite{fay2000hitchhiking} and 
SFSelect\cite{ronen2013learning} can be obtained by a dot 
product of the scaled SFS vector with their corresponding weight vector. 
\paragraph{Dynamic Data (ad-hoc).} 
}

While on the one hand, we have an extensive development of tests for
analysis of static samples, tests-of-selection for dynamic time-series
data is less studied, and often existing tests for static data are
adopted for dynamic data with two time-points. For example, Zhu et
al.~\cite{zhou2011experimental} used the ratio of the estimated
population size of case and control populations to compute test
statistic for each genomic region. Burke et al.~\cite{burke2010genome}
applied Fisher exact test to the last observation of data on case and
control populations. 
Orozco-Terwengel et al.~\cite{orozco2012adaptation} used 
Cochran-Mantel-Haenszel test~\cite{agresti2011categorical} to detect SNPs that 
their read counts are 
consitently  changed in all the replicates of two time-point data.
Turner et al. \cite{turner2011population} proposed diffStat statistic to test 
whether the distribution of change in allele frequencies of two population 
deviates from the distribution of change in allele frequencies of two drifting 
populations.
Bergland et al. \cite{bergland2014genomic}
applied $F_{st}$ to populations throughout time to signify their
differentiation from ancestral (two time-point data) as well as geographically 
different populations.  Jha et al. \cite{jha2015whole} computed test statistic
of generalized linear-mixed model directly from read counts.
Bollback et
al. \cite{bollback2008estimation} provided diffusion approximation to
the continues Wright Fisher Markov process and estimated $s$
numerically and performed standard likelihood ratio test under $\Xc^2$ 
distribution. Using (continuous-time
continuous-state) Brownian motion process, Feder et al. 
\cite{feder2014Identifying} proposed
Frequency Increment Test(FIT) for dynamic allele frequency data and suggested 
to perform statistical testing under empirical distribution of test statistic
when the number of independent samples (replicates) are small.
 More recently, Song et al.
\cite{Terhorst2015Multi} proposed empirical Gaussian Process (GP) likelihood 
ratio test for single and multiple loci dynamic allele frequency data. 


A key contribution of our paper is the development of a fast, direct
method, \comale, for detecting selection in short-term experimental evolution 
with pooled sequencing (read count) data. 
We show that \comale\ is orders of magnitude
faster, and shows higher or similar sensitivity, compared to the state-of the 
art methods for a wide range of parameters in simulated data.
In addition, we also extend SFS statistics to be computed from
dynamic data, and show their strength and weakness. In particular, we show that 
dynamic SFS statistics outperform other methods in the natural experimental 
evolution when frequency of the favored allele is high.

Before we describe \comale, it is worth understanding some of the
limitations of experimental evolution studies. A
serious constraint in adaptive experimental evolution experiment is
the \emph{sampling-time-span}, the number of generations between the
first and last sampled generations. Given a fixed amount of time for a
study, sampling-time-span depends upon the generation time of the organism, and 
can often be small relative to fixation time. For example $30$-$50$ generations 
is typical for \emph{Drosophila}, with some 
exceptions~\cite{zhou2011experimental}. 
Therefore, unless the selection coefficient is very strong the time series data 
will only capture a ``partial sweep''. 
This limitation is more critical in the controlled experimental evolutions, 
where the sampling starts at the onset of selection, and favored allele grows 
in frequency very slowly.
On the other hand, in sampling Natural
populations, the time of onset of selection, may not be known.

Second, in controlled experimental evolution experiments, populations
are evolved and inbred. This scenario in which population size
significantly drops from the large number of wild type (e.g., for
\emph{Drosophila}, ($N_e\approx10^6$) to a small number of founder
lines $F$ ($\approx 10^2$) for Experimental Evolution,
resembles a severe population bottleneck. Such a intense reduction in
effective population size have confounds both dynamic-SFS statistics and 
dynamic site allele frequency based statistics\footnote{Smaller population size 
implies stronger genetic drift. In small population experiments, drift can be 
so strong that the frequency of a site changes rapidly so that it predicted as 
a site under selection.} which leads to high false-positives.


\section{Materials and Methods}

\paragraph{Notation.} 
Consider a locus with starting derived allele frequency
$\nu_0$. Frequencies are sampled at $T$ distinct generations specified
by ${\cal T}= \{\tau_i: 1\le \tau_1<\tau_2,\ldots\le \tau_T\}$, and
denoted by $\bm{\nu}=\{\nu_1,\ldots,\nu_T\}$. Moreover, $R$ replicate
measurements are made, and we denote the $r$-th replicate frequency
data as $\bm{\nu}^{(r)}$.

\subsection{The \comale\  statistic}
To identify if the locus is evolving under positive selection, we
follow previous approaches to focus on a parametrized likelihood based
model that (a) maximizes the likelihood of the time series data
w.r.t. selection and overdominance parameters $s,h$; and, (b) computes
the log odds ratio of the likelihood of selection model to the
likelihood of neutral evolution/drift model.

\paragraph{Likelihood for Neutral Model.}
To model neutral evolution, it is natural to model the change in
frequency $\nu_t$ over time via Brownian
motion~\cite{feder2014Identifying} or Gaussian
process~\cite{Terhorst2015Multi}. Significant deviations from this
Null could be indicative of non-neutrality. However, in our
experiments, we found that the Brownian motion approximation is
inadequate for small population sizes and low starting frequencies
that are typical in experimental evolution (see Results, and
Figure~\ref{fig:markov}). In fact, other continuous models such as
Gaussian process for allele frequencies, are also susceptible to this
issue.

Instead, by computing likelihood of data using discrete-time
discrete-state-space Wright-Fisher Markov Chain, we turn the problem
of small-population size into an advantage. Consider a neutrally
evolving diploid population with $N$ individuals. Define a
$2N\times2N$ transition matrix $P$, where $P^{(\tau)}[i,j]$ denotes
probability of change in allele frequency from $\frac{i}{2N}$ to
$\frac{j}{2N}$ in $\tau$ generations, solely due to genetic drift. $P$
is defined as follows~\cite{Ewens2012Mathematical}:
\begin{eqnarray}
  P^{(1)}[i,j] &=& \pr\left(\nu_{t+1}=\frac{j}{2N} \left|
      \nu_{t}=\frac{i}{2N}\right)={2N \choose j} \right.  \nu_{t}^j
  (1-\nu_{t})^{2N-j}, \;\;\label{eq:P1}\\
  P^{(\tau)} &=&   P^{(\tau-1)}P^{(1)} \label{eq:Pt}
\end{eqnarray}
Note that pre-computing and storing $P^{(\tau)}$ is tractable and
numerically stable for controlled experimental evolution experiments
where $N\le2000$. For larger $N$, binning the frequencies is
sufficient to model frequency changes.

\paragraph{Likelihood for Selection Model.}
Assume that the site is evolving under selection constraints $s,h\in
\Rbb$, where $s$ and $h$ denote selection strength and dominance,
respectively. By definition, the relative fitness values of genotypes
0$|$0, 0$|$1 and 1$|$1 are given by $w_{00}=1$, $w_{01}=1+hs$ and
$w_{11}=1+s$. Recall that $\nu_t$ denotes the frequency of the site at
time $\tau_t\in {\cal T}$. Then, $\nusp$, the frequency at time
$\tau_t+1$ can be estimated using: \beq \hat{\nu}_{t^+} =
\mathbb{E}[\nusp|s,h,\nu_t]&=\frac{w_{11}\nu_t^2 +
  w_{01}\nu_t(1-\nu_t)}{w_{11}\nu^2_t + 2w_{01}\nu_t(1-\nu_t) +
  w_{00}(1-\nu_t)^2}\\
&=\nu_t+\frac{s(h+(1-2h)\nu_t)\nu_t(1-\nu_t)}{1+s\nu_t(2h+(1-2h)\nu_t))}.
  \label{eq:transition}
\eeq
For finite populations, let $Q^{(\tau)}_{s,h}[i,j]$ denote the
probability of transition from $\frac{i}{2N}$ to $\frac{j}{2N}$ in
$\tau$ generations. We model $Q$ as follows
(See~\cite{Ewens2012Mathematical}, eq 1.58-59, page 24):
\begin{eqnarray}
  Q^{(1)}_{s,h}[i,j] &=& \pr\left(\nusp=\frac{j}{2N} \left\lvert
      \nu_{t}=\frac{i}{2N};s,h \right .\right)={2N \choose j}
  \hat{\nu}_{t^+}^{j} (1-\hat{\nu}_{t^+})^{2N-j}\label{eq:Q1}\\
  Q^{(\tau)}_{s,h} &=& Q^{(\tau-1)}_{s,h}Q^{(1)}_{s,h}\label{eq:Qt}
  \label{eq:mkvs}   
\end{eqnarray}
For $s=0$, Eq.~\ref{eq:Q1} and~\ref{eq:Qt} are identical to
Eq.~\ref{eq:P1} and~\ref{eq:Pt}, respectively.  The likelihood of
observing the trajectory $\bm{\nu}$ is computed using:
\begin{equation}
  \Lc_\Mc(s,h|\bm{\nu}) = \pr(\bm{\nu};\nu_0,s,h)=
  \prod_{t=1}^{T} \pr(\nu_{t}|\nu_{t-1};\nu_0,s,h) = \prod_{t=1}^{T} Q^{(\delta_t)}_{s,h}[\hat{i},\hat{j}],
%  \label{eq:mkvlik}  
\end{equation}
where, $(\hat{i},\hat{j})=(\lfloor 2N\nu_{t-1}\rfloor, \lfloor
2N\nu_{t}\rfloor)$, and $\delta_t=\tau_{t}-\tau_{t-1}$. Combining the
likelihood over independent replicate samples $\bm{\nu}^{(r)}$, we
get:
\begin{equation}
  \Lc_\Mc(s,h|\{\bm{\nu}^{(r)}\}) = \prod_r   \Lc_\Mc(s,h|\bm{\nu}^{(r)}).
  \label{eq:mkvlik}
\end{equation}
Let $s^*,h^*$ denote the parameters that maximize the likelihood. The
simplest form of the test statistic for each variant is given by
\begin{eqnarray}
\Lambda_\Mc= \sgn(s^*)\log 
\left(\frac{\Lc_\Mc(s^*,h^*|\{\bm{\nu}^{(r)}\})}{\Lc_\Mc(0,0|\{\bm{\nu}^{(r)}\})}\right).
\label{eq:mcts}
\end{eqnarray}
where the sign $\sgn(s^*)$ helps focus on positive selection using a
one-sided tests.  




\paragraph{Accounting for ascertainment bias using Hidden Markov Models.}
So, far we assumed that the exact allele frequencies are given. For
most data-sets, allele frequencies are estimated computed from
genotype data with finite samples or pooled-sequencing data with
finite depth of coverage (See
Figure~\ref{fig:stateConditional}).
Moreover, multiple replicates time series data, for each site have
multiple measurements corresponding to different generations and
multiple replicates, which are not necessarily homogeneous, e.g., have
the same depth (See Figure~\ref{fig:depthHetero}). To account for
(heterogeneous) ascertainment bias, we extend the Markov chain in
Eq.~\ref{eq:mkvlik} to a Hidden Markov Model for pooled-seq data.

Consider a variant position being sampled at time point $\tau_t\in
{\cal T}$. We denote the pooled-seq data for that variant as $x_t =
\langle c_t,d_t \rangle$ where $d_t, c_t$ represent the read depth,
and the count of the derived allele, respectively, at time
$\tau_t$. The dynamic data is represented by the sequence
$\mathbf{x}=x_1,x_2,\ldots,x_T$. Define an HMM with $2N+1$
states. State $i$ $(0\le i\le 2N)$ corresponds to allele frequency
$\frac{i}{2N}$. The HMM is stationary in that transition and emission
distributions do not change over time. Therefore, at any time step,
the probability that state $i$ emits $x=\langle d, c\rangle $ is given by
\begin{equation*}
{\bf e}_{i}(x) = {d \choose c} \left(\frac{i}{2N} \right)^c\left (1- \frac{i}{2N} \right)^{d-c}.
\end{equation*}
For $1\le t\le T$, let $\alpha_{t,i}$ denote the probability of
emitting the $x_1,x_2,\ldots,x_t$ and ending in state $i$ at
$\tau_t$. Then, $\alpha_{t,i}$ can be computed using Forward
procedure~\cite{durbin1998biological}:
\begin{equation}
  \alpha_{t,i} = \left( \sum_{1\le j\le 2N} \alpha_{t-1,j}\;Q^{(\delta_t)}_{s,h}[j,i] \right) {\bf e}_{i}(x_t)\;\; .
  \label{eq:hmm}
\end{equation}
where $\delta_t=\tau_t-\tau_{t-1}$. The joint likelihood of the
observed data from $R$ independent observations is given by
\begin{equation}
  \Lc_{\Hc}(s,h|\{\bm{x}^{(r)}\})=\prod_{r=1}^R\Lc_{\Hc}(s,h|\bm{x}^{(r)}) = \prod_{r=1}^R \sum_i\alpha_{T,i}^{(r)}\;\;.
  \label{eq:hmmlik}
\end{equation}
Similar to Eq.~\ref{eq:mcts}, let $s^*,h^*$ denote the parameters that
maximize likelihood. The modified likelihood statistic for pool-seq
data is given by
\begin{eqnarray}
\Lambda_{\cal H} &=& \sgn(s^*)\log 
\left(\frac{\Lc_{\cal H}(s^*,h^*|\{\bm{x}^{(r)}\})}{\Lc_{\cal H}(0,0|\{\bm{x}^{(r)}\})}\right).
\label{eq:hmmml}
\end{eqnarray}

\paragraph{Composite Likelihood.}
Consider a genomic region defined by a collection of segregating sites
$L$, with little or no recombination between sites. To test if some
site in the region is under selection, we would like to compute
composite likelihood functions under selection and drift regimes.  The
dynamic frequency of a site $\ell \in L$ is governed by selection,
drift, \emph{and also} linkage with other sites. Computing exact
likelihood requires estimating the linkage throughout time which is
computationally expensive.

Following previous authors, we simplify the computation by multiplying
individual SNP likelihoods, to get a Composite Likelihood Ratio score
(CLR)~\cite{nielsen2005genomic,williamson2007localizing}. CLR is a
popular and effective method for detecting regions under
selection~\cite{vitti2013detecting}. 

Let $\Lambda_{\cal M}(\ell)$ (or, $\Lambda_{\cal H}(\ell)$) denote the
likelihood ratio score for each site $\ell$ in $L$. Under positive
selection, a site $\ell$ increases in frequency, and sites on the same
lineage correspondingly increase in frequency too, at the expense of
sites on other branches, which decrease or drift in frequency. In
computing composite likelihood for a region $L$, we only include sites
whose likelihood ratio score is above the $\pi^{th}$ percentile of the
scores in that region. For percentile cut-off $\pi$, let
$L_{\pi}\subseteq L$ denote the set of sites whose likelihood ratio
scores had percentile $\pi$ or better. For all $\pi$, the modified CLR
statistic for Markov chain and HMM is computed using: 

\begin{eqnarray}
  {\cal M}_{\pi} &=& \frac{1}{|L_{\pi}|}\sum_{\ell \in L_{\pi}} \Lambda_{\cal
  M}(\ell),\\
{\cal H}_{\pi} &=& \frac{1}{|L_{\pi}|}\sum_{\ell \in L_{\pi}} \Lambda_{\cal H}(\ell)\; .
  \label{eq:pihmm}
\end{eqnarray}



\paragraph{Estimating parameters.}
\label{sec:regression}
Depending on data (read count or allele frequency) the optimal value
of the parameters can be found by 
\beqn
s^*,h^*&=&\underset{s,h}{\arg\max} \sum_r^R \log
\left(\Lc_M(s,h|\bm{\nu}^{(r)}\right),\;\; \mbox{ or, }\label{eq:mcmle}\\
s^*,h^*&=&\underset{s,h}{\arg\max} \sum_r^R \log
\left(\Lc_H(s,h|\bm{x}^{(r)}\right).\label{eq:hmmmle}
\eeqn
where likelihoods are defined in Eq.~\ref{eq:mkvlik} and
Eq.~\ref{eq:hmmlik}, respectively. For a given $h$,
Equations~\ref{eq:mcmle} and \ref{eq:hmmmle} are quasiconvex
(unimodal) functions in $s$ (see Appendix \ref{app:likelihood}) and
admit efficient algorithms, (e.g., bisection). The bisection
methods can be used to compute the optimum parameters within error
$\epsilon$ in $\Oc(\log_2(1/\epsilon))$ steps.
\subsection{Extending Site Frequency Spectrum  based tests for  time series data}
\label{sec:sfs-ts}
The site frequency spectrum (SFS) is a mainstay of tests of neutrality
and selection, and can be computed using allele frequencies (does not
need haplotypes). Following Fu, 1995~\cite{fu1995statistical}, any
linear combination of the site frequencies is an estimate of
$\theta$. However, under non-neutral conditions, different linear
combinations behave differently. Therefore, many popular tests of
neutrality either compute differences of two estimates of $\theta$, or
perform cross-population tests comparing the $\theta$ estimates in two
different
populations~\cite{achaz2009frequency,ronen2013learning,sabeti2007genome}. 

We asked if SFS-based tests could be adapted for time-series data. A
simple approach is to use cross-population SFS tests on the
populations at time $0$ (before onset of selection), and at time
sample $\tau_t$, for each $t$. However, these tests are not
independent. Evans et al.  \cite{evans2007non} developed diffusion
equations for evolution of SFS in time series, but they are difficult
to solve. Here, we explicitly derive many of the common test
statistics as functions of frequency of the favored site in the form
$S_t=f_S(\nu_t,\nu_0)$, where $\nu_t$ itself can be written as a
function of $s,t$ (Eq.~\ref{eq:nut}). This allows us to compute
$\Lc_S(s,h; \{S_t\})$ for many SFS based statistics. Then, a
likelihood ratio, similar to Eq.~\ref{eq:lrt}, provides\footnote{The
  likelihood of the data to the model is the least-squares loss
  between model $D_t(\hat{(s)})$ and the observed $D$.} a predictor
for detecting selection in each window.
\begin{description}
\item [Tajima's D.] Let $D_t$ denote the value of Tajima's D in a hard
  sweep at time $t$. We show (Appendix \ref{app:td}), that
  \begin{equation}
    D_t=D_0-\log(1-\nu_t) \frac{W_0}{\log(2N)} -\nu_t^2 \Pi_0
    \label{eq:tdt}    
  \end{equation}
  where $W_0$ and $\Pi_0$ are Watterson and Tajima estimates of
  $\theta$ at the the initial generation.  
\item[Fay Wu's H.] We show (Appendix \ref{app:h}), that the dynamics
  of the $H$ statistic are directly related to average of Haplotype
  Allele Frequency (HAF) score~\cite{ronen2015predicting}, and can be
  written as a function of $\vu_t$ as follows:
  \begin{equation}
    nH_t= \theta \nu_t \left(\frac{\nu_t+1}{2} -
      \frac{1}{(1-\nu_t)n+1}\right) + \theta
    (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)
    \label{eq:ht}
  \end{equation}	
%  This good approximation only in hard sweep with strong
%  selection, Figures~\ref{fig:dynamic-weak},\ref{fig:dynamic-strong} (second row, left).
	
\item[SFSelect.] The SFSelect statistic was proposed by Ronen et
  al.~\cite{ronen2013learning} to predict selection by empirically
  learning the so-called \emph{optimal} weights using Support Vector
  Machines.
\end{description}

\paragraph{$p$-values}
By Wilks’ theorem~\cite{williams2001weighing}, the
standard likelihood ratio statistic (Eq.~\ref{eq:mcts} without
$\sgn(s^*)$ factor) is asymptotically distributed according to
$\Xc^2$. However, Feder et al.~\cite{feder2014Identifying} pint out
that the empirical distribution provide more accurate $p$-values than
$\Xc^2$ when the number of independent samples (replicates) is small.
Here we compute $p$-value for the test statistic using the empirical
distribution of the negative controls.


\paragraph{Dominance.}
The value of the overdominance parameter can provide an insight into the kind of adaptation \emph{using population frequency data}\footnote{It is trivial for genotyped data.}. In fact, for $s>0$ we have \cite{gillespie2010population}  
\begin{center}
	\begin{tabular}{l|c}
		condition & comment\\
		\hline
		$h<0$ &  underdominance\\
		$h=0$ & recessive adaptive allele\\
		$h=0.5$ & directional selection\\
		$h=1$&	dominant adaptive allele	\\
		$h>1$ &overdominance
		\end{tabular}
		\end{center}
		
\paragraph{Software.}
Pre-computation of 1313 transition matrices for $s\in\{-0.5,-0.49,\ldots,0.5 \}$ and $h\in \{-1,-0.75,\ldots,2\}$ took in second on a desktop computer with a Core i7 CPU and 16GB of RAM.



\subsection{Simulations}
For each experiment a diploid population is created and evolved as 
follows. 
\begin{enumerate}[I.]
	\item {\bf Creating initial founder line haplotypes}
	First using msms program, we created neutral populations for $F$ 
	founding 
	haplotypes with \emph{default} parameters \texttt{\$./msms <F> 1 
	-t 
		<2$\mu$LNe> 
		-r <2rNeL> 
		<L>} 
	where $F=200$ is number of founder lines, $N_e=10^6$ is 
	effective 
	population size, $r=2*10^{-8}$ is recombination rate and 
	$\mu=2\times 
	10^{-9}$ is mutation rate and  $L=50K$ is the window size in 
	base pairs 
	which gives $\theta=2\mu N_eL=200$ and $\rho=2N_erL=2000$. 
	For 
	default 
	parameter, the expected number of segregating sites in a window 
	is 
	\beqq
	\Ebb[M]=\theta \sum_{i=1}^{F-1}\frac{1}{i}=1175
	\eeqq
	\item{\bf Creating initial diploid population} 
	To implement similar setting for experimental evolution of diploid 
	organisms, 
	initial  haplotypes first cloned to create $F$ diploid homozygotes. 
	Then 
	each 
	diploid individual is  cloned $N/F$ times to yield diploid 
	population of 
	size 
	$N$.
	\item{\bf Forward Simulation}
	Given initial diploid population, position of the site under 
	selection, 
	selection 
	strength $s$, number of replicates $R=3$, recombination rate 
	$r=2\times10^{-8}$ 
	and sampling times $\Tc=\{10,20,30,40,50\}$, \texttt{simuPop} is 
	used to 
	perform
	forward simulation and  compute allele frequencies for all of the 
	$R$ 
	replicates. Also, to avoid spurious simulation samples, simulation 
	results 
	are constrained to those that the beneficial allele escapes 
	stochastic loss 
	of genetic drift and \emph{establishes} in all the replicates. 
\end{enumerate}



\section{Results}
\ignore{
\subsection{Tests for detecting selection on time series data.}
In general, tests of selection on dynamic data use generative models
for null/alternative hypotheses to compute likelihoods of data and
perform hypothesis testing. As computing exact likelihood of the
(multiple loci) data is computationally intractable, either composite
likelihood or likelihood of single locus is computed. In this part we
first introduce a single locus statistic and then generalize it to
compute composite likelihood of a genomic region.
}
\paragraph{Modeling neutral trajectories in finite populations.} 
We tested the closeness of fit for the Markov Likelihood as a model
for neutral trajectories, compared to Brownian motion. We performed
$150$K simulations for different values of $\nu_0$
($\nu_0\in\{0.005,0.1\}$) and time $\tau$ generations $\tau\in
\{1,10,100\}$. Figure~\ref{fig:markov} shows that Brownian motion is
inadequate when $\nu_0$ is far from $0.5$, and when sampling is done
after many generations $\tau>1$. (sampling times are sparse). In most
experimental evolution scenarios, a site is unlikely to have frequency
close to $0.5$, and the starting frequencies are usually much
smaller. Moreover, sampling times are sparse, with sampling done
between $10$ and $100$ generations in \emph{Drosophila}
experiments~\cite{orozco2012adaptation, zhou2011experimental}.

In contrast, Figure~\ref{fig:markov}A-F also shows that Markov
Likelihood predictions (Eq.~\ref{eq:mkvs}) are highly consistent with
empirical data for a wide range of simulation parameters. We also
tested the predictions under selection regime by conducting $100$K
simulations with selection strength $s=0.1$ on a site with initial
frequency $\nu_0=0.005$ and sampling after $\tau$
$(\tau\in\{1,10,100\})$ generations. The empirical and theoretical
distributions tracked closely (Figure~\ref{fig:markov}G-I).

\paragraph{Power.} We compared the detection power of $\Hc$, $\Mc$
Gaussian process (GP)~\cite{Terhorst2015Multi},
FIT~\cite{feder2014Identifying} statistics.  For each experiment,
(specified with values for selection coefficient $s$, starting allele
frequency $\nu_0$, coverage $\lambda$, sampling time schedule ${\cal T}$, and 
number of
replicates $R$), we conducted $1000$ simulations. Half of these
modeled neutral evolution and the rest were under selection. Define
power of a statistic as the area under Receiver Operating Characteristic (ROC) 
curve, when false positive rate is less that 0.05, see Figure \ref{}. 
\ignore{
average true-positive rate when
false-positive rate is less than $0.05$ over $1000$ simulations. In
other words, a cutoff was set so that  $0,1,\ldots,25$ of $500$ neutral
simulations were predicted as being under selection, and the fraction of true 
positives are averaged for all 26 cutoffs.}

\paragraph{Composite Likelihood}
Before computing against other methods, we first evaluate the use of
HMM and Markov chain composite likelihoods of under different levels of 
ascertainment bias. For this comparison, we calculated power for composite 
likelihood of HMM and Markov with $\pi \in \{0,0.99,100\}$ when $\lambda\in 
\{30,100,\infty\}$.
As shown in Figure~\ref{fig:powerCLR}, 
performance of HMM is robust to the noise of finite coverage, while Markov 
chain's power decay quickly as coverage depth decreases.
Also, the best composite of likelihoods achieved when likelihoods  are averaged 
over all the SNPs in the window ($\pi=0$).
It should be noted that when $\pi=100$ the composite likelihood statistic is 
the maximum of likelihoods of the individual SNPs in the window. From 
table~\ref{tab:powerCLR}, it is evident that as detection performance is better 
and more robust as composite of likelihood is computed over all the loci in a 
window in both hard and soft sweep. Henceforth, we use $\pi=0$ to compute 
composite of likelihoods and only report performance of HMM as it is a 
generalization of Markov chain method.


To show the performance of HMM, FIT and
GP under different regimes, we computed power for different levels of ascertainment bias under hard and soft sweep.
To incorporate ascertainment bias, depth of each marker is identically and independently distributed according to Poisson($\lambda$) for  $\lambda=\{30,100,\infty\}$\footnote{Note that $\lambda=\infty $ corresponds to the case where exact population allele frequencies are given.}. All the methods except $\Hc$, convert read counts to allele frequencies and compute their test statistic.
As shown in Figure~\ref{fig:power} the HMM provide much better and more stable 
than GP and FIT (Table~\ref{tab:power}). 

\paragraph{Running Time.}
A \comale\ does not compute full likelihoods, or explicitly model
linkage between sites, the complexity of computing likelihoods is
$\Oc(TR)$, and can be efficiently vectorized for multiple replicates
and loci. Therefore, it is expected to be faster than other approaches
like Gaussian Process (GP)~\cite{Terhorst2015Multi}.

We conducted $1000$ simulations and measured running time for \comale\ 
and GP. \comale\  is $\sim 1000\times$ faster than single locus GP
(Figure \ref{fig:runTime}), while maintaining high power (Figure
\ref{fig:power}).


\paragraph{SFS for Detection in Natural Experimental Evolution.} See
figure \ref{fig:powerSFS} We did not show the SFS based statistics in
Figure~\ref{fig:power} as they do not perform better than random hypothesis. In the specific
mode of experimental evolution, we sample a restricted set of $F$
founder lines. Here, $F<<N_e$ (Figure ~\ref{fig:ee} right) creating a severe
bottleneck, and this bottleneck confounds
SFS. Figure~\ref{fig:bottleneck} demonstrates the effect of
experimental evolution on different SFS statistics under neutral
evolution for 1000 simulations. 

To show performance of SFS-based statistics in natural experimental evolution, we forward conducted 200 (100 neutral and 100 sweep) simulations for every configuration with $N_e=10K$ and accumulating new mutations. In this experiment we let start of sampling to be chosen randomly.
First we consider the case when start of sampling is chosen throughout the whole sweep, i.e., 
 $\tau_1 \sim U\left[1,t_{\nu=1}(s,N_e)\right]$ (Figure~\ref{fig:powerSFS} left) and then we tested chosen in the \emph{finale} of the sweep, i.e., $\tau_1 \sim U\left[t_{\nu=0.9}(s,N_e),t_{\nu=1}(s,N_e)\right]$ (Figure~\ref{fig:powerSFS} right), where $t_{\nu=x}(s,N_e)$ to be understood as the number of expected generations required to reach carrier frequency $x$ in a hard sweep and $U[a,b]$ is discrete uniform distribution.
We compared $\Hc$, GP, FIT with both static and dynamic SFS based statistics of SFSelect and Tajima's D. Figure~\ref{fig:powerSFS} left shows that SFS based statistics are outperformed by single locus and CLR methods. However, when sampling is performed close to fixation, i.e., when beneficial allele has frequency of 0.9 or higher, SFS based statistics significantly perform better than GP, FIT and $\Hc$ (Figure~\ref{fig:powerSFS} right). Moreover, dynamic SFS statistics outperform static SFS statistics.


\paragraph{Locating the Adaptive Mutation.}
The secondary task in identifying selection is to locate the position
of the adaptive allele. We simply consider the site with highest score
in the window as the locus of the beneficial allele. For each setting
of $\nu_0$ and $s$, we conducted 500 simulations
and computed the rank of the beneficial mutation in each
simulation. We plotted the cumulative distribution of the rank in
Figure~\ref{fig:rank}.

\comale\  works the best in hard sweep regime. For example, when
$\nu_0=0.005$ and $s=0.01$ (weakest selection), the beneficial allele
was ranked first in more than 60 experiments, and was ranked $\le 5$
in over $90$\% of experiments of hard sweep. The accuracy decreased of
locating the adaptive allele diminished in soft-sweep scenarios
(larger values of $\nu_0$). Yet, in the worst case the beneficial
allele was ranked among top 50 SNPs.

\paragraph{Strength of Selection.}
As the likelihood calculation is model based, we can also output the
model parameters $\hat{s}$ that maximized
likelihood~\eqref{eq:hmmlik}. We computed bias, $s-\hat{s}$ for each
experiment of \comale, and GP. The distribution of the bias is
presented in Figure \ref{fig:bias} for different configurations. In
general, both GP and \comale\ have biased results for weak selections,
where genetic drift dominates. However, for stronger sweeps,
e.g.$s=0.1$, \comale\ provides estimates with smaller bias and
variance.


\paragraph{SFS based statistics in time series.} 
We also considered SFS based tests including Tajima's D, Fay \& Wu's H and 
SFSelect for detecting selection. 
As these tests work only on static data, we extend them for time series data 
for both null and alternative hypotheses.
More precisely, we explicitly defined functionals $D_t$ and $H_t$ as function 
of initial carrier frequency $\nu_0$ and strength of selection $s$ (see section 
\ref{sec:sfs-ts} for details).
 To show that the proposed models of $D_t$ and $H_t$ are valid models, we 
 simulated 1000 populations and computed SFS based statistics every 10 
 generation and compared them with the proposed model.
 As shown in the Figures \ref{fig:sfs} proposed models are more consistent with data.

\paragraph{SFS based tests will fail when carrier frequency $\nu_t$ is low.}
Importantly,  Figures \ref{fig:sfs} shows the increase in variance of the trajectories associated with SFS, compared to carrier frequency. Moreover, it is difficult to distinguish SFS trajectories of selection and neutral populations, when carrier frequency is small, e.g. first 50 generations of Figures \ref{fig:sfs}.
This, observation can be verified by examining the terms in the functional form of $D_t$.
As shown in the Figure 
\ref{fig:tdterms} right, in early 
generations of hard sweep where carrier frequency is low, 
$D_t$ is either 
positive or close zero. In other words, the reduction in 
diversity become 
significant when carrier frequency is high enough.
It can be shown that this argument holds for $H_t$ in early generations 
of hard sweep, where carrier frequency is not high.




\subsection{Analysis of Real Data}
We finally apply \comale\ method to the controlled experimental evolution 
experiment 
of \cite{orozco2012adaptation}, which evolves 3 replicates of a population of 
Drosophila melanogaster for 59 generations under alternating 12-hour cycles of 
hot (28$^{\circ}$C) and cold (18$^{\circ}$C) temperatures and sequenced at 
different generation for each replicate, Figure~\ref{fig:samplingTimes}.
\paragraph{Heterogenocity of read depths.} \comale\ statistic is basically 
computed using site allele frequencies at different generations for all 
replicates. For real Pool-Seq data, however, allele frequencies is unknown and 
only read counts at each site measured. As shown in Figure 
\ref{fig:depthHetero}, \ref{fig:depth}  read depths 
are highly heterogeneous and filtering low coverage sites from data strictly 
reduces the number of SNPs. For example, by setting minimum read depth at 
each site (for all replicates and generations), the number of SNPs shrinks from 
1,544,374 to 10,387. 

\paragraph{Scan.} 
We computed \comale statistic for sliding window of 50Kbp with steps of 10Kbp 
over the whole genome  and observed that there is a negative correlation 
between $\Hc$ and the number of SNPs in each  window, 
Figure~\ref{fig:manhattan}.
More precisely Pearson correlation is $\rho(M,\Hc)=-0.03$ when all the windows 
are taken into account and $\rho(M,\Hc)=-0.27$ when restricting to the 
candidate regions. This nine-fold increase in correlation indicates that $\Hc$ 
statistic takes more extreme values when SNPs are depleted and the number of 
variants are less than expected ($\approx$1100).
To remedy this, we filtered out those windows with less than 500 SNPs and 
Pearson correlation changed from $\rho(M,\Hc)=-0.08$ to $\rho(M,\Hc)=-0.09$, 
see Figure~\ref{fig:manhattancutoffed}.
We identified 25 contiguous intervals (see 
\texttt{data/intervals.tsv}) providing signal of 
selection.

\paragraph{Gene Set Enrichment on Regions under Selection}\AI{This favors 
longer shorter lists, i.e. specific GO terms}
We found 243 out of the total 13,965 variant genes to be in the regions under 
selection and applied Fisher exact test for gene lists corresponding to the 
existing Gene Ontology terms~\ref{tab:Fisher}.


\paragraph{Gene Set Enrichment on Whole Genome} \AI{This favors longer gene 
lists, i.e. general GO terms}
We also performed another enrichment analysis on the all variant genes without 
excluding any gene as follows. For each gene we computed \comale score on the 
\emph{transcript}, \emph{synonymous-excluded}, \emph{exon}, and \emph{loss of 
function}
 SNPs with $s*\ge0$ to favor genes under positive selection. Given each subset 
 of SNPs \comale computes a different score for each gene. By ranking all the 
 variant genes according to their \comale score, we computed $p$-value for each 
 gene list using Mann–Whitney U test, aka Wilcoxon rank-sum test, 
 Tables~\ref{tab:transcriptGO},~\ref{tab:nonsynonymousGO},
~\ref{tab:exonGO},~\ref{tab:lofGO}.

Interestingly, GO terms corresponding to membrane are enriched which are known 
to be responsive to thermal adaptation\cite{hazel1995thermal}.


\input{method}
\newpage
\section{Figures}

\begin{figure}[H]
	\centering
	\includegraphics[trim={1in 0.1in 0in 
		0in},clip,width=\textwidth]{figures/{markovDists}.pdf}


              \caption{{\bf Comparison of empirical distributions of
                  allele frequencies (green) versus predictions from
                  Brownian Motion (red), and Markov
                  Likelihoods}. Panels A-F: Experiments were conducted
                under neutral evolution with different starting
                frequencies $\nu_0\in\{0.005,0.1\}$ and sampling times
                $\tau \in \{1,10,100\}$ generations. The empirical
                distribution was computed by sampling 143,900 sites
                with $\nu_0=0.005$ and 47,500 with variants
                $\nu_0=0.1$.  computed from neutrally evolving
                simulations is depicted in green lines. Panels G,H,I:
                Comparisons of Empirical and Markov chain based
                predicted allele frequency value distributions under a
                selection regime with $s=0.1$. Initial frequency was
                chosen as $\nu_0=0.005$ and sampling performed after
                $t=\{1,10,100\}$ generations.}
	\label{fig:markov}
\end{figure}

\clearpage
\newpage



\begin{figure}[H]
	\centering
	\begin{tabular}{cc}
	\includegraphics[trim=0.1in 0 .08in 0.02in , clip,width=0.5\textwidth]{figures/ControlledExperimentalEvolution.pdf}
&	\includegraphics[trim=0.1in 0 .08in 0.02in , clip,width=0.5\textwidth]{figures/NaturalExperimentalEvolution.pdf}
	\end{tabular}
	\caption{Controlled (left) and Natural (right) experimental evolution. } \label{fig:ee}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=0.4in 0 .8in 0.02in , clip,width=\textwidth]{figures/powerCLR.pdf}
	\caption{Power, average true-positive-rate when FDR$\le$0.05, for detecting selection in a 50Kbp region for Markov Chain ($\Mc$) on 1000 
		simulations for each of selection strength $s$ and initial 
		carrier frequency $\nu_0$, and composite likelihood ratio cutoff $\pi$.  } \label{fig:powerCLR}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=0.4in 0 .8in 0.02in , clip,width=\textwidth]{figures/power.pdf}
	\caption{Power, average true-positive-rate when FDR$\le$0.05, for detecting selection in a 50Kbp region for Frequency Increment Test (FIT), Gaussian Process (GP), Markov Chain ($\Mc$), Hidden Markov Model ($\Hc$) on 1000 
		simulations for each of selection strength $s$ and initial 
		carrier frequency $\nu_0$. To incorporate ascertainment bias, each setting is evaluated such that depth of each SNP is identically distributed from Poisson$(\lambda)$ for $\lambda \in \{30,100,\infty\}$. Average performance of each method is shown in Table \ref{tab:power}. Note that $\Hc$ and $\Mc$ become identical when $\lambda=\infty$.  } \label{fig:power}
\end{figure}
\begin{table}[h]
	\begin{tabular}{c||c}
		Hard Sweep & Soft Sweep\\ \\  
		\input{tables/powerHard.tex}
				&\input{tables/powerSoft.tex}
	\end{tabular}
	\caption{Average power, average true-positive-rate when FDR$\le$0.05, for 
	detecting selection in a 50Kbp region for Frequency Increment Test (FIT), 
	Gaussian Process (GP), Markov Chain ($\Mc$), Hidden Markov Model ($\Hc$) on 
	1000 
		simulations for each of selection strength $s$ and initial 
		carrier frequency $\nu_0$. To incorporate ascertainment bias, each 
		setting is evaluated such that depth of each SNP is identically 
		distributed from Poisson$(\lambda)$ for $\lambda \in 
		\{30,100,\infty\}$.}\label{tab:powerCLR}
\end{table}

\begin{table}[h]
	\centering
	\begin{tabular}{ccc}
		Hard Sweep & &Soft Sweep\\ \\  
		\input{tables/powerHardMathods.tex}
		&&\input{tables/powerSoftMethods.tex}
	\end{tabular}
	\caption{Average power, average true-positive-rate when FDR$\le$0.05, for 
		detecting selection in a 50Kbp region for Frequency Increment Test 
		(FIT), 
		Gaussian Process (GP), Markov Chain ($\Mc$), Hidden Markov Model 
		($\Hc$) on 
		1000 
		simulations for each of selection strength $s$ and initial 
		carrier frequency $\nu_0$. To incorporate ascertainment bias, each 
		setting is evaluated such that depth of each SNP is identically 
		distributed from Poisson$(\lambda)$ for $\lambda \in 
		\{30,100,\infty\}$.}\label{tab:power}
\end{table}
\newpage
\begin{table}[h]
	\begin{tabular}{c}
		\input{tables/transcriptGO}
	\end{tabular}
	\caption{GO enrichment when all the SNPs in the genes are taken into 
	account. .}\label{tab:transcriptGO}
\end{table}
\begin{table}[h]
	\begin{tabular}{c}
		\input{tables/nonsynonymousGO}
	\end{tabular}
	\caption{GO enrichment when synonymous variants are 
	excluded.}\label{tab:nonsynonymousGO}
\end{table}

\begin{table}[h]
	\begin{tabular}{c}
		\input{tables/exonGO}
	\end{tabular}
	\caption{GO enrichment on missense variants.}\label{tab:exonGO}
\end{table}
\begin{table}[h]
	\begin{tabular}{c}
		\input{tables/lofGO}
	\end{tabular}
	\caption{GO enrichment on loss of function variants.}\label{tab:lofGO}
\end{table}


\begin{table}[h]
	\begin{tabular}{c}
		\input{tables/transcriptFisher}
	\end{tabular}
	\caption{Gene set enrichment of 243 genes within 25 intervals under 
	selection using Fisher exact test.}\label{tab:Fisher}
\end{table}

\newpage
\begin{figure}[H]
	\centering
	\includegraphics[trim=.4in 0 .8in 0.6in , clip,width=\textwidth]{figures/naturalee.pdf}
	\caption{Power, average true-positive-rate when FDR$\le$0.05, for detecting selection in a 50Kbp region for Frequency Increment Test (FIT), Gaussian Process (GP), Markov Chain ($\Mc$)on 200 
		simulations hard-sweep natural experimental evolution with $N_e=10^4$, for each of selection strength $s$. In this experiment start of sampling is random, then 5 samples are taken every 10 generations. Start of sampling is chosen randomly throughout the sweep $\tau_1 \sim U\left[1,t_{\nu=1}(s,N_e)\right]$ (left) and chosen in the finale of the sweep $\tau_1 \sim U\left[t_{\nu=0.9}(s,N_e),t_{\nu=1}(s,N_e)\right]$ (right), where $t_{\nu=x}(s,N_e)$ to be understood as the number of expected generations required to reach carrier frequency $x$ in a hard sweep and $U[a,b]$ is discrete uniform distribution.} \label{fig:powerSFS}
\end{figure}

\newpage
\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{figures/rank.pdf}
	\caption{Cumulative Distribution  Function (CDF) of the distribution of the rank of the adaptive allele in 500 simulations for Markov Chain ($\Mc$), Gaussian Process (GP) and Frequency Increment Test (FIT), for different values of selection strength $s$ and initial carrier frequency. Area Under Curve (AUC) is computed as a quantitative measure ranking performance of methods for each configuration.}
	\label{fig:rank}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=.0in 0 .2in 0 , clip,width=\textwidth]{figures/bias.pdf}
	\caption{Distribution of bias ($s-s^*$) over 500 simulations for each value of strength of selection $s$ and starting carrier frequency $\nu_0$.} \label{fig:bias}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[width=0.75\textwidth]{figures/{runTime.png}}
	\caption{Average running time of \comale, HMM, GP with single,3,5,7 and 10 loci over 1000 simulations. Average running time for each method is shown in the horizontal axis ticks.} 
	\label{fig:runTime}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/manhattan.pdf}
	\caption{Manhattan plot of the composite $\Hc$ statistic (top) 
		and the number of SNPs (bottom) in 50Kbp sliding window with steps of 
		10Kbp. Regions in which their $\Hc$ statistic falls in top one 
		percentile 
		distribution is denoted with red color. Pearson correlation between $M$ 
		and 
		$\Hc$ of all windows is -0.03 and for the candidate regions is -0.27. 
		This 
		nine-fold increase in correlation implies that the $\Hc$ statistic 
		takes 
		more extreme values as the number of SNPs in the window is less than 
		expected ($\approx$1100).} 
	\label{fig:manhattan}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/{manhattan.min500snp}.pdf}
	\caption{Manhattan plot of the composite $\Hc$ statistic (top) 
		and the number of SNPs (bottom) in 50Kbp sliding window with steps of 
		10Kbp, excluding windows with less that 500 SNPs. Regions in which 
		their $\Hc$ statistic falls in top one percentile 
		distribution is denoted with red color. Pearson correlation between $M$ 
		and 
		$\Hc$ of all windows is -0.09 and for the candidate regions is -0.08. 
		This indicates the spurious effect of windows with low diversity is 
		removed from out study (potentially at cost of some false negatives). } 
	\label{fig:manhattancutoffed}
\end{figure}


%\input{intro}
%\input{method}
%\input{experiment}
%\input{conclusion}
%\newpage
%\bibliographystyle{acm}
%\bibliography{library}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%% Supplemental Material%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
%\newpage
%\setcounter{page}{1}
\setcounter{figure}{0}
\setcounter{table}{0}
\setcounter{equation}{0}
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
\renewcommand{\theequation}{S\arabic{equation}}

\section{Appendix}
\subsection{Allele frequencies under selective sweep} \label{app:af}

\beq
\eeq
where $s\in \Rbb$ is the selection coefficient and $o\in[0,1]$ is the 
overdominance parameter which for $u=0.5$ we have
\begin{equation}
x_{t+1}=x_t+\frac{sx_t(1-x_t)}{2+2sx_t}\;.
\label{eq:transition}
\end{equation}
we also have
\beq
\frac{\bfd x_t}{\bfd t} = \frac{sx_t(1-x_t)}{2+2sx_t}
\eeq
which is a differential equation that is difficult to solve. However if take 
the approximation $2+2sx_t \approx 2$, it becomes an ordinary differential 
equation that can be readily solved

\begin{equation}
\nu_t =\frac{1}{1+\frac{1-x_0}{x_0}e^{-st/2}} = \sigma(st/2+\eta(x_0)) 
\label{eq:inf-pop}
\end{equation}
where$\sigma(.)$ is the logistic
function and $\eta(.)$ is logit function (inverse of the logistic function). 

\subsection{Logistic Model for Selection.}
As maximum likelihood for estimating $s$ using Markov chain is computationally 
expensive, here we use and logistic function for modeling allele frequencies 
undergoing selective sweep.
In a pure selection process with no drift, i.e. infinite 
population size, 
dynamic of allele frequencies 
can be well approximated by the logistic function (see 
Appendix \ref{app:af} 
for 
derivation)
\beq
\nu_t=\sigma(st+\eta(\nu_0))\label{eq:nut}
\eeq
where $\sigma(x)=1/(1+e^{-x})$ is the logistic function, and 
$\eta(x)=\log(x)/\log(1-x)$ is inverse of the 
logistic, aka logit, function. Figure \ref{fig:sweep} depicts 
the behavior of 
the logistic model for site allele frequencies and the 
default sampling time 
span(STS). Without genetic drift, soft sweep is easier to 
detect, 
because the logistic function happens to have steeper slope 
in th STS than 
those of hard sweeps, due to standing variation frequency. In addition, even 
under infinite 
population size regime, it is 
difficult differentiate between weak selections and genetic 
drift (a horizontal 
line), in short (e.g. 50 generations) experimental evolutions.


As shown in Figures \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (first 
row), the approximate logistic model is consistent with 
simulated data and we 
use it to estimate the strength of selection for 
each site by solving a linear system of equations(see Section 
\ref{sec:regression} for details).

\subsection{Fay Wu's H}\label{app:h}
%\bl
In any finite population size of $n$ with $m$ segregating sites, 
allele frequencies take 
discrete values, i.e.,  $x_j \in 
\{\frac{1}{n},\frac{2}{n},\ldots,\frac{n-1}{n}\}, \ \forall j \in{1,\cdot,m}$ 
and we can write
\beq
\|\bfx\|^2= \sum_{j=1}^{m} x_j^2 = 
\sum_{i=1}^{n-1}\left(\frac{i}{n}\right)^2\xi_i= 
\frac{ (n-1)}{2n}H 
\eeq
where $\xi_i$ is the number of sites with frequency $i/n$ and $H$ is the 
Fay \& Wu's estimate of $\theta$.
%\el

Recently, Ronen et al. ~\cite{ronen2015predicting} devised the $1\dHAF$ 
statistic for identifying selection on static data, which has the expected 
value related to the $\|\bfx\|^2$:
\begin{equation} 
\Ebb[1\dHAF(t)]= n\| \bfx_t\|^2\approx ng(\nu_t)
\end{equation} 
where
\beq
g(\nu_t)= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)
\label{eq:hafscorepooled}
\eeq
where easily follows that
\beq
\theta_H(t)=\frac{n-1}{2} g(\nu_t)
\eeq

\subsection{Tajima's D}\label{app:td}
Let $D_0, \Pi_0, W_0$, be Tajima's D, Tajima's estimate of  $\theta$, and 
Watterson's estimate of $\theta$ at time zero and $D_0=\Pi_0 - W_0$.
In order to compute, $D_t=\Pi_t - W_t$ we compute $\Pi_t$ and $W_t$ separately 
as follows.

Let $P$ be the $n \times n$ matrix of pairwise heterozygosity if individuals, 
then $\Pi=\frac{1}{n^2}\sum P_{ij}$. So, if the population consist of $\nu n$ 
identical carrier haplotype (due to lack of recombination), their pairwise 
hamming distance is zero and should be subtracted from the total $\Pi_t$:
\beq
\Pi_t&= (1-\nu_t^2)\Pi_0 
\eeq

To compute $W_t$, first remember that $W_t= \frac{m_t}{S_n}$ where $m_t$ is the 
number of segregating sites at time $t$ and $S_n= \sum_i^n 1/i \approx 
\log(n)$. Also we have
\beq
\frac{W_t}{W_0}&=\frac{\frac{m_t}{S}}{\frac{m_0}{S}} \ \ \Rightarrow 
W_t=\frac{m_t}{m_0}W0 
\eeq
where $m_t$ to be interpreted as the expected number of segregating sites at 
time $t$, under neutral evolution. At time $t$, the number of individuals that 
undergone neutral evolution is $(1-\nu_t)n +1$, which leads to
\beq
\frac{m_t}{m_0}&=\frac{\log\left((1-\nu_t)n +1 \right)\theta}{\log(n)\theta} 
\approx  
\frac{\log\left((1-\nu_t)n\right)}{\log(n)} = \frac{\log(1-\nu_t)+\log(n)}{\log(n)} = 
1+ \frac{ \log(1-\nu_t)}{\log(n)} 
\eeq
putting all together 
\beq
D_t&= (1-\nu_t^2)\Pi_0 - (1+ \frac{ \log(1-\nu_t)}{\log(n)} ) W_0 = 
D_0-\log(1-\nu_t) \frac{W_0}{\log(n)} -\nu_t^2 \Pi_0
\eeq


\subsection{Linkage Disequilibrium}
Nonrandom associations between polymorphisms are established in the 
substitution process according to the phylogeny, broken by recombination events 
and reinforced by selection. Although in EE the experiments with pooled 
sequencing, LD can not be measured throughout evolution, it is still worthwhile 
to examine the behavior of LD as a result of the interaction between 
recombination and natural selection, to take into account of some of EE 
implicit constraints. 

Let $\rho_0$ be the LD at time zero between the site under selection and a 
segregating site $l$ base-pairs away, then under natural selection we have
\beq
\rho_t= \alpha_t\beta_t \rho_0=e^{-rtl} \left(\frac{H_t}{H0}\right)  
\rho_0\label{eq:ldt}
\eeq
where $H_T=2\nu_0(1-\nu_0)$ is the heterozygousity at the selected site, $r$ is 
the recombination rate/bp/gen. The decay factor $\alpha_t=e^{-rtl}$ is the 
product of recombination and growth factor $\beta_t$ (eq. 30-31 in 
\cite{Stephan2006The})is the outcome of 
selection. For $s=0.01$, $l=100K$bp, the log of decay, growth and product of 
both is depicted in Figure \ref{fig:ldf}. It is evident that, for these 
parameters LD does not start to decay until generation 1000, which would be  
problematic when $\rho_0$. For example, in the case of hard sweep, the 
selection is imposed on the site with minimum AF, which is at perfect linkage 
($|D'|=1$) with all the other loci.\footnote{This is because, between the 
	selected site and all the other sites frequency of one gamete is zero.}
We this phenomenon is shown in the Figures \ref{fig:ld2d}, \ref{fig:ld3d} where 
at generation zero the site at position 500K is at perfect linkage with all the 
other sites, and linkage of the middle site with all the genome is depicted 
for both genetic drift and natural selection, in different generations.
Also, a window of 50Kbp around the selected site is shaded in Figure 
\ref{fig:ld2d} to demonstrate the value of LD in the window under drift and 
hard sweep. This implies that the precision of locating the selection on the 
genome is tightly dependent on a set of parameters including, recombination 
rate, selection strength, initial carrier frequency, and the initial linkage.
\subsection{Likelihood Functions} \label{app:likelihood}
Actually, it is enough to show \eqref{eq:transition} is
linear fractional in $s$, which is nby definition: linear numerator
and denominator, and strictly positive denominator.
\clearpage
\newpage
\section*{Supplemental Figures}



\begin{figure}[H]
	\centering
	\includegraphics[trim=1in 0.1in 1in 0.1in,clip,width=\textwidth]{figures/sfs.pdf}
	\caption{Theoretical and Empirical SFS for a neutral population of 200 
	individuals.}	\label{fig:sfs}
\end{figure}

\begin{figure}[H]
	\includegraphics[trim=0in 0.1in 1in 0.1in,clip,width=\textwidth]{figures/AF.pdf}
	\caption{Logistic model for different selection strengths 
		for soft (left) 
		and hard (right) sweep as a function of time in 
		generations. The first 
		50 
		generations, which observations are sampled is 
		shaded.} 	 
	\label{fig:sweep}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim={0in 0.1in 0in 
		0in},clip,width=\textwidth]{figures/{tdterms}.png}
	\caption{Interactions of two terms in $D$. W.l.o.g when $D_0=0$ and 
	$\Pi_0=W_0=1$, 
		$D_t$ is sum of the 
		logarithmic $\frac{-\log(1-\nu_t)}{\log(2N)}$ and the squared term 
		$\nu_t^2$.} 	 
	\label{fig:tdterms}
\end{figure}

\begin{figure}[H]
	\centering 
	\includegraphics[trim=3.2in 0.1in 3.2in 0.2in , 
	clip,width=\textwidth]{figures/bottleneck}
	\caption{Effect of bottle neck in a typical experimental evoloution 
		experiment where a restricted number of founder lines (here $F=200$) is 
		selected out of a larger population size ($N_e=10^{-6}$). Tajima's D 
		(left), Fay Wu's H (middle) and SFSelect is computed for 1000 neutral 
		simulations and mean and 95\% confidence interval is plotted.} 
	\label{fig:bottleneck}
\end{figure}

\begin{figure}[H]
	\centering 
	\includegraphics[width=\textwidth]{figures/{msmsts}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for neutral (blue trajectories) selection with $s=0.1$ (red trajectories).}
	\label{fig:sfsts}
\end{figure}



\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/decayFactors}
	\caption{Interaction between productive factors of LD under natural 
		selection for weak selection (s=$0.01$) and a distance of 100Kb between 
		sites. In this setting, after about 1000 generations LD start to decay 
		(red 
		curve).} \label{fig:ldf}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/LDDecay2d}
	\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at position 
		500K with the rest of genome when $s=0.01$ and $r=2\times10^{-8}$. A 
		window 
		of 50Kb is shaded at the center of genome to illustrate high values of 
		linkage in both selection and drift.} \label{fig:ld2d}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{figures/LDDecay3dNeutral}
	\includegraphics[width=\textwidth]{figures/LDDecay3dSweep}
	\caption{ld} \label{fig:ld3d}
	\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at 
		position 500K with the rest of genome in genetic drift with 
		$r=2\times10^{-8}$ (top) and hard sweep with $s=0.01$ (bottom).}
\end{figure}




\begin{figure}[H]
	\centering
	\includegraphics[trim=1.2in 0 .0in 0, 
	clip,width=\textwidth]{figures/{dominance}.png}
	\caption{Dominance for $s=0.1$.} 
	\label{fig:dominance}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .0in 0, 
	clip,width=\textwidth]{figures/{qq}.pdf}
	\caption{QQ plot for a simulation.} 
	\label{fig:qq}
\end{figure}



\begin{figure}[H]
	\centering
	\includegraphics[trim=.0in 0 .0in 0, 
	clip,width=\textwidth]{figures/{samplingTimes}.pdf}
	\caption{sampling times.} 
	\label{fig:samplingTimes}
\end{figure}



\begin{figure}[H]
	\centering
	\includegraphics[trim=.0in 0 .0in 0, 
	clip,width=0.5\textwidth]{figures/{statePosterior}.pdf}
	\caption{Posterior distribution of allele frequencies (states), for 
	different values of depth $d=\{5,50,500\}$. In all the cases, the number of 
	read counts for the derived allele is one-fifth of total depth. Dealing 
	with dynamic data (and possibly multi-replicate), each site has multiple 
	measurements, that can have different depth, i.e. uncertainty.} 
	\label{fig:stateConditional}
\end{figure}


\begin{figure}[H]
	\centering
	\begin{tabular}{ccc}
				\includegraphics[trim=.1in 0 .0in 0, 
				clip,width=0.3\textwidth]{figures/{Song}.pdf} &
		\includegraphics[trim=.1in 0 .0in 0, 
		clip,width=0.3\textwidth]{figures/{BM}.pdf} &
		\includegraphics[trim=.1in 0 .0in 0, 
		clip,width=0.3\textwidth]{figures/{BF}.pdf}
	\end{tabular}
	\caption{Song, HSF, HSP.} 
	\label{fig:geneRanking}
\end{figure}


\begin{figure}[H]
	\centering
	\begin{tabular}{c}
		\includegraphics[trim=.2in 0 .0in 0, 
		clip,width=\textwidth]{figures/{depth}.pdf}
	\end{tabular}
	\caption{Scaled PDF (left) and CDF (right) of the overall read depth distribution (top) and minimum depth of sites (bottom).
		Top row depicts the overall distribution of rad of all reads.} 
	\label{fig:depth}
\end{figure}

\begin{figure}[H]
	\centering
	\begin{tabular}{c}
		\includegraphics[trim=.2in 0 .0in 0, 
		clip,width=\textwidth]{figures/{depthHetero}.pdf}
	\end{tabular}
	\caption{Read depth at four different sites. (which would be filtered if the min depth is set to 30X)} 
	\label{fig:depthHetero}
\end{figure}

%\begin{figure}[H]
%	\centering
%	\begin{tabular}{c}
%		\includegraphics[trim=.2in 0 .0in 0, 
%		clip,width=\textwidth]{figures/{siteDepthDist}.pdf}\\
%	\end{tabular}
%	\caption{Distribution of the mean and variance of the minimum depth of all sites.} 
%	\label{fig:siteDepthDist}
%\end{figure}
\newpage
\bibliographystyle{acm}
\bibliography{library}
\newpage
\end{document}
