\documentclass[11pt]{article}
\input{macros_vineet}
\input{macros_arya}
\title{Detecting Selection in Experimental Evolution Experiments}
\author[1]{Arya Iranmehr}
\author[1]{Ali Akbari}
\author[2]{Vineet Bafna}
\affil[1]{\footnotesize Electrical and Computer Engineering, University of California, San Diego, La Jolla, CA 92093, USA.}
\affil[2]{\footnotesize Computer Science \& Engineering, University of California, San Diego, La Jolla, CA 92093, USA}
\date{}

\begin{document}
\maketitle
\begin{abstract}
Experimental evolution studies are a powerful tool for observing
molecular evolution at work in a controlled environment. The availability of
next-generation sequencing has made it possible to study the genetic trajectory 
of the population at different time points. Methods for investigating 
time-series genomic data are under active development, including recent work by 
Terhorst et al., that uses a Gaussian process approximation to a 
multi-locus Wright-Fisher process.  
In this research, we revisit the problem of detecting selection in short amount 
of time using classical methods such as
Tajima's S and Fay Wu's H and a proposed method which fits a logistic
function to the trajectories of allele frequencies at every locus. An extensive
simulation study shows the proposed method has a superior performance
over conventional method both in hard and soft sweeps.
\end{abstract}



\section{Introduction}
Genetic adaptation is \emph{the} central evolutionary process and is at the
 core of some of the greatest challenges facing humanity.  HIVwould likely
  cause nothing more than harmless fever without the
ability of the virus to adapt and eventually destroy the immune system. 
Cancer would be much more straightforward to treat if not for
tumor's ability to adapt to anti-cancer drugs. 
Malaria could be treated with cheap drugs such as quinine instead of 
being one of the world's worst killers. 
Crop pests would be manageable with small doses of safe insecticides 
instead of requiring applications of ever increasing amounts of a diverse array 
of powerful chemicals. [[Example for Bacteria and TB and some references]]
In both disease and agriculture, we find ourselves in an arms race due to the
ability of organisms to adapt.

Adaptation leaves a variety of detectable signatures in
genomes \cite{nielsen2005genomic,akey2009constructing,
kreitman2000methods,messer2013population,sabeti2006positive}.
The rapid expansion of adaptive alleles in populations leads to both an excess
of functional changes between species and distortions in polymorphism patterns
known as selective sweeps~\cite{nielsen2005genomic}. The signatures of
selective sweeps, which include reduction of levels of polymorphism and the
presence of too many rare alleles~\cite{nielsen2005genomic,
przeworski2002signature}, have been the basis for assessing genomic
patterns in many genome scans for adaptation. Depending on the
\emph{1)data availability}, e.g. allele frequency vs. haplotype, single vs 
multiple  population, single-snapshot vs time-series observation and 
\emph{2)objective of the test}, e.g. detecting recent or ancient
sweep, and \emph{3)simulation model}, e.g. statistical-test vs likelihood-based 
models, empirical recipes for detecting selection are available for each
scenario \cite{sabeti2006positive}.

For example, given allele frequency, a whole family of test such as 
Tajima's \emph{D} ~\cite{tajima1989statistical}, 
Fay and Wu's \emph{H}~\cite{fay2000hitchhiking}, 
SFSelect~\cite{ronen2013learning} compute test statistic as "\emph{a}"
\footnote{Convex combination is a linear combination which weights are positive
	 and sum to one. Under a Bayesian prospective, it can be regarded as a
	  \emph{expectation} which the prior distribution is determined by the
	   weights. As with Bayesian inference, using a \emph{not-reasonable prior}
	    distribution would lead to poor results.} 
	convex 
combination of Allele Frequency Spectrum(AFS)~\cite{achaz2009frequency} 
(Figure \ref{fig:afs}. 
However, despite their simplicity and intuitiveness, it has been shown that
 ~\cite{ptak2002evidence, ramos2002statistical} AFS-based tests often fail 
 to discriminate between positive selection and a dynamic
demographic processes. Also, more recently, Frequency Increment
 Test(FIT) \cite{feder2014Identifying} is  proposed to test whether a
  time-series of allele-frequencies is subjected to a selection pressure.
\begin{figure}
	\centering
	\includegraphics[scale=0.2]{afs}
	\caption{AFS}	\label{fig:afs}
\end{figure}
\paragraph{Experimental Evolution}
Recent advances of whole genome sequencing has enabled us to sequence 
populations at a reasonable cost, such that one could design \emph{experiments} 
to study different forces of \emph{evolution} in-action. \emph{Experimental 
evolution} is the study of the evolutionary processes of a model organism in a 
controlled  
\cite{hegreness2006equivalence,lang2013pervasive,orozco2012adaptation,
	lang2011genetic,barrick2009genome,bollback2007clonal} 
or natural 
	\cite{maldarelli2013hiv,reid2011new,denef2012situ,winters2012development,
	daniels2013genetic,barrett2008natural,bergland2014genomic} environment.

Types of evolution experiments \cite{Barrick2013Genome} [to be added]

While constraints such as small population sizes, limited timescales and 
oversimplified  laboratory
 environments limits interpreting experimental evolution results, it has
  been used to test different hypothesis~\cite{kawecki2012experimental}
   regarding
mutation and adaptation, 
genetic drift and inbreeding, 
environmental variability,
sexual selection and conflict, 
kin selection and cooperation
life history and sex allocation, 
sexual reproduction and mating systems, 
behavior and cognition, 
host–parasite interactions, 
speciation
repeatability of evolution.

In addition, time-series \emph{models} is shown to be more powerful 
\cite{boyko2008assessing,desai2008polymorphism,sawyer1992population} 
than single-snapshot data.
In particular, dynamic data has been used to estimate model parameters 
including population size
\cite{williamson1999using,wang2001pseudo,pollak1983new,waples1989generalized,
	Terhorst2015Multi}
strength of selection
\cite{mathieson2013estimating,illingworth2011distinguishing,Terhorst2015Multi,
	bollback2008estimation,illingworth2012quantifying,malaspinas2012estimating,
	Steinrücken2014a, malaspinas2012estimating}, allele age
 \cite{malaspinas2012estimating}
recombination rate~\cite{Terhorst2015Multi}  and mutation
 rate~\cite{Barrick2013Genome, Terhorst2015Multi}. 

Linked-loci time-series (spatio-temporal) methods 
\cite{illingworth2011distinguishing,illingworth2012quantifying,
	Barrick2013Genome,Terhorst2015Multi} incorporate linkage into computation
 which makes it cumbersome. Theoretical diffusion approximation method has also 
 developed for neutral allele frequencies~\cite{Ewens2012Mathematical, 
 kimura1955solution}
neutral AFS~\cite{evans2007non}, neutral allele
 frequencies~\cite{song2012simple}.

Experimental evolution (including evolve and re-sequence paradigms)
have become increasingly popular as a complementary tool to understand
the forces of selection, allowing for controlled environments,
specific selection constraints. Examples of experimental evolution
studies abound in sexual populations, particularly in fruitfly. Burke
et al.~\cite{Barrick2013Genome} evolved flies for over $600$ generations 
under
selection for accelerated development, and noted that evolution for
sexual populations is very different from those of asexual populations
(e.g., Lenski): the effects of clonal interference is slower
due to recombination that allows for the incorporation of multiple
beneficial alleles, but there are fewer, unconditionally advantageous
alleles that arise at the onset of selection. Rose and
Colleagues~\cite{rose1994evolutionary} created $200$ experimentally evolved
populations selected for different traits, starting with $10$ initial
populations. Zhou et al. evolved flies to adapt to low oxygen
environment (hypoxia) for over $300$ generations, and identified many
genes involved in hypoxia tolerance~\cite{zhou2011experimental}.  Instead, 
they observed
incomplete fixation (`soft-sweeps') due in part to standing variation,
changing selection coefficients, and small fitness effects. Similarly, Jha et
 al. \cite{jha2015whole} identified genes under selection for egg size in flies.
Barrick et al. \cite{barrick2009genome} evolved a population of 
\emph{Escherichia coli} for 40,000 generations.

Much like in natural populations, many of these studies also
sequenced/genotyped only the latest population. However, the emergence
of NGS and related technologies has made it feasible to sequence the
evolving population at multiple time points in its evolution. At the
same time, for small organisms where a single animal does not provide
enough DNA, it is more effective to pool together individuals in the
population at a particular time point, in a single sequencing
experiment. Thus, instead of individual genotypes, we obtain
frequencies of the derived allele at all loci at different time
points. Methods for analyzing such time series pooled sequence data
are only just being developed.

\paragraph{Asexual vs Sexual}
\cite{bollback2008estimation,lenski1991long,lang2013pervasive} [to be 
added]


\section{Results}
\paragraph{Notation} Let $\bfX \triangleq (X_{ijk}) \in [0, 1]^{T \times 
M\times R}$ denote the population frequency where $T$ is the number of samples 
in time, $M$ is the number of segregating sites, and $R$ is the number of 
replicates.
To simplify our notation, we define $x_t$ as a allele frequency of a site in a
 replicate, and we denote it by $\nu_t$ if such a site is directly under 
 selection with the strength $s$. I.E., $\nu_t$ can be regarded as frequency of 
 the carrier in the  population.
\paragraph{Simulations}
For each experiment a diploid population is created and evolved as follows. 
\begin{enumerate}[I.]
	\item {\bf Creating initial founder line haplotypes}
	First using msms program, we created neutral populations for $F$ founding 
	haplotypes with \emph{default} parameters \texttt{\$./msms <F> 1 -t 
	<2$\mu$LNe> 
		-r <2rNeL> 
		<L>} 
	where $F=200$ is number of founder lines, $N_e=10^6$ is 
	effective 
	population size, $r=2*10^{-8}$ is recombination rate and $\mu=2\times 
	10^{-9}$ is mutation rate and  $L=50K$ is the window size in base pairs 
	which gives $\theta=2\mu N_eL=200$ and $\rho=2N_erL=2000$. For default 
	parameter, the expected number of segregating sites in a window is 
	\beqq
	\Ebb[M]=\theta \sum_{i=1}^{F-1}\frac{1}{i}=1175
	\eeqq
	\item{\bf Creating initial diploid population} 
	To implement similar setting for experimental evolution of diploid 
	organisms, 
	initial  haplotypes first cloned to create $F$ diploid homozygotes. Then 
	each 
	diploid individual is  cloned $N/F$ times to yield diploid population of 
	size 
	$N$.
	\item{\bf Forward Simulation}
	Given initial diploid population, position of the site under selection, 
	selection 
	strength $s$, number of replicates $R=3$, recombination rate 
	$r=2\times10^{-8}$ 
	and sampling times $\Tc=\{10,20,30,40,50\}$, \texttt{simuPop} is used to 
	perform
	forward simulation and  compute allele frequencies for all of the $R$ 
	replicates.
\end{enumerate}

\subsection{Dynamics of the Sweep}
In this part we study the dynamics of selective sweep based on change in allele
 frequency of single site, and allele frequency spectrum of a 50Kb window. 
 Then we examine linkage disequilibrium more carefully in the context of dynamic data.
\paragraph{Allele Frequencies}
\begin{figure}[t]
	\centering \label{fig:sweep}
	\begin{tabular}{lr}
		\includegraphics[trim={2in 0.1in 1.5in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidSoft}
		&\includegraphics[trim={2in 0.1in 1.9in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidHard}
	\end{tabular}
	\caption{soft (left) and hard (right) sweep.}
\end{figure}


\paragraph{Allele Frequency Spectrum}
\begin{figure}
	\centering \label{fig:dynamic-strong}
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for strong selection.}
\end{figure}

\begin{figure}
	\centering \label{fig:dynamic-weak}
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=4,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for weak selection.}
\end{figure}
\paragraph{Linkage Disequilibrium}
\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{spaceTimeLD}
\end{figure}

\paragraph{Challenges}
\begin{itemize}
	\item time scale
	\item population size
\end{itemize}
\begin{figure}
	\centering \label{fig:bottleneck}
	\includegraphics[scale=0.2]{bottleneck}
\end{figure}



\subsection{Detecting Selection}
\paragraph{Hard Sweep vs Soft Sweep}
\begin{figure}
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0 , clip,width=\textwidth]{power}
\end{figure}

\paragraph{Effects of carrier frequency}
\paragraph{Effects of sampling times}
\paragraph{Effects of number of replicates}
\paragraph{Single-Locus vs Multi-locus}
\paragraph{Time-series vs single-snapshot}

\subsection{Locating Selection}

\subsection{Strength of  Selection}


\section{Materials and Methods}

\subsection{Single-Locus Methods}
Consider a bi-allelic single-locus (Wright-Fisher like) model with no
mutations \cite{Ewens2012Mathematical}, discrete generations, random
mating etc. With finite population size, the allele frequencies from
generation to generation are described by the random process
$\{x_{\ell,t}\}$ which denotes the population allele frequency at
locus $\ell$ at time generation $t$. The model has access to finite
number of observations in time for each locus, where the sampling
times are given by the set $\Tc=\{\tau_1,\tau_2,\cdots, \tau_T \}$
such that $\tau_1<\tau_2,\cdots<\tau_T$. We also assume that
measurements are taken up for $R$ experimental replicates for $L$
loci. Thus, The allele frequencies of a population are given by the
tensor $\bfX \in [0,1]^{R \times L \times T}$, where $X_{r,\ell,t}$
stores the observed of allele frequency at replicate $r$, locus $\ell$
and generation $\tau_t$. However, we often omit $r$ for clarity of
exposition. A neutrally evolving allele `drifts', and for a finite
population,
\beq x_{t+1} = x_t + \epsilon_t\; .\footnote{With infinite population size
	we have $x_{t+1} = x_t$, a restatement of Hardy-Weinberg
	equilibrium.}
\label{eq:drift}
\eeq where $\epsilon_t \sim \Nc(0,1)$ is a change due to genetic drift
in generation $t$.

\paragraph{Naive two-point optimization.} We start with a simplified
scheme to serve as a baseline for results.  For each $t\in \Tc$, we
use \eqref{eq:inf-pop} to provide a naive estimate $s(t)$ for the
selection coefficient as
\begin{equation}
	s^*(t)=\frac{2}{t} \log \left( \frac{x_t(1-x_0)}{x_0 (1-x_t)} \right) = \frac{2}{t}  
	\left( \eta(x_t)-\eta(x_0)\right)
	\label{eq:naive2point}
\end{equation}
To reduce the variance we can average all the naive estimates: 
\begin{equation}
	s^*=\frac{1}{|\Tc|}\sum_{t\in \Tc}\frac{2}{t}  \left( \eta(x_t)-\eta(x_0)\right)
	\label{eq:naive}
\end{equation}


\paragraph{Maximum Likelihood Estimate.}
For locus $\ell$  and replicate $r$, the observed time-series of 
allele frequencies are given by:
\[
{\bfx_{r,\ell}} = [x_{r,\ell,\tau_1},\ldots,x_{r,\ell,\tau_T}]
\]
When the locus is under selection with coefficient $s$, let the allele
frequencies at times $t\in \Tc$ be described by
\[
\vecbold{\nu}(s) = [\nu_{\tau_1}(s),\ldots,\nu_{\tau_T}(s)]
\]
We assume that the noise due to genetic drift at each generation is
Gaussian, and independent of other generations. In addition, $x_0$ is
the same for all the replicates, and $c$. Therefore, using the
independence of replicates, $\bfx_{r,\ell,t} \sim \Nc(\bf\nu_t(s),I)$
for all replicates $r$. The likelihood of the observed allele
frequencies is given by the Gaussian
\begin{equation}
	\Lc_G(s|\bfx_{1,\ell}, \dots,\bfx_{R,\ell}) = \prod_{r=1}^R \Pr(\bfx_{r,\ell}| 
	\vecbold{\mu}=\vecbold{\nu}(s),
	\Sigma= I) 
\end{equation}
Taking logarithms and removing constant values, the problem of finding
MLE for $s$ amounts to minimizing the negative-log-likelihood function
with respect to $s$: 
\begin{equation}
	s^*=\underset{s}{\arg \min} \frac{1}{2} \sum_{r=1}^R \parallel 
	{\vecbold{\nu}(s) -
		\bfx_{r,\ell}} \parallel_2^2
	\label{eq:nlls0}
\end{equation}
which is an instance of non-linear least squares optimization.
\subsection{Linked-locus Methods}

\subsection{Likelihood Ratio Test}
Using the MLE (and other) estimates of $s$ it is desirable to perform a 
secondary task such as \emph{testing for selection} or \emph{locating the 
site under selection}. Likelihood ratio test (LRT) statistics for time series 
\cite{feder2014Identifying} have shown to be predictors for differentiating
 neutral and  natural selection evolution cases. For a single locus model, the
  likelihood ratio 
test statistics $\Lambda(s*)$ is defined
\beq \label{eq:lrt}
\Lambda(s^*) = \log \left(\frac{\Lc(\bfX|s=s*)}{\Lc(\bfX|s=0)}\right)
\eeq
where $s^*$ is the optimal solution for the maximum-likelihood procedure. 

\newpage
\section{Appendix}
\subsection{Allele frequencies under selective sweep}
Consider next, a locus under selection with coefficient $s (0\le s\le
1)$, overdominance and parameter $h (0\le h\le 1)$, so that the
probability of an individual inheriting $2$ favored alleles is
$\propto (1+s)$, relative to inheriting $0$ favored
alleles. Respectively, the probability of favoring $1$ favored allele
$\propto (1+hs)$ Then,


\begin{equation}
x_{t+1} = f(x_t;s,h) + \epsilon_t
\label{eq:trans0} 
\end{equation}
where $f: [0,1] \mapsto [0,1]$ is the
transition function given by:
\begin{equation}
f(x;s,h)=\frac{(1+s)x^2 + (1+hs)x(1-x)}{(1+s)x^2 + 2(1+hs)x(1-x) + (1-x)^2}
=x+\frac{s(h+(1-2h)x)x(1-x)}{1+sx(2h+(1-2h)x))}\;.
\end{equation}
We simplify notation by setting $h=0.5$, so that
\begin{equation}
f(x;s,0.5)=x+\frac{sx(1-x)}{2+2sx}\;.
\label{eq:hequalshalf}
\end{equation}


Our goal is to estimate $s$ given allele frequencies, and/or to decide
if $s>0$. Note that the dependence on $s$ is non-linear. Define an
auxiliary continuous (in time) function $\nu_t(s)$ as follows:
$\nu_0(s)=x_0$ for all $s$, and:
\beqq
\nu_{t+\delta t}(s) &= \nu_t(s)+\frac{\nu_t(s)(1-\nu_t(s))s\delta 
	t}{2+2s\delta t \nu_t(s)}\;,\\
\frac{d\nu_t(s)}{dt} &=\lim_{\delta t\rightarrow 0}\frac{\nu_{t+\delta t}(s) 
	-\nu_t(s)}{\delta t}\\
&=\lim_{\delta t\rightarrow 
	0}\frac{s\nu_t(s)(1-\nu_t(s))}{2+2\nu_t(s)s\delta t}\\
&= \frac{s}{2}\nu_t(s)(1-\nu_t(s)) \;.
\label{eq:ode}
\eeqq
The ODE can be readily solved for $\nu_t(s)$ as
\begin{equation}
\nu_t(s) =\frac{1}{1+\frac{1-x_0}{x_0}e^{-st/2}} = \sigma(st/2+\eta(x_0)) 
\label{eq:inf-pop}
\end{equation}
where$\sigma(.)$ is the logistic
function and $\eta(.)$ is logit function (inverse of the logistic function). Note 
that $x_t\sim\Nc(\nu_t,1)$ for all discrete times $t$,
and therefore observations $x_t$ can be used to find Maximum
Likelihood Estimate of $\nu_t$. Finally, as the sigmoid is a
one-to-one function, we can find MLE for $s$ given {\bf $\nu$}.
\subsubsection{Fay Wu's H}
Consider a collection of $m$ linked loci. Omitting replicate
information, we represent the derived allele frequencies at time $t\in
\Tc$ using:
\[
{\bf x_t} = [x_{1,t},x_{2,t}\ldots,x_{m,t}]^T 
\]
At any time $t$, suppose we have a sample of $n$ individual haplotypes
that were described by the $n\times m$ matrix $M_t$. Each row of $M_t$
represents a haplotype from the sample, and is denoted by vector
$\mathbf{h} \in \{0,1\}^m$. Let $\bfone$ denote the
$n$-dimensional unit vector. In Pooled sequencing, $M_t$ is hidden,
but by definition,
\[
\mathbf{x_t} = \frac{1}{n} M_t^T\bfone
\]
We recently devised the $1\dHAF$ statistic~\cite{ronen2015predicting} for a
haplotype, and proved some bounds on its expected value during neutral
evolution, and under selection constraints. In the current notation,
the $1\dHAF$ score of haplotype $h$ is given by
\begin{equation}
1\dHAF(\mathbf{h})=n\bfx_t \cdot \mathbf{h}
\;.
\label{eq:1-HAF_SNPmatrix}
\end{equation}
The average $1\dHAF$ score at time $t$ can be estimated as:
\begin{equation} 
\mathbb{E}[1\dHAF(t)]=\frac{1}{n}\sum_h 1\dHAF(\mathbf{h}) = n\parallel 
\mathbf{x_t}\parallel^2
\end{equation} 
However, in an ongoing selective sweep, the expected $1\dHAF$ score is
a function of the selection coefficient $s$, and the frequency $\nu$
of the carriers of the favored mutation. From~\cite{ronen2015predicting}, we
find that $ \mathbb{E}[1\dHAF(t)]\approx nz(\nu_t)$ where

\beq
z(\nu_t)= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)
\label{eq:hafscorepooled}
\eeq

Rearranging terms, we can estimate $\nu_t$ using
\begin{equation}
\hspace{-0.1in}\nu_t^*=\arg\min_{\nu_t}   \left( z(\nu_t) -\parallel 
\mathbf{x}_t\parallel^2  \right)^2
\label{eq:pooledfrequency}
\end{equation}
where $\nu_t=\sigma(st/2+\eta(\nu_0))$ and for each replicate we have
\beq
\xbb &=[\|\bfx_{\tau_1}\|^2 , \ldots, \|\bfx_{\tau_T}\|^2]^T\\
\bfz(s)&=[z(\nu_{\tau_1}(s)) , \ldots, z(\nu_{\tau_T}(s))]^T
\eeq

\beq \label{eq:nlls1}
s^*=\underset{s}{\arg \min} \frac{1}{2} \sum_{r=1}^R \parallel {\bfz_r(s) -  
	\xbb_{r}} \parallel_2^2
\eeq

\beq
AverageHAF&=n\|x\|^2= \alpha\theta_H\\
\|x\|^2&=\sum_i \left(\frac{i}{n}\right)^2\xi_i =\frac{1}{n^2}\sum_i i^2\xi_i = 
\frac{ (n-1)}{2n}\theta_H \\
\alpha&=\frac{ n-1}{2}
\eeq
where $k$ is the number of histogram bins in computing AFS.

\subsection{Tajima's D}
We can compute Tajima's D in time as a function of $s$ and initial carrier 
frequency.

\beq
D_0&=\Pi_0 - W_0, \ \ \ \ \ D_t=\Pi_t - W_t\\
\Pi_t&= (1-\nu_t^2)\Pi_0 \\
W_0&= \frac{m_0}{S_n}, \ \ \ \ \ W_t= \frac{m_t}{S_n}\\
\frac{W_t}{W_0}&=\frac{\frac{m_t}{S}}{\frac{m_0}{S}} \ \ \Rightarrow 
W_t=\frac{m_t}{m_0}W0 \\
\frac{m_t}{m_0}&=\frac{\log\left((1-\nu_t)n +1 \right)}{\log(n)} \approx  
\frac{\log\left((1-\nu_t)n\right)}{\log(n)} = \frac{\log(1-\nu_t)+\log(n)}{\log(n)} = 
1+ \frac{ \log(1-\nu_t)}{\log(n)}\\
D_t&= (1-\nu_t^2)\Pi_0 - (1+ \frac{ \log(1-\nu_t)}{\log(n)} ) W_0 = 
D_0+\log(1-\nu_t) \frac{W_0}{\log(n)} -\nu_t^2 \Pi_0\\
\frac{d D_{t}}{d \nu_{t}}&=
2\Pi_0\nu_t - \frac{\frac{W_0}{\log(n)}}{1-\nu_t}=a\nu_t + \frac{b}{1-\nu_t}
\eeq

%\input{intro}
%\input{method}
%\input{experiment}
%\input{conclusion}
\cite{achaz2009frequency,akey2009constructing,andersson2011notch,barrett2008natural,Barrick2013Genome,barrick2009genome,barton1991natural,bergland2014genomic,bollback2007clonal,bollback2008estimation,boyko2008assessing,braverman1995hitchhiking,daniels2013genetic,denef2012situ,desai2008polymorphism,Elena2003Evolution,enard2015viruses,evans2007non,Ewens2012Mathematical,fay2000hitchhiking,feder2014Identifying,feder2012ldx,Feder2016More,hegreness2006equivalence,Hudson2002Generating,illingworth2011distinguishing,illingworth2012quantifying,izutsu2015dynamics,jha2015whole,kawecki2012experimental,kimura1955solution,kreitman2000methods,lang2011genetic,lang2013pervasive,lenski1991long,malaspinas2012estimating,maldarelli2013hiv,mathieson2013estimating,messer2013population,nielsen2005genomic,orozco2012adaptation,Peng2005simuPOP,pollak1983new,przeworski2002signature,ptak2002evidence,pybus2015hierarchical,ramos2002statistical,reid2011new,Ronen2015Haplotype,ronen2015predicting,ronen2013learning,rose1994evolutionary,sabeti2006positive,sawyer1992population,schlotterer2015combining,schlotterer2014sequencing,schrider2015soft,Sheehan028175,slatkin2008linkage,song2012simple,Steinrücken2014a,Stephan2006The,tajima1989statistical,Terhorst2015Multi,tobler2014massive,wang2001pseudo,waples1989generalized,williamson1999using,winters2012development,zhou2011experimental}
\bibliographystyle{plain}
\bibliography{library}
\end{document}
