\documentclass[11pt]{article}
\input{macros_vineet}
\input{macros_arya}
\title{Detecting Selection in Experimental Evolution Experiments}
\author[1]{Arya Iranmehr}
\author[1]{Ali Akbari}
\author[2]{Vineet Bafna}
\affil[1]{\footnotesize Electrical and Computer Engineering, University of California, San Diego, La Jolla, CA 92093, USA.}
\affil[2]{\footnotesize Computer Science \& Engineering, University of California, San Diego, La Jolla, CA 92093, USA}
\date{}

\begin{document}
\maketitle
\begin{abstract}
Experimental evolution (EE) studies are powerful tools for observing
molecular evolution "in-action" in wild and controlled environments. This 
paradigm of experiments was infeasible until recently when the whole-genome 
and whole-population was made possible by next-generation sequencing. 
However, one of the primary constraints of the EE studies 
is the 
limited time for the experiment, which depend on the organism's generation 
time. 
For example, in the EE aims to study genetic adaptation, the population 
can only evolved and re-sequenced in a small number of generations, which is 
usually smaller than the time required for fixation of adaptive 
allele. Although, a 
complete library of tests-of-selection has already been developed, they are 
mainly designed 
to identify adaptation in pre/post fixation data.
In this research, we study the problem of identifying selection signatures in 
allele-frequency trajectories of small number of generations. 
We propose new Composite Likelihood Ratio 
(CLR) statistic which computes the likelihoods of null (neutral) and 
alternative (selection) model using the Wright-Fisher Markov chain model 
outperforms existing methods on extensive soft and hard sweep simulations. 
Finally, we apply the proposed statistic to the controlled experimental 
evolution of D. melanogaster to detect adaptive genes/alleles under alternating 
cold and hot temperatures.
\end{abstract}
\section{Introduction}
Genetic adaptation is \emph{the} central evolutionary process and is at the
 core of some of the greatest challenges facing humanity. 
For example, HIV's rapid evolution of drug resistance during treatment, makes 
treatments ineffective and even by prescribing a combination of drugs the speed 
of adaptation only slowed down~\cite{Feder2016More}. 
Cancer would be much more straightforward to treat if not for
tumor's ability to adapt to anti-cancer drugs 
\cite{gottesman2002mechanisms,zahreddine2013mechanisms}. 
Malaria could be treated with cheap drugs such as quinine instead of 
being one of the world's worst killers 
\cite{ariey2014molecular,nair2007recurrent}. 
Crop pests would be manageable with small doses of safe insecticides 
instead of requiring applications of ever increasing amounts of a diverse array 
of powerful chemicals\cite{daborn2001ddt}. 
Finally, antibiotic resistance \cite{spellberg2008epidemic} to new antibiotics 
could return us to the pre-Antibiotic age.
Despite enormous examples of rapid adaptation, Messer and Petrov 
\cite{messer2013population} argued that "many, if not most, cases of adaptation 
are yet to be discovered".


Model for describing genetic adaptations, \emph{selective sweep} 
\cite{smith1974hitch,kaplan1989hitchhiking} , take into account of associations 
of the beneficial mutation with its surrounding loci.
The extent of genetic loci that are in association with the adaptive allele 
depends on the amount of accumulated recombination events between adaptive 
allele and the rest of genome. In the asexual populations, where no 
recombination event occurs, the whole genome is linked to the adaptive allele 
and "clonal interference" \cite{desai2007beneficial,lang2013pervasive} best 
describes the dynamic 
genetic adaptation process.
On the other hand, in sexual population, the favored mutation is only in 
linkage-disequilibrium (LD) with its nearby polymorphisms. Hence, methods for 
identifying selective 
sweep in sexual populations often analyze  polymorphism data of a population of 
in a genomic region, rather than a single site. 

Adaptation leaves a variety of detectable signatures 
genomes, including, reduction in genetic 
variation\cite{tajima1989statistical,fay2000hitchhiking,ronen2013learning} in 
allele-frequency data, 
prevalence of long haplotypes 
	\cite{sabeti2006positive,vitti2013detecting} in haplotype (phased) data, 
	population differentiation \cite{holsinger2009genetics} in cross-population 
	data and 
	rapid increase in allele 
	frequencies \cite{bergland2014genomic} in the dynamic data.
However, in the experimental evolution with pooled-sequencing, only frequency 
data available. Given static frequency data, tests including
Tajima's \emph{D} ~\cite{tajima1989statistical}, 
Fay and Wu's \emph{H}~\cite{fay2000hitchhiking}, 
SFSelect~\cite{ronen2013learning} compute their test statistic using the y 
distribution of polymorphisms, site-frequency spectrum (SFS), in a genomic 
region. Despite their simplicity and intuitiveness, it has been shown that
  SFS-based tests often fail are prone to pathological false-positive/negatives
  due to ascertainment biases and  demography
  ~\cite{ptak2002evidence, ramos2002statistical,akey2009constructing, 
  nielsen2003correcting}. 

However, identifying selection in dynamic data is not well studied yet.
Feder et. al \cite{feder2014Identifying}  proposed Frequency Increment
 Test(FIT) for dynamic frequency data and Likelihood Ratio Test (LRT) for 
 frequency/haplotype dynamic data. Bergland et. al \cite{bergland2014genomic} 
 applied $F_{st}$ to multiple populations throughout time to identify selection.
 
\paragraph{Experimental Evolution}
Recent advances of whole genome sequencing has enabled us to sequence 
populations at a reasonable cost, such that one could design experiments
to study different forces of evolution in real time. \emph{Experimental 
evolution} is the study of the evolutionary processes of a model organism in a 
controlled  
\cite{hegreness2006equivalence,lang2013pervasive,orozco2012adaptation,
	lang2011genetic,barrick2009genome,bollback2007clonal,oz2014strength} 
or natural 
	\cite{maldarelli2013hiv,reid2011new,denef2012situ,winters2012development,
	daniels2013genetic,barrett2008natural,bergland2014genomic} environment.

Among different types of evolution experiments \cite{Barrick2013Genome} in this 
paper we focus on Adaptive evolution, fixed population size, multicellular 
sexual organisms. While constraints such as small population sizes, limited 
timescales and oversimplified  laboratory
 environments limits interpreting experimental evolution results, it has
  been used to test different hypothesis~\cite{kawecki2012experimental}
   regarding
mutation and adaptation, 
genetic drift and inbreeding, 
environmental variability,
sexual selection and conflict, 
kin selection and cooperation
life history and sex allocation, 
sexual reproduction and mating systems, 
behavior and cognition, 
host–parasite interactions, 
speciation
repeatability of evolution.

In addition, time-series \emph{models} is shown to be more powerful 
\cite{boyko2008assessing,desai2008polymorphism,sawyer1992population} 
than single-snapshot data.
In particular, dynamic data has been used to estimate model parameters 
including population size
\cite{williamson1999using,wang2001pseudo,pollak1983new,waples1989generalized,
	Terhorst2015Multi}
strength of selection
\cite{mathieson2013estimating,illingworth2011distinguishing,Terhorst2015Multi,
	bollback2008estimation,illingworth2012quantifying,malaspinas2012estimating,
	Steinrücken2014a, malaspinas2012estimating}, allele age
 \cite{malaspinas2012estimating}
recombination rate~\cite{Terhorst2015Multi}  and mutation
 rate~\cite{Barrick2013Genome, Terhorst2015Multi}. 

Linked-loci time-series (spatio-temporal) methods 
\cite{illingworth2011distinguishing,illingworth2012quantifying,
	Barrick2013Genome,Terhorst2015Multi} incorporate linkage into computation
 which makes it cumbersome. Theoretical diffusion approximation method has also 
 developed for neutral allele frequencies~\cite{Ewens2012Mathematical, 
 kimura1955solution}
neutral SFS~\cite{evans2007non}, neutral allele
 frequencies~\cite{song2012simple}.

Experimental evolution (including evolve and re-sequence paradigms)
have become increasingly popular as a complementary tool to understand
the forces of selection, allowing for controlled environments,
specific selection constraints. Examples of experimental evolution
studies abound in sexual populations, particularly in fruitfly. Burke
et al.~\cite{Barrick2013Genome} evolved flies for over $600$ generations 
under
selection for accelerated development, and noted that evolution for
sexual populations is very different from those of asexual populations
(e.g., Lenski): the effects of clonal interference is slower
due to recombination that allows for the incorporation of multiple
beneficial alleles, but there are fewer, unconditionally advantageous
alleles that arise at the onset of selection. Rose and
Colleagues~\cite{rose1994evolutionary} created $200$ experimentally evolved
populations selected for different traits, starting with $10$ initial
populations. Zhou et al. evolved flies to adapt to low oxygen
environment (hypoxia) for over $300$ generations, and identified many
genes involved in hypoxia tolerance~\cite{zhou2011experimental}.  Instead, 
they observed
incomplete fixation (`soft-sweeps') due in part to standing variation,
changing selection coefficients, and small fitness effects. Similarly, Jha et
 al. \cite{jha2015whole} identified genes under selection for egg size in flies.
Barrick et al. \cite{barrick2009genome} evolved a population of 
\emph{Escherichia coli} for 40,000 generations.

Much like in natural populations, many of these studies also
sequenced/genotyped only the latest population. However, the emergence
of NGS and related technologies has made it feasible to sequence the
evolving population at multiple time points in its evolution. At the
same time, for small organisms where a single animal does not provide
enough DNA, it is more effective to pool together individuals in the
population at a particular time point, in a single sequencing
experiment. Thus, instead of individual genotypes, we obtain
frequencies of the derived allele at all loci at different time
points. Methods for analyzing such time series pooled sequence data
are only just being developed.


\section{Background}
\paragraph{Notation} Let $\bfX \triangleq (X_{ijk}) \in [0, 1]^{T \times 
M\times R}$ denote the population frequency where $T$ is the number of samples 
in time, $M$ is the number of segregating sites, and $R$ is the number of 
replicates. Samples in time are taken such that $\tau_1<\tau_2,\cdots<\tau_T$.
To simplify our notation, we define $x_t$ as a allele frequency of a site in a
 replicate, and we denote it by $\nu_t$ if such a site is directly under 
 selection with the strength $s$. I.E., $\nu_t$ can be regarded as frequency of 
 the carrier in the  population.
\paragraph{Simulations}
For each experiment a diploid population is created and evolved as follows. 
\begin{enumerate}[I.]
	\item {\bf Creating initial founder line haplotypes}
	First using msms program, we created neutral populations for $F$ founding 
	haplotypes with \emph{default} parameters \texttt{\$./msms <F> 1 -t 
	<2$\mu$LNe> 
		-r <2rNeL> 
		<L>} 
	where $F=200$ is number of founder lines, $N_e=10^6$ is 
	effective 
	population size, $r=2*10^{-8}$ is recombination rate and $\mu=2\times 
	10^{-9}$ is mutation rate and  $L=50K$ is the window size in base pairs 
	which gives $\theta=2\mu N_eL=200$ and $\rho=2N_erL=2000$. For default 
	parameter, the expected number of segregating sites in a window is 
	\beqq
	\Ebb[M]=\theta \sum_{i=1}^{F-1}\frac{1}{i}=1175
	\eeqq
	\item{\bf Creating initial diploid population} 
	To implement similar setting for experimental evolution of diploid 
	organisms, 
	initial  haplotypes first cloned to create $F$ diploid homozygotes. Then 
	each 
	diploid individual is  cloned $N/F$ times to yield diploid population of 
	size 
	$N$.
	\item{\bf Forward Simulation}
	Given initial diploid population, position of the site under selection, 
	selection 
	strength $s$, number of replicates $R=3$, recombination rate 
	$r=2\times10^{-8}$ 
	and sampling times $\Tc=\{10,20,30,40,50\}$, \texttt{simuPop} is used to 
	perform
	forward simulation and  compute allele frequencies for all of the $R$ 
	replicates. Also, to avoid spurious simulation samples, simulation results 
	are constrained to those that the beneficial allele escapes stochastic loss 
	of genetic drift and \emph{establishes} in all the replicates. 
\end{enumerate}

\paragraph{Hard and Soft Sweep}
In this paper, we only study (positive) selection with single beneficial allele 
under selective sweep, i.e. favored allele is in linkage disequilibrium (LD) 
with its nearby loci. Selective sweeps are classified by the amount of 
variation exist in the individuals carrying adaptive allele. By definition, 
hard sweep is the case when all the carriers coalesce after onset of selection 
and thus diversity between carriers is at its minimum. 
In general, soft sweep is difficult to detect. We conduct experiments, so that 
evaluate prediction performance of the proposed method under hard and soft 
sweeps with different selection strengths. Since here we consider random mating 
with no de novo mutations, soft sweep can only happen in standing variation 
where the site under selection is at frequency is larger than $1/F$. In 
contrast, hard sweep experiments are those that their adaptive allele is at its 
minim frequency, $1/F$, at the onset of selection.


\paragraph{Challenges}
The main constraint in the experimental evolution experiments, is the 
sampling-time-span (STS) $\tau=\tau_T-\tau_0$, the number of generations 
between the first and last generation in the time-series data, which primarily 
determined organism generation time. In other words, for wide range of values 
of $s$ (not very strong selection), STS will be significantly smaller than 
fixation time. As a result, methods for detecting selection on static data, 
which work the best when the beneficial allele is close to fixation, would not 
work properly for dynamic data.

Moreover, in controlled experimental evolution, a population is evolve and 
inbreed under a selective pressure. This scenario which population size is 
effectively reduces from large number of wild type (e.g.D. melanogaster 
$N_e\approx10^6$ ) to a small number (typically $F$ is between 100-1000) of 
founder lines for EE, resembles a severe population bottleneck. This phenomena 
makes genetic drift quite strong and confounds polymorphism summary statistics 
and makes difficult to detecting selection weaker selections.

\newpage
\section{Results}
Selective sweep is a complex random process which under infinite site assumption
only depends on selection strength,
population size and the underlying genealogy at the onset of 
selection. 
As a result, the change in different population statistics, including 
\emph{site 
allele frequencies}(AF), \emph{site frequency spectrum}(SFS) and \emph{linkage 
disequilibrium}(LD), can be attributed to the selection event.
However, other independent processes such as genetic drift, 
population expansion/bottleneck and recombination cause AF,SFS and LD to 
change, respectively. Therefore, obtaining a clear understanding of dynamics of 
selective sweep  requires disentangling effects of natural selection from 
those of neutral evolution and demography.

 In the following, we examine the dynamics of AF and SFS in the experimental 
 evolution and evaluate their power in identification of selection 
 events. 
 The "identification" of a selection event can be done in 
 different levels of detail. At coarsest level, identification can be done by 
 determining whether a region 
 (e.g. a window of 50Kbp, or a gene) on genome is under selection. In the rest, 
 we consider this task to 
 be the task of \emph{detection}.
 Then, finding the causal 
 mutation/allele would be a more elaborate identification of selection, 
 henceforth, \emph{locating} selection. 
 Finally, estimating model parameters such as strength of selection and 
 overdominance at the site fully describes the selective sweep.
 



\paragraph{Allele Frequencies}
Selection and genetic drift are the two main 
\emph{independent} evolutionary 
forces that cause change in allele frequencies, each 
generation. 
In the simplest case, models for neutral evolution of a 
single-locus can be 
used to derive a null distribution in order to detect the 
region under 
selection or locate the 
adaptive allele.
For example, Feder et al. 
\cite{feder2014Identifying} proposed a (continues-time 
continues-state) 
Brownian motion process for modeling variation of allele 
frequencies under 
genetic drift. Basically, given current allele frequency 
$\nu_t$ at a site 
Brownian motion approximation assumes future-generations allele 
frequencies are drawn from the Gaussian distribution
\beq
\nu_{t+\tau}\sim 
\Nc\left(\nu_t,\frac{2\nu_t(1-\nu_t)}{N_e}\tau\right) 
\label{eq:brown}
\eeq
Using \eqref{eq:brown}, Frequency Increment Test (FIT) 
\cite{feder2014Identifying} computes p-value of 
time-series data by a Student's t-test. However, Brownian 
motion is a poor 
approximation of the actual genetic drift process when 1) 
$\nu_t$ is not close 
to $0.5$, 2) when $\tau$ is large. A the start of selection 
site frequencies 
are distributed according to SFS of a neutral population, and 
thus most of the 
sites are at low frequencies Figure \ref{fig:afs}. Thus if the selecting site 
is 
chosen randomly 
with high probability its frequency will be far 0.5. 
Moreover, in many 
experimental evolution experiments STS $\tau$ is chosen be 
large, e.g. of the order of 10 to 
100 generations.


Instead, probabilities of trajectories can be computed using 
a 
(discrete-time discrete-state-space) Markov Chain with 
transition matrix $P$
\beq
P_{i,j}=\pr\left(\nu_{t+1}=\frac{j}{2N} \left| 
\nu_{t}=\frac{i}{2N}\right)={2N 
	\choose j} \right. 
\nu_{t}^j (1-\nu_{t})^{2N-j}  
\eeq
which transition probabilities for $\tau$ generations ahead 
can be computed by 
powering the transition matrix to $\tau$.
Figure \ref{fig:markov} illustrates expected and observed distribution of 
frequencies after 
1, 10 and 100 generations when starting frequency is 0.005 (top) and 0.1 
(bottom) for Brownian motion and Markov chain. Brownian motion 
exhibits poor prediction of distributions of allele frequencies when starting 
frequency is far from 
0.5. On the other hand, Markov chain provides an accurate distribution of 
allele frequencies 
for future generations, under genetic drift.

On the other hand, alternative models for single-locus selection can be 
utilized to estimate strength of selection and robustly 
detect and locate selection by 
evaluating two hypotheses in order to perform likelihood ratio tests.
In a pure selection process with no drift, i.e. infinite 
population size, 
dynamic of allele frequencies 
can be well approximated by the logistic function (see 
Appendix \ref{app:af} 
for 
derivation)
\beq
\nu_t=\sigma(st+\eta(\nu_0))\label{eq:nut}
\eeq
where $\sigma(x)=1/(1+e^{-x})$ is the logistic function, and 
$\eta(x)=\log(x)/\log(1-x)$ is inverse of the 
logistic, aka logit, function. Figure \ref{fig:sweep} depicts 
the behavior of 
the logistic model for site allele frequencies and the 
default sampling time 
span(STS). Without genetic drift, soft sweeps are easier to 
detect, 
because the logistic function happens to have steeper slope 
in th STS than 
those of hard sweeps. In addition, even in infinite 
population size, it is 
difficult differentiate between weak selections and genetic 
drift (a horizontal 
line), in 50 generations.


As shown in Figures \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (first 
row), the approximate logistic model is consistent with 
simulated data and we 
use it to estimate the strength of selection for 
each site by a simple regression (see \ref{sec:regression} 
for details).



\paragraph{Site Frequency Spectrum} SFS (Figure 
\ref{fig:afs}) is the 
distribution of allele 
frequencies in a genomic region.
Historically, SFS of static data has been extensively used to 
"detect" genetic 
adaption and demographic changes in a population. Basically, 
SFS is measure of diversity in a genomic region computed from allele 
frequencies. 
In general, reduction in diversity is a signal of 
selection, and detecting selection based on the reduction in 
genomic diversity is a subtle task, because 
\begin{enumerate}[(i)]
	\item  in soft sweeps the genomic diversity is not is not 
	necessarily 
	reduced.
	\item even in hard sweeps with no recombination, the 
	reduction diversity is 
	significant only when the SFS sample is taken close to 
	fixation (not too 
	far prior or after fixation)
	\item SFS change in the same way for both selection an 
	demography changes. 
\end{enumerate}

Although conditions (hard sweep with no recombination, not 
far from 
fixation, and random-mating and constant size population ) 
for detecting 
selection based on SFS are very restrictive, SFS-based tests 
are simple and 
inexpensive to use and often used in combination with 
other tests~\cite{}. 

In practice, SFS distribution is \emph{sclarized} by a 
weighted 
linear combination\cite{achaz2009frequency}. More precisely, 
the weights should be positive and sum to 
one, which called a convex combination, aka discrete 
expectation. For 
example, 
test statistics for Tajima's $D$\cite{tajima1989statistical}, 
Fay Wu's $H$~\cite{fay2000hitchhiking} and 
SFSelect\cite{ronen2013learning} can be obtained by a dot 
product of SFS vector with their weight vector. 

Whether the dynamic of SFS in times-series can improve the detection power in 
the conditions (i-iii) is not well-studied yet. Evans et al. 
\cite{evans2007non} developed diffusion equations for 
evolution of SFS in time series, they are difficult to solve. 
Hence, we analyze and visualize scalar test statistics $D$, $H$ and SFSelect 
in the dynamic data under different selection regimes.

To adapt SFS-based test for dynamic data, one can simply design statistical 
tests to examine whether the 
observations of $D$, $H$ and SFSelect statistics are 
significantly deviated 
from the null distribution, i.e. constant values in time.
On the other hand, we can model the dynamics of SFS as a 
function of selection 
strength to 1)have a model for estimating model parameter 2)obtain an insight 
regarding how does SFS statistics behave under different 
stages of selection.

As mentioned before, restricting population to $F$ founder lines ($F<<N_e$) 
resembles a severe bottleneck event, which confound SFS. Figure 
\ref{fig:bottleneck} demonstrates effect of controlled experimental evolution 
on different SFS statistics under neutral evolution for 1000 simulations. The 
mean of 
neutral simulations can be used to empirically filter out the effect of 
bottleneck in dynamic data. 

\begin{enumerate}[I.]
	\item {\bf Tajima's D}. As shown in the Appendix \ref{app:td}, dynamic of 
	Tajima's D in hard sweep is
	\beq
	D_t=D_0-\log(1-\nu_t) \frac{W_0}{\log(2N)} -\nu_t^2 \Pi_0\label{eq:tdt}
	\eeq

	where $W_0$ and $\Pi_0$ are Watterson and Tajima estimate 
	of $\theta$ at 
	the first generation.
	Given that $\nu_t$ defined in term of $\nu_0$ and $s$ in 
	\eqref{eq:nut}, we 
	can use \eqref{eq:tdt} to estimate the strength of 
	selection by performing a nonlinear 
	regression to find 
	$s$. Then, a likelihood ratio, similar to \eqref{eq:lrt}, 
	provides\footnote{The 
	likelihood of the data to the model is the least-squares 
	loss between model 
	$D_t(\hat{(s)})$ and the observed $D$.} a predictor for 
	detecting selection in each window. Figures \ref{fig:dynamic-weak} and 
	\ref{fig:dynamic-strong} (last 
	row, left), shows that the parametric model with the actual value of $s$ 
	provides is consistent with 1000 simulations, in strong and weak 
	selections. Also, \eqref{eq:tdt} represents dynamic of $D$ in hard 
	sweep, and the 
	model becomes invalid as initial carrier frequency increases, Figure 
	\ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (last 
	row, right).
	
	Finally, even in a infinite population size setting, 
	differentiating the between selection and drift is difficult in early 
	generations. W.l.o.g when $D_0=0$ and $\Pi_0=W_0=1$, 
	$D_t$ is sum of the 
	logarithmic $\frac{-\log(1-\nu_t)}{\log(2N)}$ and the squared term 
	$\nu_t^2$, blue and green curves in 
	Figure 
	\ref{fig:tdterms}. As shown in the Figure 
	\ref{fig:tdterms} right, in early 
	generations of hard sweep where carrier frequency is low, 
	$D_t$ is either 
	positive or close zero. In other words, the reduction in 
	diversity become 
	significant when carrier frequency is high.
	
	
	
	\item {\bf Fay Wu's H} As shown in the Appendix \ref{app:h}, dynamic $H$ 
	statistic is directly related to average HAF \cite{ronen2015predicting} and 
	can be written
	\beq
	nH_t= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
	\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right) 
	\label{eq:ht}
	\eeq
	which is a good approximation only in hard sweep with strong selection, 
	Figures 
	\ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (second
	row, left). Similar to $D$, it can be shown that $H$ statistics for natural 
	selection behave similar to that of neutral evolution in early generations 
	of hard sweep, where carrier frequency is not high.
	
	\item {\bf SFSelect} statistic proposed by Ronen et 
	al. \cite{ronen2013learning} to predict selection by empirically learning 
	the so-called \emph{optimal} weights using Support Vector Machines. 
	Figure \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (forth
	row), depicts the dynamics of SFSelect for selection and neutral evolution 
	in time, which essentially behave similar to $D$ and $H$, i.e. poor power 
	of detection in soft sweep, and early generations of hard sweep.
\end{enumerate}




\subsection{Detecting Selection}
To evaluate detection power of the proposed method, least-squares (LS), we 
compare it with Gaussian process (GP) \cite{Terhorst2015Multi}, FIT 
\cite{feder2014Identifying},  SFselect \cite{ronen2015predicting}, 
$D$\cite{tajima1989statistical}, and $H$ \cite{fay2000hitchhiking} statistics. 
We also take time-series models of $D$ and $H$ into comparison by fitting 
models \eqref{eq:tdt} \eqref{eq:ht} to data. Finally, since SFSelect is 
positive and 
monotone, we considered the aggregation of SFSelect statistics 
along time-series as score for detecting selection.

For every setting, i.e. strength of selection, initial carrier frequency, 
sampling time span and number of replicates, we conduct 200 simulations which 
half of them are neutral and the rest are under selection. Then prediction 
score of each method for all the simulations is used to compute Receiver 
Operating Characteristic (ROC) curve \cite{fawcett2006introduction}, which is 
the primary tool for evaluating binary decision making hypotheses.
Area under ROC  curve (AUC), is measure of predictive performance of a binary 
hypothesis, but it computes the score under all possible setting of 
false-positive rate (FPR). However, here we restrict AUC 
\cite{mcclish1989analyzing} to those region that has FPR$\le$0.1, and 
henceforth we call 0.1-AUC$\times$1000 as power of a method\footnote{Power can 
be interpreted as average true positive rate when FPR$\le$0.1.}. Note that 
power 
rages between 0 to 100 and power of 5 corresponds to random prediction, i.e. 
null hypothesis.

First we compare all the methods under default parameter settings, i.e., 
$T=5$, $\tau=50$ and $R=3$, Figure \ref{fig:power}. It evident that $LS$ method 
provides better detection power in the hard and soft sweep regimes for 
different values of selection strength. Also, except for $H$ in strong 
selections, SFS-based scores exhibit poor performance, comparable to random 
prediction. Also, GP and FIT methods whcih are single-locus methods, reveal 
their best performance in softest sweep scenario, $\nu_0=0.1$.

Also, we assessed the behavior of LS method for different values of 
$\tau\in\{10,50,100\}$, 
Figure \ref{fig:powert}. For the default set of parameters, and $\tau=100$, LS 
method provides significantly better performance, while for $\tau=10$ detection 
power is comparable to those of other methods for $\tau=50$ in Figure 
\ref{fig:power}. Effects of number of replicates on the power of LS method is 
examined for $R\in\{3,5,10\}$, Figure \ref{fig:powerr}. Interestingly, 
increasing number of replicates does not necessarily improve power, yet they 
provide comparable power to the case with $R=3$. This observation implies that 
estimates of power for each method tightly depend on the replicates used for 
evaluation, and computed power of a method potentially under/over-estimated due 
to finite sample size, i.e. \emph{estimation error}. Thus, comparison using 
small number of replicate is meaningful only when difference in performance is 
significant and consistent for different configuration, which is the case for 
LS method in Figure \ref{fig:power}. Finally, Figure \ref{fig:powerrt}, 
demonstrates the power for $R=\{3,10\}$ and $\tau={50,100}$, which shows power 
for $R=3$ and $R=10$ exhibits the similar behavior under different regimes fo 
$\nu_0$, $s$ and $\tau$.

\subsection{Locating the Adaptive Mutation}
The secondary task in identifying selection is to locate the position of the 
adaptive allele. Since de novo mutations are excluded, and linkage is 
reinforced after selection, it is difficult to incorporate phylogeny into this 
task. We consider the site with highest score as the locus of the beneficial 
allele. For each setting of $\nu_0$ and $s$, we conducted 100 
simulations and computed the rank of the beneficial mutation in the sorted 
scores. Then for each configuration distribution of ranks of 100 
simulation is presented via their Cumulative Distribution Function (CDF), 
Figure \ref{fig:rank}. LS works the best in hard sweep regime. For example, 
when $\nu_0=0.005$ and $s=0.01$ (weakest selection), the beneficial allele is 
ranked first, in more than 60 experiments, and take rank of less than 5 in more 
than 90\% of experiments of hard sweep. Accuracy of the locating the adaptive 
allele diminished as sweep become softer, i.e. larger values of $\nu_0$. Yet, 
in the worst case the beneficial allele is tanked among top 50 SNPs.

\subsection{Strength of  Selection}
Finally, estimating the model parameters such as strength of selection provides 
the finest detail of identification of a selective sweep process, because given 
the model parameters we can predict the state of population, e.g. fixation 
time, in future generations. We computed bias, $|s-\hat{s}|$ for each 
experiment of LS, and GP. The distribution of the bias is presented in Figure 
\ref{fig:bias} for different configurations. In general, both GP and  LS are 
biased for weak selections, which genetic drift dominates. However, for 
stronger selections, e.g.$s=0.1$, LS provides estimates with smaller bias and 
variance.

\subsection{Analysis of Real Data}
We finally apply LS method to the controlled experimental evolution experiment 
of \cite{orozco2012adaptation}, which evolves 5 replicates of a population of 
Drosophila melanogaster for 37 generations under alternating 12-hour cycles of 
hot (28$^{\circ}$C) and cold (18$^{\circ}$C) temperatures. Three replicates are 
sampled at the firs generation, 2 replicates at generation 15, one replicate at 
generation 23, one replicate at generation 27 and three replicates at 
generation 37.

Figure \ref{app:af} demonstrates Manhattan plot for the scan of data. Genes 
will be extracgted ASAP.

\section{Materials and Methods}
\subsection{Least-Squares Estimate of $s$} 
\label{sec:regression}
In a simplest case, we have 2 observations, $\nu_0$ and $\nu_t$ and we wish to 
find 
$\hat{s}$ so that $\nu_t=\sigma(\hat{s}t/2+\eta(\nu_0))$ which can be solved 
analytically:
\begin{equation}
\hat{s}=\frac{2}{t} \log \left( \frac{\nu_t(1-\nu_0)}{\nu_0 (1-\nu_t)} \right) 
= 
\frac{2}{t}  \left( \eta(\nu_t)-\eta(\nu_0)\right)
\label{eq:naive2point}
\end{equation}
which selection strength can be interpreted as the normalized frequency 
increment in log-odds space.

However, with more than two observations, the system become over determined and 
we need to solve $T$ set of equations. By re-arranging 
\eqref{eq:naive2point}, we can write a set of linear equations

\beq
t s&=2(\eta(\nu_t)-\eta(\nu_0)) \ \ \ \ \ \forall t \in \Tc \\ 
\eeq
which can be written in matrix form $\bft s =\bfb$ with
\beq
\bft=\begin{bmatrix}
	\tau_1\\
	\vdots \\
	\tau_T
\end{bmatrix}
\bfb=\begin{bmatrix}
 2(\eta(\nu_{\tau_1})-\eta(\nu_0))\\
	\vdots \\
2\eta(\nu_{\tau_T})-\eta(\nu_0))
\end{bmatrix}, \ \ \ \ \ 
\eeq
and the solution for each replicate can be computed using pseudo inverse

\beq
\hat{s}_r=\bft^+\bfb=(\bft^T\bft)^{-1}\bft^T\bfb=\frac{\bft^T\bfb}{\|\bft\|^2}
\eeq
and for multiple independent replicates with the same sampling times we have
\beq
\hat{s}= \frac{1}{R} \sum_r^R \hat{s}_r = \bft^+\bar{\bfb}
\eeq
where $\bar{\bfb}$ is the average of $\bfb$ over $R$ replicates.
Finally, for each replicate, instead of $T$ equations where each observation is 
compared with $x_0$, we can take all possible pairs and solve ${T \choose 2}$ 
set of equations.
\subsection{Composite Likelihood Ratio Test}

Given an estimate for $s$, the Markov chain with transition 
probabilities 
(\cite{Ewens2012Mathematical}, eq 1.58-59)
\beq
Q_{i,j}(s)&=\pr\left(\nu_{t+1}=\frac{j}{2N} \left| 
\nu_{t}=\frac{i}{2N}\right)={2N 
	\choose j} \right. 
\hat{\nu}_{t+1}^{j} (1-\hat{\nu}_{t+1})^{2N-j},\\
\hat{\nu}_{t+1}&=\sigma(\hat{s}+\eta(\nu_t))\label{eq:markovs}
\eeq
computes probabilities of time-series under the alternative 
model. Figure \ref{fig:markovs} shows that the predicted distributions of 
frequencies using Markov chain \ref{eq:markovs} is consistent with simulation 
data.

Likelihood of the trajectories under null and alternative 
hypothesis can be 
computed to perform likelihood ratio tests which provide 
predictors that are 
robust to pathological false-positives 
\cite{feder2014Identifying}. 
Specifically, for each site, we define the likelihood-ratio 
predictor
\beq \label{eq:lrt}
\Lambda(\hat{s}) = \hat{s}\log 
\left(\frac{\Lc(\bfx|s=\hat{s})}{\Lc(\bfx|s=0)}\right)
\eeq
where $\bfx$ is the vector of trajectories and $\Lc$ is the 
likelihood function 
defined by the Markov chain with transition 
\eqref{eq:markovs}.

Given likelihood ratio \eqref{eq:lrt} for each SNP we can further exclude false 
positives by combining likelihood ratio scores of the polymorphisms in a 
genomic region by computing Composite 
Likelihood 
Ratio(CLR)\cite{nielsen2005genomic,williamson2007localizing,vitti2013detecting}.
For each window, CLR is computed by taking average of top one percentile of the 
likelihood ratios.

\section{Discussion}


\section{Appendix}
\subsection{Allele frequencies under selective sweep} \label{app:af}
Let in a dialellic diploid Wright-Fisher model with single locus selection, 
with genotypes 0|0, 0|1 and 1|1 , $w_{00}$, $w_{01}$ and  $w_{11}$ be the 
relative fitness of each genotype at a locus. Also, consider $x$ to be 
frequency of the allele 1 and allele 1 be the favored allele, then
\beq
&w_{00}=1,&w_{01}=1+us,&w_{11}=1+s
\eeq
\beq
x_{t+1}&=\frac{w_{11}x_t^2 + w_{01}x(1-x_t)}{w_{11}x^2_t + 2w_{01}x_t(1-x_t) + 
w_{00}(1-x_t)^2} =x_t+\frac{s(h+(1-2u)x_t)x_t(1-x_t)}{1+sx_t(2u+(1-2u)x_t))} 
\eeq
where $s\in \Rbb$ is the selection coefficient and $o\in[0,1]$ is the 
overdominance parameter which for $u=0.5$ we have
\begin{equation}
x_{t+1}=x_t+\frac{sx_t(1-x_t)}{2+2sx_t}\;.
\label{eq:transition}
\end{equation}
we also have
\beq
\frac{\bfd x_t}{\bfd t} = \frac{sx_t(1-x_t)}{2+2sx_t}
\eeq
which is a differential equation that is difficult to solve. However if take 
the approximation $2+2sx_t \approx 2$, it becomes an ordinary differential 
equation that can be readily solved

\begin{equation}
\nu_t =\frac{1}{1+\frac{1-x_0}{x_0}e^{-st/2}} = \sigma(st/2+\eta(x_0)) 
\label{eq:inf-pop}
\end{equation}
where$\sigma(.)$ is the logistic
function and $\eta(.)$ is logit function (inverse of the logistic function). 

\subsubsection{Fay Wu's H}\label{app:h}
\bl
In any finite population size of $n$ with $m$ segregating sites, 
allele frequencies take 
discrete values, i.e.,  $x_j \in 
\{\frac{1}{n},\frac{2}{n},\ldots,\frac{n-1}{n}\}, \ \forall j \in{1,\cdot,m}$ 
and we can write
\beq
\|\bfx\|^2= \sum_{j=1}^{m} x_j^2 = 
\sum_{i=1}^{n-1}\left(\frac{i}{n}\right)^2\xi_i= 
\frac{ (n-1)}{2n}H 
\eeq
where $\xi_i$ is the number of sites with frequency $i/n$ and $H$ is the 
Fay \& Wu's estimate of $\theta$.
\el

Recently, Ronen et al. ~\cite{ronen2015predicting} devised the $1\dHAF$ 
statistic for identifying selection on static data, which has the expected 
value related to the $\|\bfx\|^2$:
\begin{equation} 
\Ebb[1\dHAF(t)]= n\| \bfx_t\|^2\approx ng(\nu_t)
\end{equation} 
where
\beq
g(\nu_t)= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)
\label{eq:hafscorepooled}
\eeq
where easily follows that
\beq
\theta_H(t)=\frac{n-1}{2} g(\nu_t)
\eeq

\subsection{Tajima's D}\label{app:td}
Let $D_0, \Pi_0, W_0$, be Tajima's D, Tajima's estimate of  $\theta$, and 
Watterson's estimate of $\theta$ at time zero and $D_0=\Pi_0 - W_0$.
In order to compute, $D_t=\Pi_t - W_t$ we compute $\Pi_t$ and $W_t$ separately 
as follows.

Let $P$ be the $n \times n$ matrix of pairwise heterozygosity if individuals, 
then $\Pi=\frac{1}{n^2}\sum P_{ij}$. So, if the population consist of $\nu n$ 
identical carrier haplotype (due to lack of recombination), their pairwise 
hamming distance is zero and should be subtracted from the total $\Pi_t$:
\beq
\Pi_t&= (1-\nu_t^2)\Pi_0 
\eeq

To compute $W_t$, first remember that $W_t= \frac{m_t}{S_n}$ where $m_t$ is the 
number of segregating sites at time $t$ and $S_n= \sum_i^n 1/i \approx 
\log(n)$. Also we have
\beq
\frac{W_t}{W_0}&=\frac{\frac{m_t}{S}}{\frac{m_0}{S}} \ \ \Rightarrow 
W_t=\frac{m_t}{m_0}W0 
\eeq
where $m_t$ to be interpreted as the expected number of segregating sites at 
time $t$, under neutral evolution. At time $t$, the number of individuals that 
undergone neutral evolution is $(1-\nu_t)n +1$, which leads to
\beq
\frac{m_t}{m_0}&=\frac{\log\left((1-\nu_t)n +1 \right)\theta}{\log(n)\theta} 
\approx  
\frac{\log\left((1-\nu_t)n\right)}{\log(n)} = \frac{\log(1-\nu_t)+\log(n)}{\log(n)} = 
1+ \frac{ \log(1-\nu_t)}{\log(n)} 
\eeq
putting all together 
\beq
D_t&= (1-\nu_t^2)\Pi_0 - (1+ \frac{ \log(1-\nu_t)}{\log(n)} ) W_0 = 
D_0-\log(1-\nu_t) \frac{W_0}{\log(n)} -\nu_t^2 \Pi_0
\eeq


\subsection{Linkage Disequilibrium}
Nonrandom associations between polymorphisms are established in the 
substitution process according to the phylogeny, broken by recombination events 
and reinforced by selection. Although in EE the experiments with pooled 
sequencing, LD can not be measured throughout evolution, it is still worthwhile 
to examine the behavior of LD as a result of the interaction between 
recombination and natural selection, to take into account of some of EE 
implicit constraints. 

Let $\rho_0$ be the LD at time zero between the site under selection and a 
segregating site $l$ base-pairs away, then under natural selection we have
\beq
\rho_t= \alpha_t\beta_t \rho_0=e^{-rtl} \left(\frac{H_t}{H0}\right)  
\rho_0\label{eq:ldt}
\eeq
where $H_T=2\nu_0(1-\nu_0)$ is the heterozygousity at the selected site, $r$ is 
the recombination rate/bp/gen. The decay factor $\alpha_t=e^{-rtl}$ is the 
product of recombination and growth factor $\beta_t$ (eq. 30-31 in 
\cite{Stephan2006The})is the outcome of 
selection. For $s=0.01$, $l=100K$bp, the log of decay, growth and product of 
both is depicted in Figure \ref{fig:ldf}. It is evident that, for these 
parameters LD does not start to decay until generation 1000, which would be  
problematic when $\rho_0$. For example, in the case of hard sweep, the 
selection is imposed on the site with minimum AF, which is at perfect linkage 
($|D'|=1$) with all the other loci.\footnote{This is because, between the 
	selected site and all the other sites frequency of one gamete is zero.}
We this phenomenon is shown in the Figures \ref{fig:ld2d}, \ref{fig:ld3d} where 
at generation zero the site at position 500K is at perfect linkage with all the 
other sites, and linkage of the middle site with all the genome is depicted 
for both genetic drift and natural selection, in different generations.
Also, a window of 50Kbp around the selected site is shaded in Figure 
\ref{fig:ld2d} to demonstrate the value of LD in the window under drift and 
hard sweep. This implies that the precision of locating the selection on the 
genome is tightly dependent on a set of parameters including, recombination 
rate, selection strength, initial carrier frequency, and the initial linkage.



\section{Figures}

\begin{figure}[H]
	\centering
	\includegraphics[scale=0.2]{afs}
	\caption{SFS}	\label{fig:afs}
\end{figure}

\begin{figure}[H]
	\begin{tabular}{lr}
		\centering
		\includegraphics[trim={2in 0.1in 1.5in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidSoft}
		&\includegraphics[trim={2in 0.1in 1.9in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidHard}
	\end{tabular}
	\caption{Logistic model for different selection strengths 
		for soft (left) 
		and hard (right) sweep as a function of time in 
		generations. The first 
		50 
		generations, which observations are sampled is 
		shaded.} 	 
	\label{fig:sweep}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim={0in 0.1in 0in 
		0in},clip,width=\textwidth]{{markov0.005}.png}\\
	\includegraphics[trim={0in 0.1in 0in 
		0in},clip,width=\textwidth]{{markov0.1}.png}
	\caption{Predicted distributions of allele frequencies of 
		sites with 
		initial frequency $\nu_0=0.005$ (top) and $\nu_0=0.1$ 
		(bottom), after 
		$t=\{1,10,100\}$ generations, under neutral evolution, by 
		Brownian mothion 
		and Markov chain. Observed distribution of 143900 sites 
		with $\nu_0=0.005$ 
		and 47500 with variants $\nu_0=0.1$ computed from 
		neutrally evolving 
		simulations is depicted in green lines.} 	 
	\label{fig:markov}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[trim={0in 0.1in 0in 
		0in},clip,width=\textwidth]{{markov.s0.1}.png}
	\caption{Predicted distributions of allele frequencies of 
		sites with 
		initial frequency $\nu_0=0.005$ and selection 
		strength $s=0.1$, after 
		$t=\{1,10,100\}$ generations,by Markov chain. 
		Observed distribution of the adaptive allele 
		(with$\nu_0=0.005$) in 
		10000 simulation is shown in green.} 	 
	\label{fig:markovs}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim={0in 0.1in 0in 
		0in},clip,width=\textwidth]{{tdterms}.png}
	\caption{Interactions of two terms in $D$.} 	 
	\label{fig:tdterms}
\end{figure}

\begin{figure}[H]
	\centering 
	\includegraphics[trim=3.2in 0.1in 3.2in 0.2in , 
	clip,width=\textwidth]{bottleneck}
	\caption{Effect of bottle neck in a typical experimental evoloution 
		experiment where a restricted number of founder lines (here $F=200$) is 
		selected out of a larger population size ($N_e=10^{-6}$). Tajima's D 
		(left), Fay Wu's H (middle) and SFSelect is computed for 1000 neutral 
		simulations and mean and 95\% confidence interval is plotted.} 
	\label{fig:bottleneck}
\end{figure}

\begin{figure}[H]
	\centering 
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for strong selection.}
	\label{fig:dynamic-strong}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=4,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for weak selection.}
	\label{fig:dynamic-weak}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{decayFactors}
	\caption{Interaction between productive factors of LD under natural 
		selection for weak selection (s=$0.01$) and a distance of 100Kb between 
		sites. In this setting, after about 1000 generations LD start to decay 
		(red 
		curve).} \label{fig:ldf}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{LDDecay2d}
	\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at position 
		500K with the rest of genome when $s=0.01$ and $r=2\times10^{-8}$. A 
		window 
		of 50Kb is shaded at the center of genome to illustrate high values of 
		linkage in both selection and drift.} \label{fig:ld2d}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{LDDecay3dNeutral}
	\includegraphics[width=\textwidth]{LDDecay3dSweep}
	\caption{ld} \label{fig:ld3d}
	\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at 
		position 500K with the rest of genome in genetic drift with 
		$r=2\times10^{-8}$ (top) and hard sweep with $s=0.01$ (bottom).}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0 , clip,width=\textwidth]{power}
	\caption{Predictive performance of different method is evaluated on 200 
		simulations for different values of selection strength $s$ and initial 
		carrier frequency $\nu_0$.} \label{fig:power}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0 , clip,width=\textwidth]{powerTime}
	\caption{Predictive performance of the proposed method is evaluated 
		on 200 simulations for different values of sampling-time-span 
		$\tau$ and initial carrier frequency $\nu_0$.} \label{fig:powert}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0, 
	clip,width=\textwidth]{powerReplicate}
	\caption{Predictive performance of the proposed method is evaluated on 200 
		simulations for different number of replicates $R$ and initial carrier 
		frequency $\nu_0$.} \label{fig:powerr}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0, 
	clip,width=\textwidth]{powerReplicateTime}
	\caption{Predictive performance of the proposed method is evaluated on 200 
		simulations for different number of replicates $R$, 
		sampling-time-span $\tau$ and initial carrier 
		frequency $\nu_0$.} \label{fig:powerrt}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0, 
	clip,width=\textwidth]{bias}
	\caption{Comparison between the distribution of bias of the proposed method 
		with Gaussian Process (GP), over 100 simulations for different values 
		of 
		selection strength $s$ and soft-sweep with $\nu_0=0.1$.} 
		\label{fig:bias}
\end{figure}



\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{rank}
	\caption{CDF of the rank of the adaptive allele in 100 simulations.} 
	\label{fig:rank}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{{real.manhattan}.png}
	\caption{Manhattan plot of scan for selection using CLR on windows of 
	size 50Kb and sliding window of 10Kb.} 
	\label{fig:real}
\end{figure}

%\input{intro}
%\input{method}
%\input{experiment}
%\input{conclusion}
\newpage
\bibliographystyle{plain}
\bibliography{library}
\end{document}
