\documentclass[11pt]{article}
\input{macros_vineet}
\input{macros_arya}
\title{Detecting Selection in Experimental Evolution Experiments}
\author[1]{Arya Iranmehr}
\author[1]{Ali Akbari}
\author[2]{Vineet Bafna}
\affil[1]{\footnotesize Electrical and Computer Engineering, University of California, San Diego, La Jolla, CA 92093, USA.}
\affil[2]{\footnotesize Computer Science \& Engineering, University of California, San Diego, La Jolla, CA 92093, USA}
\date{}

\begin{document}
\maketitle
\begin{abstract}
Experimental evolution studies are a powerful tool for observing
molecular evolution at work in a controlled environment. The availability of
next-generation sequencing has made it possible to study the genetic trajectory 
of the population at different time points. Methods for investigating 
time-series genomic data are under active development, including recent work by 
Terhorst et al., that uses a Gaussian process approximation to a 
multi-locus Wright-Fisher process.  
In this research, we revisit the problem of detecting selection in short amount 
of time using classical methods such as
Tajima's S and Fay Wu's H and a proposed method which fits a logistic
function to the trajectories of allele frequencies at every locus. An extensive
simulation study shows the proposed method has a superior performance
over conventional method both in hard and soft sweeps.
\end{abstract}

\section{Introduction}
Genetic adaptation is \emph{the} central evolutionary process and is at the
 core of some of the greatest challenges facing humanity.  HIVwould likely
  cause nothing more than harmless fever without the
ability of the virus to adapt and eventually destroy the immune system. 
Cancer would be much more straightforward to treat if not for
tumor's ability to adapt to anti-cancer drugs. 
Malaria could be treated with cheap drugs such as quinine instead of 
being one of the world's worst killers. 
Crop pests would be manageable with small doses of safe insecticides 
instead of requiring applications of ever increasing amounts of a diverse array 
of powerful chemicals. [[Example for Bacteria and TB and some references]]
In both disease and agriculture, we find ourselves in an arms race due to the
ability of organisms to adapt.

Adaptation leaves a variety of detectable signatures in
genomes \cite{nielsen2005genomic,akey2009constructing,
kreitman2000methods,messer2013population,sabeti2006positive}.
The rapid expansion of adaptive alleles in populations leads to both an excess
of functional changes between species and distortions in polymorphism patterns
known as selective sweeps~\cite{nielsen2005genomic}. The signatures of
selective sweeps, which include reduction of levels of polymorphism and the
presence of too many rare alleles~\cite{nielsen2005genomic,
przeworski2002signature}, have been the basis for assessing genomic
patterns in many genome scans for adaptation. Depending on the
\emph{1)data availability}, e.g. allele frequency vs. haplotype, single vs 
multiple  population, single-snapshot vs time-series observation and 
\emph{2)objective of the test}, e.g. detecting recent or ancient
sweep, and \emph{3)simulation model}, e.g. statistical-test vs likelihood-based 
models, empirical recipes for detecting selection are available for each
scenario \cite{sabeti2006positive}.

For example, given allele frequency, a whole family of test such as 
Tajima's \emph{D} ~\cite{tajima1989statistical}, 
Fay and Wu's \emph{H}~\cite{fay2000hitchhiking}, 
SFSelect~\cite{ronen2013learning} compute test statistic as "\emph{a}"
\footnote{Convex combination is a linear combination which weights are positive
	 and sum to one. Under a Bayesian prospective, it can be regarded as a
	  \emph{expectation} which the prior distribution is determined by the
	   weights. As with Bayesian inference, using a \emph{not-reasonable prior}
	    distribution would lead to poor results.} 
	convex 
combination of Allele Frequency Spectrum(AFS)~\cite{achaz2009frequency} 
(Figure \ref{fig:afs}. 
However, despite their simplicity and intuitiveness, it has been shown that
 ~\cite{ptak2002evidence, ramos2002statistical} AFS-based tests often fail 
 to discriminate between positive selection and a dynamic
demographic processes. Also, more recently, Frequency Increment
 Test(FIT) \cite{feder2014Identifying} is  proposed to test whether a
  time-series of allele-frequencies is subjected to a selection pressure.
\begin{figure}
	\centering
	\includegraphics[scale=0.2]{afs}
	\caption{AFS}	\label{fig:afs}
\end{figure}
\paragraph{Experimental Evolution}
Recent advances of whole genome sequencing has enabled us to sequence 
populations at a reasonable cost, such that one could design \emph{experiments} 
to study different forces of \emph{evolution} in-action. \emph{Experimental 
evolution} is the study of the evolutionary processes of a model organism in a 
controlled  
\cite{hegreness2006equivalence,lang2013pervasive,orozco2012adaptation,
	lang2011genetic,barrick2009genome,bollback2007clonal,oz2014strength} 
or natural 
	\cite{maldarelli2013hiv,reid2011new,denef2012situ,winters2012development,
	daniels2013genetic,barrett2008natural,bergland2014genomic} environment.

Types of evolution experiments \cite{Barrick2013Genome} [to be added]

While constraints such as small population sizes, limited timescales and 
oversimplified  laboratory
 environments limits interpreting experimental evolution results, it has
  been used to test different hypothesis~\cite{kawecki2012experimental}
   regarding
mutation and adaptation, 
genetic drift and inbreeding, 
environmental variability,
sexual selection and conflict, 
kin selection and cooperation
life history and sex allocation, 
sexual reproduction and mating systems, 
behavior and cognition, 
host–parasite interactions, 
speciation
repeatability of evolution.

In addition, time-series \emph{models} is shown to be more powerful 
\cite{boyko2008assessing,desai2008polymorphism,sawyer1992population} 
than single-snapshot data.
In particular, dynamic data has been used to estimate model parameters 
including population size
\cite{williamson1999using,wang2001pseudo,pollak1983new,waples1989generalized,
	Terhorst2015Multi}
strength of selection
\cite{mathieson2013estimating,illingworth2011distinguishing,Terhorst2015Multi,
	bollback2008estimation,illingworth2012quantifying,malaspinas2012estimating,
	Steinrücken2014a, malaspinas2012estimating}, allele age
 \cite{malaspinas2012estimating}
recombination rate~\cite{Terhorst2015Multi}  and mutation
 rate~\cite{Barrick2013Genome, Terhorst2015Multi}. 

Linked-loci time-series (spatio-temporal) methods 
\cite{illingworth2011distinguishing,illingworth2012quantifying,
	Barrick2013Genome,Terhorst2015Multi} incorporate linkage into computation
 which makes it cumbersome. Theoretical diffusion approximation method has also 
 developed for neutral allele frequencies~\cite{Ewens2012Mathematical, 
 kimura1955solution}
neutral AFS~\cite{evans2007non}, neutral allele
 frequencies~\cite{song2012simple}.

Experimental evolution (including evolve and re-sequence paradigms)
have become increasingly popular as a complementary tool to understand
the forces of selection, allowing for controlled environments,
specific selection constraints. Examples of experimental evolution
studies abound in sexual populations, particularly in fruitfly. Burke
et al.~\cite{Barrick2013Genome} evolved flies for over $600$ generations 
under
selection for accelerated development, and noted that evolution for
sexual populations is very different from those of asexual populations
(e.g., Lenski): the effects of clonal interference is slower
due to recombination that allows for the incorporation of multiple
beneficial alleles, but there are fewer, unconditionally advantageous
alleles that arise at the onset of selection. Rose and
Colleagues~\cite{rose1994evolutionary} created $200$ experimentally evolved
populations selected for different traits, starting with $10$ initial
populations. Zhou et al. evolved flies to adapt to low oxygen
environment (hypoxia) for over $300$ generations, and identified many
genes involved in hypoxia tolerance~\cite{zhou2011experimental}.  Instead, 
they observed
incomplete fixation (`soft-sweeps') due in part to standing variation,
changing selection coefficients, and small fitness effects. Similarly, Jha et
 al. \cite{jha2015whole} identified genes under selection for egg size in flies.
Barrick et al. \cite{barrick2009genome} evolved a population of 
\emph{Escherichia coli} for 40,000 generations.

Much like in natural populations, many of these studies also
sequenced/genotyped only the latest population. However, the emergence
of NGS and related technologies has made it feasible to sequence the
evolving population at multiple time points in its evolution. At the
same time, for small organisms where a single animal does not provide
enough DNA, it is more effective to pool together individuals in the
population at a particular time point, in a single sequencing
experiment. Thus, instead of individual genotypes, we obtain
frequencies of the derived allele at all loci at different time
points. Methods for analyzing such time series pooled sequence data
are only just being developed.

\paragraph{Asexual vs Sexual}
\cite{bollback2008estimation,lenski1991long,lang2013pervasive} [to be 
added]


\section{Results}
\paragraph{Notation} Let $\bfX \triangleq (X_{ijk}) \in [0, 1]^{T \times 
M\times R}$ denote the population frequency where $T$ is the number of samples 
in time, $M$ is the number of segregating sites, and $R$ is the number of 
replicates.
To simplify our notation, we define $x_t$ as a allele frequency of a site in a
 replicate, and we denote it by $\nu_t$ if such a site is directly under 
 selection with the strength $s$. I.E., $\nu_t$ can be regarded as frequency of 
 the carrier in the  population.
\paragraph{Simulations}
For each experiment a diploid population is created and evolved as follows. 
\begin{enumerate}[I.]
	\item {\bf Creating initial founder line haplotypes}
	First using msms program, we created neutral populations for $F$ founding 
	haplotypes with \emph{default} parameters \texttt{\$./msms <F> 1 -t 
	<2$\mu$LNe> 
		-r <2rNeL> 
		<L>} 
	where $F=200$ is number of founder lines, $N_e=10^6$ is 
	effective 
	population size, $r=2*10^{-8}$ is recombination rate and $\mu=2\times 
	10^{-9}$ is mutation rate and  $L=50K$ is the window size in base pairs 
	which gives $\theta=2\mu N_eL=200$ and $\rho=2N_erL=2000$. For default 
	parameter, the expected number of segregating sites in a window is 
	\beqq
	\Ebb[M]=\theta \sum_{i=1}^{F-1}\frac{1}{i}=1175
	\eeqq
	\item{\bf Creating initial diploid population} 
	To implement similar setting for experimental evolution of diploid 
	organisms, 
	initial  haplotypes first cloned to create $F$ diploid homozygotes. Then 
	each 
	diploid individual is  cloned $N/F$ times to yield diploid population of 
	size 
	$N$.
	\item{\bf Forward Simulation}
	Given initial diploid population, position of the site under selection, 
	selection 
	strength $s$, number of replicates $R=3$, recombination rate 
	$r=2\times10^{-8}$ 
	and sampling times $\Tc=\{10,20,30,40,50\}$, \texttt{simuPop} is used to 
	perform
	forward simulation and  compute allele frequencies for all of the $R$ 
	replicates.
\end{enumerate}

\subsection{Dynamics of the Sweep}
Selective sweep is a random process which depends on selection strength
 and population size and affects \emph{site allele frequencies} , \emph{allele 
 frequency spectrum} and \emph{linkage disequilibrium}.
 In these part we study the dynamics of these quantities along selective sweep 
 to propose an efficient method to identify and locate selection.
\paragraph{Allele Frequencies}
Selection and genetic drift are two main \emph{independent} evolutionary forces 
that cause change in allele frequency at a segregating site.
 Whether the change in allele frequencies is caused by genetic drift or natural
  selection or both is the main question in detecting selection in the dynamic
   data. Basically, the strength of selection is measured by selection 
   coefficient $s$ and the strength of drift is proportional to 
   $\frac{1}{N_e}$, and to detect selection $N_es>1$ should hold.
  

More precisely, under genetic drift, variations in allele frequencies is a 
random walk, which can be modeled via Moran process~\cite{moran1962statistical}.
In fact, joint distribution of allele frequencies grows exponentially with 
number of observations which makes computing probabilities extremely difficult 
(due to tedious marginalizations).

However, it is possible to approximate change in allele frequencies
with a Brownian motion~\cite{Ewens2012Mathematical}. 
Brownian motion is a random process which is identified by zero-mean 
identically and independently distributed Gaussian increments. Feder et al. 
\cite{feder2014Identifying} designed a neutrality test, Frequency Increment 
Test (FIT), which tests \emph{scaled} increments in observed allele frequencies 
under Null distribution, i.e. $\Nc(0,1/N_e)$\footnote{Note that the strength of 
drift is proportional to $\frac{1}{N_e}$}.

In the selective sweep with infinite population size, dynamic of allele frequencies 
can be approximated by the logistic function (see Appendix \ref{app:af} for 
derivation)
\beq
\nu_t=\sigma(st+\eta(\nu_0))\label{eq:nut}
\eeq
where $sigma(.)$ is the logistic function, and $\eta(.)$ is inverse of the 
logistic 
function, aka logit.
Depending on the 
value of allele frequency at the time selection started, selective sweep is often 
categorized in two main regimes: hard and soft sweep, Figure \ref{fig:sweep}. 
Hard sweep occurs when the initial allele (carrier) frequency $\nu_0=1/N_e$ is 
at 
minimum, and soft-sweep otherwise. This small difference could have nontrivial
consequences~\cite{schrider2015soft} [[explain more]].

As shown in Figures \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (first 
row), the approximate logistic model seems to be consistent with data in both 
hard and soft-sweep with both weak and strong selection. Hence, we use the 
logistic model for allele frequencies to estimate the strength of selection for each 
site.
\begin{figure}[t]
	\begin{tabular}{lr}
		\centering
		\includegraphics[trim={2in 0.1in 1.5in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidSoft}
		&\includegraphics[trim={2in 0.1in 1.9in 
			0in},clip,page=2,width=0.5\textwidth]{sigmoidHard}
	\end{tabular}
	\caption{soft (left) and hard (right) sweep.} 	 \label{fig:sweep}
\end{figure}


\paragraph{Allele Frequency Spectrum} AFS is the histogram of allele 
frequencies in a fixed window, which depends on allele frequencies of all sites. 
While Evans et al. \cite{evans2007non} developed diffusion equations for 
evolution of AFS, they are difficult to solve. However, here instead of analyzing 
the whole distribution of AFS, we \emph{sclarize} the distribution and study the 
behavior of it. Scalarization can be done by a weighted linear cobmination of the 
AFS vector. We choose Tajima's D, Fay Wu's H and SFSelect statistics weights to 
scalarize the AFS. In other words, we analyze test statistics for Tajima, Fay \& Wu 
and SFSelect tests.
\begin{enumerate}[I.]
	\item {\bf Tajima D} As shown in the Appendix \ref{app:td}, dynamic of 
	Tajima's D in hard sweep is
	\beq
	D_t=D_0-\log(1-\nu_t) \frac{W_0}{\log(2N)} -\nu_t^2 \Pi_0\label{eq:tdt}
	\eeq
	where $\nu_t$ defined in term of $\nu_0$ and $s$ in \eqref{eq:nut}, $W_0$ 
	and $\Pi_0$ are Watterson and Tajima estimate of $\theta$ at the first 
	generation. Given $W_0$, $\Pi_0$ and $\nu_t \in (0,1)$, $D_t$ 
	\eqref{eq:tdt} is a one-to-one map of $\nu_t$, which exhibits the same 
	trend as carrier frequency $\nu_t$, but potentially with higher variance.
	Figures \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (last 
	row, left), shows that the parametric model is reasonable, but observations 
	of $D_t$ has higher variance than $\nu_t$ which makes it harder to 
	distinguish between neutral and under sweep population. 
	
	In addition to their variance, the mean of the processes in both cases 
	(even for strong selection)
	seems to be entangled to each other in early generations  that could lead 
	to high rates of false-positives (or false negatives). We conjecture that, 
	this is because Tajima's D is a weighted average of AFS and this averaging 
	cancels out small fluctuations of AFS in early generations.

	Finally, in derivation of \eqref{eq:tdt} it has been assumed that the 
	population undergoes a hard-sweep, and the model becomes invalid as the 
	initial carrier frequency deviates from that of hard-sweep, Figure 
	\ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (last 
	row, right).
	
	\item {\bf Fay Wu's H} As shown in the Appendix \ref{app:h}, dynamic of Fay 
	Wu's H in 	hard sweep is
	\beq
	nH_t= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
	\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right) 
	\label{eq:ht}
	\eeq
	which is an acceptable approximation only in strong selection, Figures 
	\ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (second
	row, left). Similar to $D_t$, $H_t$ is an injection of $\nu_t$ which only 
	increases its variance and centers the means of two processes in early 
	generations. Likewise, \eqref{eq:ht} is only developed for hard sweep which 
	makes it impractical to use it on real data.
	
	\item {\bf SFSelect} is also another test statistic proposed by Ronen et 
	al. \cite{ronen2013learning} to predict selection by empirically learning 
	the so-called \emph{optimal} weights using Support Vector Machines. 
	Figure \ref{fig:dynamic-weak} and \ref{fig:dynamic-strong} (forth
	row), depicts the dynamics of SFSelect for selection and neutral evolution 
	in time which differentiates better than D and H statistics only in the 
	hard sweep scenario.
\end{enumerate}
\begin{figure}
	\centering 
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=1,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for strong selection.}
	\label{fig:dynamic-strong}
\end{figure}

\begin{figure}
	\centering
	\includegraphics[trim={2in 0.5in 2in 
		0in},clip,page=4,width=\textwidth]{{GlobalDynamics.TimeSeries}.pdf}
	\caption{Mean and 95\% CI of 1000 simulations for weak selection.}
	 \label{fig:dynamic-weak}
\end{figure}
\paragraph{Linkage Disequilibrium}
Nonrandom associations between polymorphisms are established in the 
substitution process according to the phylogeny, broken by recombination events 
and reinforced by selection. Although in EE the experiments with pooled 
sequencing, LD can not be measured throughout evolution, it is still worthwhile 
to examine the behavior of LD as a result of the interaction between 
recombination and natural selection, to take into account of some of EE 
implicit constraints. 

Let $\rho_0$ be the LD at time zero between the site under selection and a 
segregating site $l$ base-pairs away, then under natural selection we have
\beq
\rho_t= \alpha_t\beta_t \rho_0=e^{-rtl} \left(\frac{H_t}{H0}\right)  
\rho_0\label{eq:ldt}
\eeq
where $H_T=2\nu_0(1-\nu_0)$ is the heterozygousity at the selected site, $r$ is 
the recombination rate/bp/gen. The decay factor $\alpha_t=e^{-rtl}$ is the 
product of recombination and growth factor $\beta_t$ (eq. 30-31 in 
\cite{Stephan2006The})is the outcome of 
selection. For $s=0.01$, $l=100K$bp, the log of decay, growth and product of 
both is depicted in Figure \ref{fig:ldf}. It is evident that, for these 
parameters LD does not start to decay until generation 1000, which would be  
problematic when $\rho_0$. For example, in the case of hard sweep, the 
selection is imposed on the site with minimum AF, which is at perfect linkage 
($|D'|=1$) with all the other loci.\footnote{This is because, between the 
selected site and all the other sites frequency of one gamete is zero.}
We this phenomenon is shown in the Figures \ref{fig:ld2d}, \ref{fig:ld3d} where 
at generation zero the site at position 500K is at perfect linkage with all the 
other sites, and linkage of the middle site with all the genome is depicted 
for both genetic drift and natural selection, in different generations.
Also, a window of 50Kbp around the selected site is shaded in Figure 
\ref{fig:ld2d} to demonstrate the value of LD in the window under drift and 
hard sweep. This implies that the precision of locating the selection on the 
genome is tightly dependent on a set of parameters including, recombination 
rate, selection strength, initial carrier frequency, and the initial linkage.


\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{decayFactors}
	\caption{Interaction between productive factors of LD under natural 
	selection for weak selection (s=$0.01$) and a distance of 100Kb between 
	sites. In this setting, after about 1000 generations LD start to decay (red 
	curve).} \label{fig:ldf}
\end{figure}


\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{LDDecay2d}
	\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at position 
	500K with the rest of genome when $s=0.01$ and $r=2\times10^{-8}$. A window 
	of 50Kb is shaded at the center of genome to illustrate high values of 
	linkage in both selection and drift.} \label{fig:ld2d}
\end{figure}


\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{LDDecay3dNeutral}
	\includegraphics[width=\textwidth]{LDDecay3dSweep}
		\caption{ld} \label{fig:ld3d}
			\caption{Decay of LD ($|D'|$ measure) of the minimum AF site at 
			position 500K with the rest of genome in genetic drift with 
			$r=2\times10^{-8}$ (top) and hard sweep with $s=0.01$ (bottom).}
\end{figure}


\paragraph{Challenges}
The nature of experimental evolution, suggests that the duration of the 
experiment is ruled by the organism generation time, and given a limited time 
for the experiment, the population can be evolved for sampling-time-span (STS) 
$\tau=\tau_T-\tau_0$  number of generations. This can be regarded as a 
limitation in a EE experiment when STS is significantly smaller than fixation 
time for a selection with strength $s$ imposed on a site with frequency 
$\nu_0$. For example, in Figure \ref{fig:sweep} the, STS is 50 generations and 
the frequency of the site under selection is $0.005$ (right, hard sweep) and 
$0.1$ (left, soft sweep). 
For $s=0.01$, the mean fixation time is approximately 2100 and 1500 generations 
for the hard and soft sweep, receptively, which makes it extremely larger than 
$STS$. This can be observed by noticing the small slope of the green line in 
the shaded area, which is similar to that of genetic drift.

Moreover, the Figure \ref{fig:sweep} depicts selective sweep for an infinite 
population size, i.e., no drift. In practice, not only the population size is 
limited, but also it is severely bottlenecked to the number of founder-lines to 
inbreed. For example in the case of EE with D. melanogaster, the population 
size of millions is restricted to hundreds which makes genetic drift quite 
strong to disturb (and with small probability, reinforce) the signal for 
selection.



To quantitatively show the effects of STS and population size on the signal of 
selection, it has been shown that (in the appendix), the null distribution of 
the selection strength (under neutral evolution) is approximately distributed 
as $\Nc(0,\sigma^2_{\nu_0}/R\tau N_e)$ while the alternative distribution of 
selection strength is distributed as $\Nc(s,\sigma^2_{\nu_0}/R\tau N_e)$. The 
variance of both distributions inversely depend on population size, 
STS, and the number of replicates. In order to separate 
these distributions 
\begin{enumerate}
	\item the mean of the distributions be far from each other, i.e. selection 
	strength should be large.
	\item the variance of the distributions be small, i.e. population size, 
	number of replicates or STS be large.
\end{enumerate}
Interestingly some of these parameters can be determined both in a controlled 
or natural experimental evolution and by the trade-offing between them one can 
achieve the desired predictive power.
For example, in a controlled EE, when STS is constrained to be small number of 
generation, due to long generation times of the model organism, the population 
size can be increased by gathering more founder lines. Alternatively, the small 
\begin{figure}
	\centering \label{fig:bottleneck}
	\includegraphics[trim=3.2in 0.1in 3.2in 0.2in , 
	clip,width=\textwidth]{bottleneck}
		\caption{Effect of bottle neck in a typical experimental evoloution 
		experiment where a restricted number of founder lines (here $F=200$) is 
		selected out of a larger population size ($N_e=10^{-6}$). Tajima's D 
		(left), Fay Wu's H (middle) and SFSelect is computed for 1000 neutral 
		simulations and mean and 95\% confidence interval is plotted.} 
		\label{fig:bott}
\end{figure}



\subsection{Detecting Selection}
\paragraph{Hard Sweep vs Soft Sweep}
\begin{figure}
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0 , clip,width=\textwidth]{power}
		\caption{Predictive performance of different method is evaluated on 200 
		simulations for different values of selection strength $s$ and initial 
		carrier frequency $\nu_0$.} \label{fig:power}
\end{figure}



\paragraph{Effects of carrier frequency}
\paragraph{Effects of sampling times}
\begin{figure}
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0 , clip,width=\textwidth]{powerTime}
			\caption{Predictive performance of the proposed method is evaluated 
			on 200 simulations for different values of sampling-time-span 
			$\tau$ and initial carrier frequency $\nu_0$.} \label{fig:powert}
\end{figure}
\paragraph{Effects of number of replicates}
\begin{figure}
	\centering
\includegraphics[trim=2.2in 0 2.2in 0, clip,width=\textwidth]{powerReplicate}
\caption{Predictive performance of the proposed method is evaluated on 200 
simulations for different number of replicates $R$ and initial carrier 
frequency $\nu_0$.} \label{fig:powerr}
\end{figure}
\begin{figure}
	\centering
\includegraphics[trim=2.2in 0 2.2in 0, 
clip,width=\textwidth]{powerReplicateTime}
	\caption{Predictive performance of the proposed method is evaluated on 200 
			simulations for different number of replicates $R$, 
			sampling-time-span $\tau$ and initial carrier 
			frequency $\nu_0$.} \label{fig:powerrt}
\end{figure}

\subsection{Strength of  Selection}
\begin{figure}
	\centering
	\includegraphics[trim=2.2in 0 2.2in 0, 
	clip,width=\textwidth]{bias}
	\caption{Comparison between the distribution of bias of the proposed method 
	with Gaussian Process (GP), over 100 simulations for different values of 
	selection strength $s$ and soft-sweep with $\nu_0=0.1$.} \label{fig:bias}
\end{figure}


\subsection{Locating the adaptive mutation}
\begin{figure}
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{rank}
	\caption{CDF of the rank of the adaptive allele in 100 simulations.} 
	\label{fig:rank}
		\end{figure}
\section{Materials and Methods}
\newpage

\subsection{Least-Squares Estimate of $s$}
In a simplest case, we have 2 observations, $x_0$ and $x_t$ and we wish to find 
$\hat{s}$ so that $x_t=\sigma(\hat{s}t/2+\eta{x_0})$ which can be solved 
analytically:
\begin{equation}
\hat{s}=\frac{2}{t} \log \left( \frac{x_t(1-x_0)}{x_0 (1-x_t)} \right) = 
\frac{2}{t}  \left( \eta(x_t)-\eta(x_0)\right)
\label{eq:naive2point}
\end{equation}
which selection coefficient can be regarded as for the
selection coefficient as normalized increment in log-odds space.

However, with more than two observations, the system become over determined and 
we need to solve $T$ set of equations. By re-arranging 
\eqref{eq:naive2point}, we can write a set of linear equations

\beq
\frac{t}{2}s&=\eta(x_t)-\eta(x_0) \ \ \ \ \ \forall t \in \Tc \\ 
\eeq
which can be written in matrix form $\bft s =\bfb$ with
\beq
\bft=\begin{bmatrix}
	\tau_1\\
	\vdots \\
	\tau_T
\end{bmatrix}
\bfb=\begin{bmatrix}
 2(\eta(x_{\tau_1})-\eta(x_0))\\
	\vdots \\
2\eta(x_{\tau_T})-\eta(x_0))
\end{bmatrix}, \ \ \ \ \ 
\eeq
and the solution is can be computer using psudo inverse

\beq
\hat{s}=\bft^+\bfb=(\bft^T\bft)^{-1}\bft^T\bfb=\frac{\bft^T\bfb}{\|\bft\|^2}
\eeq
it can be shown that 
\beq
s^*= \frac{1}{R} \sum_r^R \hat{s}_r = \bft^+\bar{\bfb}
\eeq
where $\bar{\bfb}$ is the average of $\bfb$ over $R$ replicates.
Finally, for each replicate, instead of $T$ equations where each observation is 
compared with $x_0$, we can take all possible pairs and solve ${T \choose 2}$ 
set of equations.


\newpage
\subsection{Null and Alternative Distributions of $s$}
Consider a diallelic single-locus (Wright-Fisher like) model with no
mutations \cite{Ewens2012Mathematical}, discrete generations, random
mating etc. With finite population size, the allele frequencies from
generation to generation are described by the random process
$\{x_{\ell,t}\}$ which denotes the population allele frequency at
locus $\ell$ at time generation $t$. The model has access to finite
number of observations in time for each locus, where the sampling
times are given by the set $\Tc=\{\tau_1,\tau_2,\cdots, \tau_T \}$
such that $\tau_1<\tau_2,\cdots<\tau_T$. We also assume that
measurements are taken up for $R$ experimental replicates for $L$
loci. Thus, The allele frequencies of a population are given by the
tensor $\bfX \in [0,1]^{R \times L \times T}$, where $X_{r,\ell,t}$
stores the observed of allele frequency at replicate $r$, locus $\ell$
and generation $\tau_t$. However, we often omit $r$ for clarity of
exposition. A neutrally evolving allele `drifts', and for a finite
population,
\beq x_{t+1} = x_t + \epsilon\; .\footnote{With infinite population size
	we have $x_{t+1} = x_t$, a restatement of Hardy-Weinberg
	equilibrium.} \ \ \ \ \ \epsilon \sim \Nc(0,\frac{h_t}{N_e})  \
\label{eq:drift}
\eeq
where $h_t=2\nu_t(1-\nu_t)$ is heterozygousity at time $t$.
Using Brownian motion approximation, we have 
\beq x_{\tau} = x_0 + \epsilon_\tau,  \ \ \ \ \ \epsilon \sim 
\Nc(0,\frac{h_0 \tau}{N_e})
\eeq

\newpage
\section{Appendix}
\subsection{Allele frequencies under selective sweep} \label{app:af}
Let in a dialellic diploid Wright-Fisher model with single locus selection, 
with genotypes 0|0, 0|1 and 1|1 , $w_{00}$, $w_{01}$ and  $w_{11}$ be the 
relative fitness of each genotype at a locus. Also, consider $x$ to be 
frequency of the allele 1 and allele 1 be the favored allele, then
\beq
&w_{00}=1,&w_{01}=1+us,&w_{11}=1+s
\eeq
\beq
x_{t+1}&=\frac{w_{11}x_t^2 + w_{01}x(1-x_t)}{w_{11}x^2_t + 2w_{01}x_t(1-x_t) + 
w_{00}(1-x_t)^2} =x_t+\frac{s(h+(1-2u)x_t)x_t(1-x_t)}{1+sx_t(2u+(1-2u)x_t))}
\eeq
where $s\in \Rbb$ is the selection coefficient and $o\in[0,1]$ is the 
overdominance parameter which for $u=0.5$ we have
\begin{equation}
x_{t+1}=x_t+\frac{sx_t(1-x_t)}{2+2sx_t}\;.
\label{eq:hequalshalf}
\end{equation}
we also have
\beq
\frac{\bfd x_t}{\bfd t} = \frac{sx_t(1-x_t)}{2+2sx_t}
\eeq
which is a differential equation that is difficult to solve. However if take 
the approximation $2+2sx_t \approx 2$, it becomes an ordinary differential 
equation that can be readily solved

\begin{equation}
\nu_t =\frac{1}{1+\frac{1-x_0}{x_0}e^{-st/2}} = \sigma(st/2+\eta(x_0)) 
\label{eq:inf-pop}
\end{equation}
where$\sigma(.)$ is the logistic
function and $\eta(.)$ is logit function (inverse of the logistic function). 

\subsubsection{Fay Wu's H}\label{app:h}
\bl
In any finite population size of $n$ with $m$ segregating sites, 
allele frequencies take 
discrete values, i.e.,  $x_j \in 
\{\frac{1}{n},\frac{2}{n},\ldots,\frac{n-1}{n}\}, \ \forall j \in{1,\cdot,m}$ 
and we can write
\beq
\|\bfx\|^2= \sum_{j=1}^{m} x_j^2 = 
\sum_{i=1}^{n-1}\left(\frac{i}{n}\right)^2\xi_i= 
\frac{ (n-1)}{2n}H 
\eeq
where $\xi_i$ is the number of sites with frequency $i/n$ and $H$ is the 
Fay \& Wu's estimate of $\theta$.
\el

Recently, Ronen et al. ~\cite{ronen2015predicting} devised the $1\dHAF$ 
statistic for identifying selection on static data, which has the expeceted 
value related to the $\|\bfx\|^2$:
\begin{equation} 
\Ebb[1\dHAF(t)]= n\| \bfx_t\|^2\approx ng(\nu_t)
\end{equation} 
where
\beq
g(\nu_t)= \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
\theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)
\label{eq:hafscorepooled}
\eeq
where easily follows that
\beq
\theta_H(t)=\frac{n-1}{2} g(\nu_t)
\eeq

\subsection{Tajima's D}\label{app:td}
Let $D_0, \Pi_0, W_0$, be Tajima's D, Tajima's estimate of  $\theta$, and 
Watterson's estimate of $\theta$ at time zero and $D_0=\Pi_0 - W_0$.
In order to compute, $D_t=\Pi_t - W_t$ we compute $\Pi_t$ and $W_t$ separately 
as follows.

Let $P$ be the $n \times n$ matrix of pairwise heterozygosity if individuals, 
then $\Pi=\frac{1}{n^2}\sum P_{ij}$. So, if the population consist of $\nu n$ 
identical carrier haplotype (due to lack of recombination), their pairwise 
hamming distance is zero and should be subtracted from the total $\Pi_t$:
\beq
\Pi_t&= (1-\nu_t^2)\Pi_0 
\eeq

To compute $W_t$, first remember that $W_t= \frac{m_t}{S_n}$ where $m_t$ is the 
number of segregating sites at time $t$ and $S_n= \sum_i^n 1/i \approx 
\log(n)$. Also we have
\beq
\frac{W_t}{W_0}&=\frac{\frac{m_t}{S}}{\frac{m_0}{S}} \ \ \Rightarrow 
W_t=\frac{m_t}{m_0}W0 
\eeq
where $m_t$ to be interpreted as the expected number of segregating sites at 
time $t$, under neutral evolution. At time $t$, the number of individuals that 
undergone neutral evolution is $(1-\nu_t)n +1$, which leads to
\beq
\frac{m_t}{m_0}&=\frac{\log\left((1-\nu_t)n +1 \right)\theta}{\log(n)\theta} 
\approx  
\frac{\log\left((1-\nu_t)n\right)}{\log(n)} = \frac{\log(1-\nu_t)+\log(n)}{\log(n)} = 
1+ \frac{ \log(1-\nu_t)}{\log(n)} 
\eeq
putting all together 
\beq
D_t&= (1-\nu_t^2)\Pi_0 - (1+ \frac{ \log(1-\nu_t)}{\log(n)} ) W_0 = 
D_0-\log(1-\nu_t) \frac{W_0}{\log(n)} -\nu_t^2 \Pi_0
\eeq


\subsection{Markov Chain}
Without loss of generality the allele frequencies in a finite diploid population 
take discrete values in $\{i/2N\}$ where $i\in\{1,\ldots,2N-1\}$. Thus, the 
process of allele frequency can be regarded as a discrete Markov chain and in 
the folloing we specify transition probabilities for neutral evolution and natural 
selection.
\paragraph{Neutral Evolution}
The randomness imposed by a genetic drift can be regarded as a random 
noise distributed by a bionomial sampling. Therefore
\beq
p_{i,j}\triangleq \pr\left(x_{t+1}=\frac{j}{2N} \left| 
x_{t}=\frac{i}{2N}\right)={2N 
\choose j} \right. 
x_{t}^j (1-x_{t})^{2N-j} 
\eeq

\paragraph{Natural Selection}

\beq
p^s_{i,j}\triangleq \pr\left(x_{t+1}=\frac{j}{2N} \left| 
x_{t}=\frac{i}{2N}\right)={2N 
	\choose 2Ny_t} \right. 
y_{t}^{2Ny_t} (1-y_{t})^{2N(1-y_{t})} 
\eeq


\section{Old Stuff}
We can compute the null distribution for selection strength of $s_N$ under 
neutrality (see appendix 
\ref{app:null}):
\beq
s_N \sim  \Nc(0,\frac{16}{\tau N_eR h_0 } )
\eeq

Since in the selective sweep, both drift and selection contribute to the change 
in allele frequencies, they also have an effect in the signal of selection 
(selection strength) in a variant. In appendix \ref{app:adds} we show that 
these effects are additive, so the alternative distribution is
\beq
s_{A} \sim \Nc(s,\frac{16}{\tau N_eR h_0} )
\eeq
where $\tau$ is sampling-time-span, and $h_0$ is initial heterozygosity 
($2\nu_0(1-nu_0)$) .


\subsection{Null Distribution for Selection Strength}\label{app:null}

\bl
Let $X$ is a random variable with $\Ebb[X]=\mu_X$ and 
$\Var[X]=\sigma_X^2$
\beq
\Ebb\left[f(X)\right] & {} = \Ebb\left[f(\mu_X+ 
\left(X - \mu_X\right))\right] \\
& \approx \Ebb\left[f(\mu_X) + f'(\mu_X) \left(X-\mu_X\right)+ 
\frac{1}{2}f''(\mu_X) \left(X - \mu_X\right)^2 \right]=\\
&=f(\mu_X)  + \frac{1}{2}f''(\mu_X) \sigma^2\\
\Var\left[f(X)\right]&\approx 
\left(f'(\Ebb\left[X\right])\right)^2 \Var\left[X\right]
= \left(f'(\mu_X)\sigma_X\right)^2.
\eeq
Taking $f(X)=\log(\frac{X}{1-X}) $, we have $f'(X)=\frac{1}{X(1-X)}$  and 
$f''(X)=\frac{1-2X}{(X(1-X))^2}$, then
\beq
\Ebb[\log(X)]&\approx \log\left(\frac{\mu_X}{1-\mu_X}\right)- 
\frac{1-2\mu_X}{(\mu_X(1-\mu_X))^2} \sigma^2_X\\
\Var[\log(X)]&\approx \frac{\sigma_X^2}{(\mu_X(1-\mu_X))^2}
\eeq \label{lemma:log}
\el
therefore $\mu_X=\nu_0$ $\sigma_X=\frac{2\nu_0(1-\nu_0)t}{N_e}$
\beq
\log\left( \frac{\nu_t}{1-\nu_t} \right) \sim \Nc\left( \eta(\nu_0)-   
\frac{2(1-2\nu_0)}{N\nu_0(1-\nu_0)} , 
\frac{2t}{N\nu_0(1-\nu_0)} \right)
\eeq
which follows
\beq
s \sim \Nc(0,\frac{4t}{N h_0 } (\frac{2}{t})^2)= \Nc(0,\frac{16}{tN h_0} )
\eeq 
also for R replicates we have
\beq
s \sim  \Nc(0,\frac{16}{tNR h_0 } )
\eeq 
\subsection{Additivity of $s$}\label{app:adds}
In a selective sweep, the change in AF is caused by both drift and selection 
which means that we can quantify the observed selection as combination of 
drift 
and sweep selection, $s_d$ and $s_s$ respectively:
\beq
x_t&=x_0 +\delta \Rightarrow x_t=\sigma(st+\eta(x_0)) \\
x_t&=x_0 +\delta_s +\delta_d\\
x_t-x'+x'&=x_0 +\delta_s +\delta_d \rightarrow 
\begin{cases}
	x'=x_0+\delta_d \rightarrow x'=\sigma(ts_d+\eta(x_0)) \\
	x_t=x'+\delta_s \rightarrow x_t=\sigma(ts_s+\eta(x'))
\end{cases} 
\eeq
\beq
x_t&=\sigma(ts_s+\eta(x'))=\sigma(ts_s+\eta(\sigma(ts_d+\eta(x_0)) 
))=\sigma(ts_s+ts_d+\eta(x_0))\Rightarrow s=s_d+s_s
\eeq
It should be noted that $\nu_t$ and $1-\nu_t$ are identically distributed, 
around their means:
\beq
\nu_t&\sim \Nc(\nu_0,\frac{h_0 t}{N_e}) , \ \ \ \ \ 
1-\nu_t\sim \Nc(1-\nu_0,\frac{h_0 t}{N_e}),
\eeq
\beq
\frac{1-\nu_t}{1-\nu_0}&\sim \Nc(1,\frac{\sigma^2}{(1-\nu_0)^2}) \Rightarrow 
\log \left( \frac{1-\nu_t}{1-\nu_0} \right)  \sim 
\Nc(0,\frac{\sigma^2}{(1-\nu_0)^2})
\eeq

\beq
s=&\frac{2}{t} \log \left( \frac{x_t(1-x_0)}{x_0 (1-x_t)} \right) = 
\frac{2}{t}\left\{ \log \left( \frac{x_t}{x_0 } \right) -
\log \left( \frac{(1-x_t)}{ (1-x_0)} \right) \right\}
\eeq

which we can extend
which for $\mu_X=1$ and $\sigma_X=\frac{\sigma^2}{\nu_0^2}$
\beq
\log\left(\frac{\nu_t}{\nu_0}\right)&\sim \Nc(- 
\frac{\sigma^2}{2\nu_0^2},\frac{\sigma^2}{\nu_0^2}) \\
\log\left(\frac{1-\nu_t}{1-\nu_0}\right)&\sim
\Nc(\frac{\sigma^2}{2(1-\nu_0)^2},\frac{\sigma^2}{(1-\nu_0)^2})\\
\log\left(\frac{\nu_t}{\nu_0}\right)-\log\left(\frac{1-\nu_t}{1-\nu_0}\right)&\sim
\Nc(\frac{\sigma^2}{2(1-\nu_0)^2}-
\frac{\sigma^2}{2\nu_0^2},\frac{\sigma^2}{(1-\nu_0)^2}+\frac{\sigma^2}{\nu_0^2})\\
s &\sim \Nc \left(0, 
\frac{4}{t^2} 
\left(\frac{\sigma^2}{\nu_0^2}+\frac{\sigma^2}{(1-\nu_0)^2}\right)\right) \\
s &\sim \Nc \left(0, 
\frac{4}{R \tau^2} 
\left(\frac{\sigma^2}{\nu_0^2}+\frac{\sigma^2}{(1-\nu_0)^2}\right)\right) 
\eeq
basically we for $\nu_0<0.5$ we have
\beq
s &\sim \Nc \left(0, 
\frac{4}{R \tau^2} 
\left(\frac{2\sigma^2}{(1-\nu_0)^2}\right)\right) \\
s &\sim \Nc \left(0, 
\frac{4}{R \tau^2} 
\left(\frac{\sigma^2}{\nu_0(1-\nu_0)}\right)\right) =\Nc(0,\frac{2t}{N_e})
\eeq

cite



Also let $\Phi_D$ and $\Phi_A$  be CDF of Null and Alternative distribution, 
respectively, then we have
\beq
POWER= \pr_A(s>\Phi_D^{-1}(0.9))=((1-\Phi_A(\Phi_D^{-1}(0.9)))-0.1)\times100
\eeq
\newpage
\paragraph{Linked-loci} suppose infinite pop size and no recomb and hard 
sweep. consider $\nu$ and $x$ be AF of carrier and linked site with perfect 
linkage. Then
\beqq
\delta x_t &= \begin{cases}
	\delta \nu   \ \ \ \   x_0=\nu_0 \\
	\alpha\delta \nu   \ \ \   \alpha \le0, \ x_0>\nu_0 
\end{cases} \\
s_x &= \eta(x_t) -\eta( x_0)\\
s_\nu &= \eta(\nu_t) - \eta(\nu_0)\\
s_x -s_\nu&= \eta(x_t) -\eta( x_0)-\eta(\nu_t) + \eta(\nu_0)\\
s_x -s_\nu&= s_{x_t,\nu_t} - s_{x_0,\nu_0}\\
s_x &= s_\nu+  s_{x_t,\nu_t} - s_{x_0,\nu_0}\\
\eeqq


\subsection{Likelihood Ratio Test}
Using the MLE (and other) estimates of $s$ it is desirable to perform a 
secondary task such as \emph{testing for selection} or \emph{locating the 
	site under selection}. Likelihood ratio test (LRT) statistics for time 
	series 
\cite{feder2014Identifying} have shown to be predictors for differentiating
neutral and  natural selection evolution cases. For a single locus model, the
likelihood ratio 
test statistics $\Lambda(s*)$ is defined
\beq \label{eq:lrt}
\Lambda(s^*) = \log \left(\frac{\Lc(\bfX|s=s*)}{\Lc(\bfX|s=0)}\right)
\eeq
where $s^*$ is the optimal solution for the maximum-likelihood procedure. 

%\input{intro}
%\input{method}
%\input{experiment}
%\input{conclusion}
\cite{achaz2009frequency,akey2009constructing,andersson2011notch,barrett2008natural,Barrick2013Genome,barrick2009genome,barton1991natural,bergland2014genomic,bollback2007clonal,bollback2008estimation,boyko2008assessing,braverman1995hitchhiking,daniels2013genetic,denef2012situ,desai2008polymorphism,Elena2003Evolution,enard2015viruses,evans2007non,Ewens2012Mathematical,fay2000hitchhiking,feder2014Identifying,feder2012ldx,Feder2016More,hegreness2006equivalence,Hudson2002Generating,illingworth2011distinguishing,illingworth2012quantifying,izutsu2015dynamics,jha2015whole,kawecki2012experimental,kimura1955solution,kreitman2000methods,lang2011genetic,lang2013pervasive,lenski1991long,malaspinas2012estimating,maldarelli2013hiv,mathieson2013estimating,messer2013population,nielsen2005genomic,orozco2012adaptation,Peng2005simuPOP,pollak1983new,przeworski2002signature,ptak2002evidence,pybus2015hierarchical,ramos2002statistical,reid2011new,Ronen2015Haplotype,ronen2015predicting,ronen2013learning,rose1994evolutionary,sabeti2006positive,sawyer1992population,schlotterer2015combining,schlotterer2014sequencing,schrider2015soft,Sheehan028175,slatkin2008linkage,song2012simple,Steinrücken2014a,Stephan2006The,tajima1989statistical,Terhorst2015Multi,tobler2014massive,wang2001pseudo,waples1989generalized,williamson1999using,winters2012development,zhou2011experimental}
\bibliographystyle{plain}
\bibliography{library}
\end{document}
