\section{Results}
\paragraph{Modeling neutral trajectories in finite populations.} 
We tested the closeness of fit for the Markov Likelihood as a model
for neutral trajectories, compared to Brownian motion. We performed
$150$K simulations for different values of $\nu_0$
($\nu_0\in\{0.005,0.1\}$) and time $\tau$ generations $\tau\in
\{1,10,100\}$.  Fig.~\ref{fig:markov} shows that Brownian motion is
inadequate when $\nu_0$ is far from $0.5$, and when sampling is done
after many generations $\tau>1$. (sampling times are sparse). In most
experimental evolution scenarios, a site is unlikely to have frequency
close to $0.5$, and the starting frequencies are usually much
smaller. Moreover, sampling times are sparse. In typical
\emph{Drosophila} experiments for example, $<10$ time points are
samples in a span of $100$ generations from the onset of
selection~\cite{orozco2012adaptation, zhou2011experimental}.

In contrast, Fig.~\ref{fig:markov}A-F also shows that Markov
Likelihood predictions (Eq.~\ref{eq:mkvs}) are highly consistent with
empirical data for a wide range of simulation parameters. We also
tested the model under selection by conducting $100$K simulations with
selection strength $s=0.1$ on a site with initial frequency
$\nu_0=0.005$ and sampling after $\tau$ $(\tau\in\{1,10,100\})$
generations. The empirical and theoretical distributions tracked
closely (Fig.~\ref{fig:markov}G-I).

\paragraph{Detection Power.} 
To evaluate performance of each method, define power as the fraction
of true positives identified with false-positive rate $\le 0.05$
(Suppl.~Fig.~\ref{fig:powerROC}). Before comparing against other
methods, we first evaluated the use of \comale-H ($\Hc$) and \comale-M
($\Mc$) with different percentile-cutoffs $\pi$ (Eq.~\ref{eq:pihmm})
under different sequence coverage settings. We chose the sequence
depth of each marker by identically and independently sampling from a
Poisson distribution with parameter $\lambda\in \{30,100,\infty\}$. We
computed power for ${\cal H}_\pi$ and ${\cal M}_{\pi}$, with $\pi \in
\{0,0.99,100\}$. See Fig.~\ref{fig:powerCLR}. The performance of
\comale-H is robust to change in coverage, while the power of
\comale-M decays for low coverage values. Also, the composite
likelihood for all variants ($\pi=0$) shows the highest power
(Fig.~\ref{fig:powerCLR}, and Table~\ref{tab:power}).  Therefore, we
used ${\cal H} = {\cal H}_0$ as the default statistic for all
subsequent calculations.

We compared the power of $\Hc$, $\Mc$ Gaussian process
(GP)~\cite{Terhorst2015Multi}, FIT~\cite{feder2014Identifying}, and
CMH~\cite{agresti2011categorical} statistics.  All methods other than
\comale\, and CMH convert read counts to allele frequencies prior to
computing the test statistic. For each experiment, (specified with
values for selection coefficient $s$, starting allele frequency
$\nu_0$, coverage $\lambda$, sampling time schedule ${\cal T}$, and
number of replicates $R$), we conducted $1000$ simulations. Half of
these modeled neutral evolution and the rest were under
selection. \comale\ shows the highest power across these range of
parameters (Fig.~\ref{fig:power}). However, for perfect coverage
($\lambda=\infty$), and high selection coefficient $s=0.1$, GP has
comparable power.




\paragraph{Running Time.}
As \comale\ does not compute full likelihoods, or explicitly model
linkage between sites, the complexity of computing likelihoods is
$\Oc(TR)$, and can be efficiently vectorized for multiple replicates
and loci. Therefore, it is expected to be faster than full-likelihood
approaches like Gaussian Process (GP)~\cite{Terhorst2015Multi}. We
conducted $1000$ simulations and measured running times for \comale\
and GP, which is the only method that estimates the parameters of
selection. \comale\ is $\sim 10^3\times$ faster than single locus GP
(Fig.~\ref{fig:runTime}), while maintaining high power. These times
have a practical consequence. To run GP in the single locus mode on
the entire \emph{Drosophila} genome of a small sample (1.5M variant
sites), would take 1444 CPU-hours ($\approx$ 1 CPU-month). In
contrast, even the unoptimized version of \comale\ ran in 22 CPU-hours
in addition to $15$ CPU-mins. for pre-computation of transition
matrices.  Moreover, the core computations of \comale\ are matrix
products, and they can be efficiently vectorized. We vectorized
operations using \texttt{numba} package, which reduced the run time to
$20+$ CPU-mins., not counting pre-computation.


\paragraph{SFS for Detection in Natural Samples.} We did not show the
SFS based statistics in Fig.~\ref{fig:power} as they did not perform
better than random. In many experimental evolution settings, we sample
a restricted set of $F$ founder lines, where $F<<N_e$
(Fig.~\ref{fig:ee}B). This creates a severe bottleneck, confounding
SFS. Suppl.~Fig.~\ref{fig:bottleneck} demonstrates the effect of
experimental evolution on different SFS statistics under neutral
evolution for 1000 simulations. A second problem with using SFS for
experimental evolution is that the sampling starts right after the
onset of experimentally induced selection, and the favored allele may
not reach high enough frequency to modify the site frequency spectrum
(Suppl. Fig.~\ref{fig:sweep}).

However, in experiments involving naturally occurring populations,
even if the span of the time-series is small, the onset of selection
might occur many generations prior to sampling. To test performance of
SFS-based statistics in natural evolution, we conducted 200 (100
neutral and 100 sweep) forward simulations for different values of
$s,\lambda$ using $N_e=10K$ and accumulating new mutations. The start
of sampling was done at a randomly picked time subsequent to the onset
of selection in two distinct scenarios. Let $t_{\nu=x}(s,N_e)$ denote
the expected time (in generations) required to reach carrier frequency
$x$ in a hard sweep and $U[a,b]$ denote discrete uniform distribution
in the interval $[a,b]$. First we considered the case when start of
sampling is chosen throughout the whole sweep. i.e., $\tau_1 \sim
U\left[1,t_{\nu=1}(s,N_e)\right]$ (Fig.~\ref{fig:powerSFS}A). Next, we
considered sampling start time chosen nearer to fixation of the
favored allele, i.e., $\tau_1 \sim
U\left[t_{\nu=0.9}(s,N_e),t_{\nu=1}(s,N_e)\right]$
(Fig.~\ref{fig:powerSFS}B). In both scenarios, sampling was done over
$5$ time points within $50$ generations of $\tau_1$. We compared
$\Hc$, GP, FIT with both static and dynamic SFS based statistics of
SFSelect and Tajima's D. Fig.~\ref{fig:powerSFS}A shows that SFS based
statistics are outperformed by single locus and CLR methods. However,
when sampling is performed close to fixation, i.e., when the favored
allele has frequency of 0.9 or higher, SFS based statistics perform
significantly better than GP, FIT and $\Hc$
(Fig.~\ref{fig:powerSFS}b). Moreover, dynamic SFS statistics
outperform static SFS statistics, demonstrating that in these regimes,
SFS based statistics could be used to detect selection.


\paragraph{Site-localization.}
To identify the favored site, we simply ranked each site in the
detected window according to the likelihood ratio scores
(Eqns.~\ref{XX}). For each setting of $\nu_0$ and $s$, we conducted
500 simulations and computed the rank of the favored mutation in each
simulation. We plotted the cumulative distribution of the rank in
Fig.~\ref{fig:rank}. In all configurations, \comale\ ranked favored
allele higher than GP, In particular, for strong ($s=0.1$), the
favored allele was among the top \VB{XXX} ranked variants in
\VB{XXX99}\% of the soft sweep simulations and in \VB{95\%} of
hard-sweep simulations.

When we look at the top-ranked variant, we see a trade-off between
detection and site-localization. In hard-sweep with strong selection, the high
linkage allows for strong detection (Fig.~\ref{XXX}), but weaker
site-localization. The situation is reversed for soft sweeps due to
standing variation (Fig,~\ref{XXX}).
 
\paragraph{Strength of Selection.}
As the \comale\ likelihood calculation is model based, we can also
compute the model parameters $\hat{s}$ that maximized
likelihood~\eqref{eq:hmmlik}. We computed bias, $s-\hat{s}$ for each
experiment of \comale, and compared it against GP. The distribution of
the error (bias) for 100$\times$ coverage is presented in
Fig.~\ref{fig:bias100} for different configurations.
Suppl.~Fig.~\ref{fig:bias30},~\ref{fig:bias300},~\ref{fig:biasinf}
provide the distribution of estimation errors for 30$\times$,
300$\times$ and infinite coverage, respectively.  In general, both GP
and \comale\ have biased results for weak selection, where genetic
drift dominates. However, for stronger sweeps, ($s=0.1$), \comale\
provides estimates with smaller bias and variance. Standard deviation
of the bias of $\Hc$ is 0.028 and 0.011 in hard and soft sweep
scenarios, while it is 0.036 and 0.013 for GP, respectively.



\ignore{
\paragraph{SFS based statistics in time series.} \VB{We need to delete, and 
move some of this earlier.}
We also considered SFS based tests including Tajima's D, Fay \& Wu's H and 
SFSelect for detecting selection. 
As these tests work only on static data, we extend them for time series data 
for both null and alternative hypotheses.
More precisely, we explicitly defined functionals $D_t$ and $H_t$ as function 
of initial carrier frequency $\nu_0$ and strength of selection $s$ (see section 
\ref{sec:sfs-ts} for details).
 To show that the proposed models of $D_t$ and $H_t$ are valid models, we 
 simulated 1000 populations and computed SFS based statistics every 10 
 generation and compared them with the proposed model.
 As shown in the Fig.~\ref{fig:sfs} proposed models are more consistent with 
 data.

 \paragraph{SFS based tests will fail when carrier frequency $\nu_t$
   is low.} \VB{We need to delete, and move some of this earlier.}
 Importantly, Fig.~\ref{fig:sfs} shows the increase in variance of
 the trajectories associated with SFS, compared to carrier
 frequency. Moreover, it is difficult to distinguish SFS trajectories
 of selection and neutral populations, when carrier frequency is
 small, e.g. first 50 generations of Fig.~\ref{fig:sfs}.  This,
 observation can be verified by examining the terms in the functional
 form of $D_t$.  As shown in the Fig.~\ref{fig:tdterms} right, in
 early generations of hard sweep where carrier frequency is low, $D_t$
 is either positive or close zero. In other words, the reduction in
 diversity become significant when carrier frequency is high enough.
 It can be shown that this argument holds for $H_t$ in early
 generations of hard sweep, where carrier frequency is not high.

}





\subsection{\data}\label{sec:dmel}
We applied \comale\ to a controlled experimental evolution
experiment~\cite{orozco2012adaptation}, where $3$ replicate samples
were chosen from a population of \emph{Drosophila melanogaster} for 59
generations under alternating 12-hour cycles of hot (28$^{\circ}$C)
and cold (18$^{\circ}$C) temperatures and sequenced. The sampling read
depths are highly heterogeneous
(Suppl. Figs.~~\ref{fig:depth},~\ref{fig:depthHetero}). Filtering low
coverage sites from data can dramatically reduce data available for
analysis. For example, by setting minimum read depth at a site to be
$30$, allowing the site to be retained only if the depth for all time
points, and all replicates exceeded $30$, the number of sites analyzed
would drop from from 1,544,374 to 10,387. The \comale\ statistic
automatically adjusts for all coverage values. We computed the
\comale\ ${\cal H}$ statistic for sliding window of 50Kbp with steps
of 10Kbp over the whole genome. We filtered out genomic regions
containing fewer than $500$ SNPs, as those regions were close to the
centromere or telomere. Fig.~\ref{fig:manhattancutoffed} shows a
Manhattan plot of scores. Our results are consistent with previous
studies~\cite{orozco2012adaptation} in showing an over-representation
of significant variants on Chromosome 3R.

The 1\%-ile cut-off (dashed line in Fig.~\ref{fig:manhattancutoffed})
reveals $21$ distinct intervals (Table~\ref{tab:intervals}) spanning
$243$ of the $13,965$ genes that encoded variants. We found $36$ GO
terms associated with ``Biological Process'' to be over-represented in
the $243$ genes using the Fisher exact test for
significance. Table~\ref{tab:Fisher} describes all $13$ terms that
were significant and contained at least $3$ genes. The effect of high
temperature on metabolism is profound, and it is not surprising that
the enriched GO terms include many generic stress and stress response
genes. They also include $5$ of $7$ genes with aminoacylase activity,
and $4$ of $17$ genes involved in peroxidase activity. Stress induced
formation of reactive oxygen species (ROS) can have deleterious
effects, and peroxidase activity has been seen in other organisms. In
many plants, including strawberry, high temperature adaptation led to
an increase in expression of ROS scavenging genes, including
peroxidase, and a reduction in total protein content due to protein
denaturation, and inhibited synthesis~\cite{gulen2004effect}, and
these effects are also observed in potato. The genes include Pxd
(CG3477), the peroxidase component of the
chorion~\cite{konstandi2005enzymatic}. Chorion peroxidases are
conjectured to play a role in ROS defense during egg formation in
mosquitoes~\cite{li2006major}.


One of the difficulties in genome-wide association studies is that the
number of polymorphisms are different between genes, i.e., longer
genes contain more SNPs and larger number of polymorphisms increase
the chance of showing association for larger genes.  Although the
\comale\ statistic for genes does not favor longer genes, we performed
SNP-based GO enrichment and compared enrichment versus the window
based approach. We tested associations of top 1\%-ile of the SNPs
within the $21$ candidate intervals (Table~\ref{tab:intervals}) using
\texttt{Gowinda}~\cite{kofler2012gowinda} and found 325 GO terms (see
supplemental table S4) with FDR $\le 0.001$. $9$ of the $13$
previously enriched terms (Table~\ref{tab:Fisher}), overlapped with
the $325$ \texttt{Gowinda} terms (Fisher exact $p$-val: $10^{-11}$)
suggesting consistency of SNP based and window-based analysis.

\ignore{
\paragraph{Dominance.}
The value of the overdominance parameter can provide an insight into the kind 
of adaptation \emph{using population frequency data}. In fact, for $s>0$ we 
	have~\cite{gillespie2010population}  
	\AI{results to be added }
\begin{center}
	\begin{tabular}{l|c}
		condition & comment\\
		\hline
		$h<0$ &  underdominance\\
		$h=0$ & recessive adaptive allele\\
		$h=0.5$ & directional selection\\
		$h=1$&	dominant adaptive allele	\\
		$h>1$ &overdominance
	\end{tabular}
\end{center}}
