%\documentclass[handout]{beamer} % \pause disabled
\documentclass[t]{beamer} % \pause enabled
\input{macros-arya}
\usepackage{datetime}
\def\vecbold#1{{\boldsymbol#1}}

\newdateformat{monthyeardate}{%
  \monthname[\THEMONTH], \THEYEAR}

%\usepackage[linesnumbered,ruled]{algorithm2e}
\usetheme{Madrid}
\setbeamertemplate{footline}
{
  \leavevmode%
  \hbox{%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{author in head/foot}%
    \usebeamerfont{author in head/foot}\insertshortauthor
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{title in head/foot}%
    \usebeamerfont{title in head/foot}\insertshorttitle
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,right]{date in head/foot}%
    \usebeamerfont{date in head/foot}\insertshortdate{}\hspace*{2em}
    \insertframenumber{} / \inserttotalframenumber\hspace*{2ex} 
  \end{beamercolorbox}}%
  \vskip0pt%
}
\def\dHAF{\text{-HAF}}
\makeatother
\title{Genomic Time-Series Data Analysis}
\subtitle{}
\author[Arya Iranmehr]
{%
  \texorpdfstring{
      \centering
      Arya Iranmehr\\
      \href{mailto:airanmehr@ucsd.edu}{airanmehr@ucsd.edu}
  }
  {Arya Iranmehr}
}
%\email{}
\institute{
Bafna Lab\\
University of California, San Diego}
\date{
\monthyeardate\today}
\subject{Optimization}
\usefonttheme{serif} % default family is serif
%\AtBeginSection[]
%  {
%     \begin{frame}<beamer>
%     \frametitle{Agenda}
%     \tableofcontents[currentsection,currentsubsection]
%     \end{frame}
%  }
%\AtBeginSubsection[]
%{
%  \begin{frame}<beamer>{Outline}
%    \tableofcontents[currentsection,currentsubsection]
%  \end{frame}
%}

\begin{document}
\begin{frame}
  \titlepage
\end{frame}




\begin{frame}
Experimental Evolution
\begin{figure}
\includegraphics[trim={0 2.1in 0 0},clip, scale=0.6]{eg.jpg} 
\end{figure}
\end{frame}
\begin{frame}{Simulations}
\begin{enumerate}[(i)]
\item For each simulation, population of $F=200$ founder lines is created in  \texttt{msms} program with parameters 
\begin{itemize}
\item Length of genome $L=10^6$Kbp.
\item Population size $N_e=10^6$.
\item Recombination rate $r=2\times10^{-8}$.
\item Mutation rate $\mu=10^{-9}$ which give $\theta = 200$ and $m\approx 1200$ .
\end{itemize}

\item Using $F=200$ founder lines a population of $N=1000$ diploid homozygote individuals are created.
\item Using \texttt{SimuPop}, the initial population is forward simulated for 10 replicates and AF is recorded every 10 generation for a complete hard sweep. 
\item I.e., a site with minimum allele frequency of $1/F$ is always under selection.
\item for each $s=\{0.1, 0.05, 0.02, 0.01, 0\}$ 100 simulations are done.
\end{enumerate}
\end{frame}

\begin{frame}{Biological Goals}
\begin{itemize}
\item Given the genome time series data (allele frequencies) we are interested in
\begin{enumerate}[I]
\item Detecting selection 
\item Locating selection
\begin{enumerate}[(i)]
 \item Identify gene under selection
 \item Identify the mutation 
\end{enumerate}
\item Estimating model parameters
\end{enumerate}
\item  problems I and II are easier, and can be addressed using model-free methods such as Tajima's D.
\end{itemize}
\end{frame}

\begin{frame}{History Detecting and locating Selection}
\begin{itemize}
\item Traditionally, detecting/locating selection is done by analysing Allele Spectrum Frequency (AFS) from a window (50Kbp here).
\item different linear combinations of AFS vector gives different tests of selection.
\item to put it in context, Tajima's D, FayWu's H and SFSelect are multi-locus, model-free methods for detecting and locating selection which use a \emphr{ single snapshot} of data.
\item However, we extend these methods to times-series framework, just by summing all test statistics through time.
\end{itemize}
\end{frame}


\begin{frame}{Questions}
\begin{enumerate}[(i)]
\item Are multi-locus methods better than single locus ones?
\item Are time-series methods outperform single-snapshot methods?
\item Are model-based approaches (likelihood based methods such as GP) better than classical \emphr{model-free} statistical-tests for detecting selection?
\item When detecting and locating selection is trival and when is not?
\item How does the initial frequency of the carrier affect the performance?
\end{enumerate}


\end{frame}

\begin{frame}{Hard Sweep}
\begin{itemize}
\item With this setting around a quarter of sites are at MAF at time 0.
\item We will consider hard-sweep, i.e., site under selection frequency=$1/F=1/Ne=0.005$=MAF. 
\item if we pick a site randomly with high prob, it'll be low freq.
\end{itemize}
\hspace{-0in}\includegraphics[scale=0.17]{AFS}
\end{frame}

\begin{frame}{Single Locus Logistic Model: Carrier Frequency}
Dynamics of sweep extremely depend on $s$ as well as initial carrier frequency.Sampling times are very important prediction performance!

\begin{figure}
\centering
\hspace{-0in}\includegraphics[trim={2in 0.5in 1.5in 0in},clip,page=2,width=0.5\textwidth]{sigmoidSoft}
\hspace{-0in}\includegraphics[trim={2in 0.5in 1.9in 0in},clip,page=2,width=0.5\textwidth]{sigmoidHard}
\end{figure}
{\tiny
\begin{enumerate}
\item Left) Terhorst et al. considered soft sweep. The time-window which  they sampled is shaded. 
\item Right) With the same sampling scheme, it is difficult to make predictions base on observations under 50 generations.
\end{enumerate}
}
\end{frame}

\begin{frame}{A Naive Single Locus Model}
\begin{itemize}
\item Roughly, just divide your last and first observation, it gives you a number, which can be regarded as a \emphr{predictor} for detecting selection. I.e., larger values of the predictor imply evidence of selection.
\item Using start and end AF at a site, we can solve the logistic equation to get an estimate of $s$ for that site.
\end{itemize}

\end{frame}

\begin{frame}{Multi-Locus Methods}
\begin{itemize}
\item In hard sweep, site under selection has perfect LD with all the other sites at time 0.\item With hard-sweep, it is difficult to aim locating mutation in less than 2K generations.
\item 50Kb window seems to be a reasonable resolution.
\end{itemize}
\begin{figure}
\centering
\hspace{-0in}\includegraphics[scale=0.17]{spaceTimeLD}
\caption{Decay of LD in space through time under neutral evolution, when selected site is at position 500K. 50Kb window centered at the site is shaded.}

\end{figure}
\end{frame}

\begin{frame}{Multi Locus Model: Avg Haplotype Allele Frequency}
\begin{itemize}
\item Analogous to, TajimaD,H and SFSelect, AHAF is a statistic for a (50Kb) window can be computed instantly, (Euclidean norm of AF vector).
\item However, Ronen et. al. derived dynamics of AHAF in term of $\nu_t$ for hard sweep,  and therefore $s$.
\end{itemize}
\centering
\hspace{-0.1in}\includegraphics[scale=0.2]{haf}

\end{frame}



\begin{figure}
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,width=\textwidth]{populationDynamics.pdf}
  \caption{A simulations for s=0.1. Mean of 10 replicates is shown as observation.
  Complete sweep is divided into three epochs based on carrier frequency.}
\end{figure}

\begin{frame}{Global Trends of sweep}
\begin{itemize}
\item In the following slides, global trend of Carrier freq, and all the window-based statistics are presented for each $s$.
\item There are two models: logistic (single locus) and AHAF (multi-locus).
\item 1000 simulations done and 95\% confidence interval is shown for each curve. (sweep/neutral).
\end{itemize}

\end{frame}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=1,width=\textwidth]{Globaldynamics.pdf}
\end{figure}
{\tiny 1) Before generation 70, detection is difficult.\newline 2)Window based methods discriminate two cases with high confidence in phase 2 and 3.\newline 3) Discrimination in AHAF trend is harder than other methods.}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=2,width=\textwidth]{Globaldynamics.pdf}
\end{figure}
{\tiny 1) Before generation 130, detection is difficult.\newline 2)Window based methods discriminate two cases with high confidence in phase 2 and 3.\newline 3) Discrimination in AHAF trend is harder than other methods.}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=3,width=\textwidth]{Globaldynamics.pdf}
\end{figure}
{\tiny 1) Before generation 300, detection is difficult.\newline 2)Window based methods discriminate two cases with high confidence in phase 2 and 3.\newline 3) Discrimination in AHAF trend is harder than other methods.}


\begin{figure}[H]
  \centering
  		    \includegraphics[trim={2in 0.5in 2in 0in},clip,page=4,width=\textwidth]{Globaldynamics.pdf}
\end{figure}
{\tiny 1) Before generation 600, detection is difficult.\newline 2)Window based methods discriminate two cases with high confidence in phase 2 and 3.\newline 3) Discrimination in AHAF trend is harder than other methods.}

{\tiny Start of phases chosen when $\nu_t=[0.1,0.5,0.9]$ and AF sampled every 10 generations for 50 generations.}
\begin{figure}
\hspace{-0in}\includegraphics[scale=0.16]{samplingTimes}
\end{figure}
\vspace{-0.5in}



\begin{frame}{Task 1: Detecting Selection Experiments}
\begin{enumerate}[I]
\item To detect selection we need a predictor
\begin{itemize}
\item GP and HAF use Empirical Likelihood Ratio (ELR)
\beqq
ELR=\frac{Likelihood(Data|s=s^*)}{Likelihood(Data|s=0)}
\eeqq
\item Naive, use the ration of the first and last AF at a site
\item Model-free methods use their test-statistics values.
\end{itemize}
\item Single locus method (Naive and GP), use max of predictor over all the sites of the window.
\item performance measure is the percentage of Area under ROC curve, for regions with false-positive-rate is less than 0.1.
\end{enumerate}
\end{frame}



\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=1,,scale=0.2]{aucfull}
\end{figure}
{\tiny \begin{enumerate}
\item Problem seems to be trivial if we have access to 5 time samples in a wide times series window (200,500,1000,2000 generations).
\item when $s$ is small, SFSelect outperforms the others.
\end{enumerate}}




\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=1,,scale=0.2]{aucwin50.pdf}
\end{figure}
{\tiny \begin{enumerate}
\item Model-free methods perform very well in all epochs for sampling window = 50 , s=0.1.
\item Multi-locus methods are better than single locus ones.
\item Model-free methods are better than model-based ones.
\end{enumerate}}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=2,scale=0.2]{aucwin50.pdf}
\end{figure}
{\tiny \begin{enumerate}
\item Model-free methods perform very well in last two epochs for sampling window = 50 , s=0.05.
\item Multi-locus methods are better than single locus ones.
\item Model-free methods are better than model-based ones.
\end{enumerate}}


\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=3,,scale=0.2]{aucwin50.pdf}
\end{figure}
{\tiny \begin{enumerate}
\item Model-free methods perform very well in last two epochs for sampling window = 50 , s=0.02.
\item Multi-locus methods are better than single locus ones.
\item Model-free methods are better than model-based ones.
\end{enumerate}}


\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=4,scale=0.2]{aucwin50.pdf}
\end{figure}
{\tiny \begin{enumerate}
\item Model-free methods perform very well in last two epochs for sampling window = 50 , s=0.01.
\item Multi-locus methods are better than single locus ones.
\item Model-free methods are better than model-based ones.
\end{enumerate}}


\begin{frame}{Task 2: Locating Selection}
Locating selection upto a 50Kb window amount to perform an sliding window detection test and select window with maximum predictor.
\end{frame}

\begin{frame}{Task 3: Estimating Strength Selection}
\begin{itemize}
\item Only model based method can estimate $s$.
\item Performance measure is absolute value of bias $|s-\hat{s}|$.
\end{itemize}
\end{frame}
\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=1,,scale=0.2]{bias.pdf}
\end{figure}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=2,,scale=0.2]{bias.pdf}
\end{figure}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=3,,scale=0.2]{bias.pdf}
\end{figure}

\begin{figure}[H]
  \centering
  		    \includegraphics[trim={0in 0.5in 0in 0in},clip,page=4,,scale=0.2]{bias.pdf}
\end{figure}


\begin{frame}{summary}
\begin{itemize}
\item Classical model-free (multi-locus) methods work better for detecting and locating selection. 
\item HAF method is simple and its estimate of strength of selection is comparable to that of GP. However for detecting selection, HAF have too many FPs.
\item Verdict: Use classical model-free methods for detecting selection, if it exist, fit a HAF model to estimate $s$.
\item Sampling times are critical. In fact, given 5 samples along full sweep, the problem of detection is trivial.
\item To be verified: Due to high LD in experimental evolution in less than 2K generations, we can only determine window under selection, in HARD SWEEP.
\end{itemize}

\end{frame}


%\end{frame}


%
%\begin{frame}{Computational Performance}
%\begin{figure}[H]
%  \centering
%    \includegraphics[width=0.6\textwidth]{times}
%\end{figure}
%\end{frame}





\begin{frame}{\ }
\vspace{1	in}
\begin{center}
	\huge{Thanks!}\\
\end{center}

\end{frame}

\begin{frame}{Single Locus Model: Carrier Frequency}
\begin{itemize}
\item By differentiating update equations ($x_{t+1}=x_t+\frac{sx_t(1-x_t)}{2+2sx_t}$ w.r.t. $t$ and solving differential equation, we have
\beqq
\nu_t=\sigma\left(st/2+\eta(\nu_0)\right)
\eeqq
where $\sigma(.)$ is logistic function and $\eta=\sigma^{-1}$ is the logit function and $\nu_t$ is the frequency of the carrier at time $t$.
\end{itemize}
\hspace{-0.6in}\includegraphics[scale=0.2]{sigmoid}
\end{frame}
\begin{frame}{Multi Locus Model: Average Haplotype Allele Frequency}

\begin{equation*}
\resizebox{.99\hsize}{!}{
$\mathbb{E}[1\dHAF(t)]= \|\mathbf{x}_t\|^2\approx \theta \nu_t \left(\frac{\nu_t+1}{2} - \frac{1}{(1-\nu_t)n+1}\right) +
 \theta (1-\nu_t)\left(\frac{n+1}{2n}-\frac{1}{(1-\nu_t)n+1}\right)$
}
\end{equation*}
where $ \mathbf{x}_t$ is vector of AF at time $t$ and $\nu_t=\sigma\left(st/2+\eta(\nu_0)\right)$.

\hspace{-0.1in}\includegraphics[scale=0.2]{haf}
\end{frame}
\begin{frame}{Critics to GP}
\begin{itemize}
\item Its too complex, likely to overfit with small number of iid replicates.
\item Although the likelihood model is based on different parameters, in practice, it can learn only one parameter at a time.
\item not tractable, its time complexity is quartic!
\item worse, each iteration requires \texttt{maxGeneration} recursion which makes it very hard to analyse late epochs of sweeps.
\item In addition to PoolSeq data, it requires initial population haplotypes.
\item Despite its elegant theory, it has not compared with classical methods.
\item In practice, single locus scan is performed and multi-locus (with 3-7 seg. sites.) model is fitted at regions of interests.
\end{itemize}
\end{frame}
\end{document}


