\begin{abstract}
  Experimental evolution (EE) studies are powerful tools for observing
  molecular evolution ``in-action'' from populations sampled in
  controlled and natural environments. The advent of next generation
  sequencing technologies has made whole-genome and whole-population
  sampling possible, even for eukaryotic organisms with large genomes,
  and allowed us to locate the genes and variants responsible for
  genetic adaptation. While majority of the existing literature on dynamic data 
  analysis, is mainly devoted to the settings with large population sizes, allele 
  frequency as input data and wide time spans, these assumptions does not 
  hold in many EE studies.
	
  In this article, we propose a method -Composition of Likelihoods for
  Evolve-And-Resequence experiments (\comale)- to identify selection in 
  short-term (as 
  well as long-term), EE of \emph{small} sexual populations. Also, \comale\  
  takes whole-genome sequence of pool of individuals (pool-seq) as input, 
  and properly addresses heterogeneous ascertainment bias, due to uneven 
  coverages.
  \comale\ also provides unbiased estimates of model
  parameters, including selection strength and overdominance, and population 
  size, while being computationally efficient.
	Extensive simulations
  show that \comale\ achieves higher power in detecting and localizing
  selection over a wide range of parameters. Moreover, \comale\ statistic is 
  robust to variation of
  coverage.  
  We applied 
  \comale\ statistic to previously published datasets, including,
  \datadm and study of outcrossing Yeast populations. Result XXXX
\end{abstract}