\section{Discussion}
We developed a computational method, \comale, that can detect regions
and variants under selection E\&R experiments of sexual
populations. Using extensive simulations, we show that \comale\
outperforms existing methods in detecting selection, locating the
favored allele, and estimating selection parameters.  Also, while
being computationally efficient, \comale\ provide means for estimating
populations size and hypothesis testing.

Many factors such as small population size, finite coverage, linkage
disequilibrium, finite sampling for sequencing, duration of the
experiment and the small number of replicates can limit the power of
tools for analyzing E\&R.  Here, by an discrete modeling, \comale\
estimates population size, and provides unbiased estimates of
$s,h$. It adjusts for heterogeneous coverage of pool-seq data, and
exploits presence of linkage within a region to compute composite
likelihood ratio statistic.


It should be noted that, even though we described \comale\ for small
fixed-size populations, the statistic can be adjusted for other
scenarios, including changing population sizes when the demography is
known. For large populations, transitions can be computed for a fixed
size of frequencies instead of raw counts.

\ignore{
In our simulations, we found that the power of detection can be
severely affected by the sampling schedule as well as initial
frequency of the favored allele.  In general, while E\&R studies are
powerful, they also pose some challenges that
 are not adequately considered by other tools. 
One serious constraint is the sampling time
span, the gap between the first and last sampled generations, which
depends upon the generation time of the organism. It can be very small
relative to the time of fixation of the favored allele. In \dmel for
example, $30$-$50$ generations are typical~\cite{kofler2013guide},
although there are some notable
exceptions~\cite{zhou2011experimental}.  Therefore, unless the
selection coefficient is very strong, the time series data will only
capture a `partial sweep'. This limitation is more pronounced in
controlled experimental evolution, where the sampling often starts at
the onset of selection. In particular, in small populations it is difficult to 
differentiate between drift and natural selection is an incomplete sweep. }



\ignore{
In controlled experimental evolution experiments, populations are
evolved and inbred. As this scenario involves picking a small number
of founders, the effective population size significantly drops from
the large number of wild type (e.g., for \dmel, $N_e\approx10^6$) to a
small number of founder lines ($F$ $\approx 10^2$). This creates a severe 
population
bottleneck. The bottleneck confounds SFS-based statistics and makes
it difficult to fit a model or test a hypothesis
(\ref{fig:bottleneck}).  Hence, statistical testing based
on SFS statistic provides poor performance in controlled experiments
where the initial sampling time is close to the onset of selection.
However, SFS-based methods perform very well when sampling is started
long after the onset of selection (e.g., sampling from natural
populations). The larger time gap from the onset of selection provides
an opportunity for the site frequency spectrum to shift away from
neutrality.}



The comparison of hard and soft sweep scenarios showed that initial
frequency of the favored allele can have an nontrivial effect on the
statistical power for identifying selection. Interestingly, while in
stronger selections it is easier to detect regions of selection, it is
difficult to locate favored allele in those regions.


There are many directions to improve the analyses presented here.  In
particular, we plan to focus our attention on other organisms with
more complex life cycles, experiments with variable population size
and longer sampling-time-spans. As evolve and resequencing experiments
continue to grow, deeper insights into adaptation will go hand in hand
with improved computational analysis.



