\begin{figure}[H]
		\centering
(A)\\
	\includegraphics[trim=0in 0in .08in 0.02in , 
	clip,width=0.7\textwidth]{ExperimentalEvolution.pdf}\\

\begin{tabular}{l|l}
		\hline\\
		(B) &(C)\\
		\raisebox{0.5in}{
\includegraphics[trim=0in 0.in 0in 
0.0in,clip,width=0.5\textwidth]{HMMGM.pdf}}
& 	\includegraphics[trim=0in 0.in 0in 
0.0in,clip,width=0.6\textwidth]{AF.pdf}	

\end{tabular}
\hspace{-1in}
\caption{{\bf Evolve and Resequence Selection Experiments on \dmel.}
  (A) Typical configuration in which time-series data is collected for
  \dmel.  (B) Graphical model showing dependence of the random
  variables in the single-locus model used to compute \comale\
  statistics. Only shaded variables are observed.  (C) Mean and the
  95\% confidence interval of the theoretical \VB{Please change the
    insets to (i,ii,iii,iv)} (a,b) and empirical (b,d) trajectories of
  the favored allele for hard (a,b) and soft (c,d) sweep scenarios and
  $N=1000$.  The first $50$ generations are shaded in gray to
  represent the sampling span of sampling in short-term experiments,
  implying the difficulty in predicting selection at early stages of
  selective sweep.  }
\label{fig:1}
\end{figure}


%\begin{figure}[H]
%	\includegraphics[trim=0in 0.in 0in 
%	0.1in,clip,width=\textwidth]{AF.pdf}
%	\caption{{
%	\label{fig:sweep}
%\end{figure}


%\begin{figure}[H]
%	\centering
%	\includegraphics[trim=0in 0in .08in 0.02in , 
%	clip,width=\textwidth]{ExperimentalEvolution.pdf}
%	\caption{ {\bf Two settings for collecting genomic time series            
%	data.}\\
%		 Different settings in which dynamic data is
%          collected are depicted with typical parameters for \emph{D.
%            melanogaster}. In both settings, 6 samples (vertical red
%          dashed lines) are taken every 10 generation.  When sampling
%          from naturally evolving populations (A), the time of onset
%          of selection is unknown, and population size is larger. For
%          (controlled) experimental evolution (B), founder lines are first
%          sampled from a natural population to create a homogeneous
%          population. Then, multiple replicates of this population are
%          evolved and sampled over time. }
%	\label{fig:ee}
%\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{{markovDists}.pdf}
        \caption{{\bf Comparison of empirical distributions of allele
            frequencies (red) versus predictions from Brownian
            Motion (green), and Markov chain (blue).}\\
          Comparison of empirical and theoretical distributions under
          neutral evolution (panels A-F) and selection (panels G-M)
          with different starting frequencies $\nu_0\in\{0.005,0.1\}$
          and sampling times of $\Tc=\{0,\tau\}$, where $\tau \in
          \{1,10,100\}$.  For each panel, the empirical distribution
          was computed over 100,000 simulations.  Brownian motion
          (Gaussian approximation) provides poor approximations when
          initial frequency is far from 0.5 (A) or sampling is sparse
          (B,C,E,F). In addition, Brownian motion can only provide
          approximations under neutral evolution. In contrast, Markov
          chain consistently provides a good approximation in all
          cases.}
	\label{fig:markov}
\end{figure}
\ignore{
\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0,
	clip,width=\textwidth]{{CLRQ}.pdf}
	\caption{{\bf Average power of \comale\ statistics as function
            of percentile cutoffs.}\\ Detection power is averaged for
          $s\in\{0.025,0.05,0.075,0.1\}$ for \comale\ statistics, $\Hc_{\pi}$ and
          $\Hc^{+}_{\pi}$. We denote
          $\pi^{*}=\arg\max_{\pi}\{\mbox{Power}(\Hc),\mbox{Power}(\Hc^{+})\}$. Average
          power was computed using $8000$ simulations for each choice
          of $\pi$. The appropriate choice of $\pi$ can be used to
          improve performance for different coverage values.
          In all simulations, 3 replicates are evolved and sampled at generations
          $\Tc=\{0,10,20,30,40,50\}$.}
	\label{fig:clrq}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{powerCLR.pdf}
	\caption{{\bf Comparison of power of Markov chain and HMM.}\\
          Detection power for Markov chain ($\Mc$) and HMM ($\Hc$) under
          hard (A-C) and
          soft sweep (D-F) scenarios, for different coverage $\lambda$ and
          selection strength $s$.  The
          $y$-axis measures power -- sensitivity with false positive
          rate FPR $\le 0.05$ -- for $2000$ simulations of $50$Kbp
          regions. The horizontal line reflects the power of a random
          classifier.
          In all simulations, 3 replicates are evolved and sampled at generations
          $\Tc=\{0,10,20,30,40,50\}$.
           } \label{fig:powerCLR}
\end{figure}
}

\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{power.pdf}
	\caption{ {\bf Power calculations for detection of selection.}\\
          Detection power for \comale ($\Hc$), Frequency Increment
          Test (FIT), Gaussian Process (GP), and CMH under hard (A-C)
          and soft sweep (D-F) scenarios. $\lambda$, $s$ denote the
          mean coverage and selection coefficient, respectively. The
          $y$-axis measures power -- sensitivity with false positive
          rate FPR $\le 0.05$ -- for $2,000$ simulations of $50$Kbp
          regions. The horizontal line reflects the power of a random
          classifier.
          In all simulations, 3 replicates are evolved and sampled at generations 
          $\Tc=\{0,10,20,30,40,50\}$.}
           \label{fig:power}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[width=0.75\textwidth]{{runTime.pdf}}
	\caption{{\bf Running time.}\\ Box plots of running time per
          variant (CPU-secs.) of \comale ($\Hc$), CMH, FIT, and
          GP with single, 3, 5, 7, and 10 loci over 1000 simulations
          conducted on a workstation with Intel Core i7
          processor. The average running time for each method is shown
          on the x-axis.
          In all simulations, 3 replicates are evolved and sampled at generations 
          $\Tc=\{0,10,20,30,40,50\}$.}
	\label{fig:runTime}
\end{figure}

\ignore{
\begin{figure}[H]
	\centering
	\includegraphics[width=0.7\textwidth]{naturalee.pdf}
	\caption{{\bf Power of SFS based statistics.}\\ Power of
          detecting selection for Frequency Increment Test (FIT),
          Gaussian Process (GP), \comale ($\Hc$) on hard sweep
          natural experimental evolution with $N_e=10^4$ and coverage
          $\lambda=\infty$. The measurements are conducted for a range
          of selection coefficients, $s$. Each point represents the
          mean of 200 simulations. For each simulation, sampling
          starts at a randomly chosen time, and subsequently 5
          replicate samples are acquired every $10$ generations.  (A)
          Start of sampling is chosen randomly throughout the sweep
          $\tau_1 \sim U\left[1,t_{\nu=1}(s,N_e)\right]$, where
          $t_{\nu=x}(s,N_e)$ denotes is the expected time to reach
          carrier frequency $x$ in a hard sweep and $U[a,b]$ is
          discrete uniform distribution. (B) The start of sampling is
          chosen near fixation of the favored allele, i.e. $\tau_1
          \sim
          U\left[t_{\nu=0.9}(s,N_e),t_{\nu=1}(s,N_e)\right]$.}
          \label{fig:powerSFS}
\end{figure}
}

\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{{rank100.0}.pdf}
	\caption{{\bf Ranking performance for 100$\times$ coverage.}\\
          Cumulative Distribution Function (CDF) of the distribution
          of the rank of the favored allele in 1000 simulations for
          \comale\ ($H$), Gaussian Process (GP), CMH, and Frequency
          Increment Test (FIT), for different values of selection
          coefficient $s$ and initial carrier frequency. Note that the
          individual variant \comale\ score ($H$) is used to rank
          variants.  The Area Under Curve (AUC) is computed as an overall
          quantitative measure to compare the performance of methods
          for each configuration. In all simulations, 3 replicates are evolved and 
          sampled at generations 
          $\Tc=\{0,10,20,30,40,50\}$.}
	\label{fig:rank}
\end{figure}



\begin{figure}[H]
	\centering
\includegraphics[width=0.7\textwidth]{{bias.100}.pdf}
\caption{{\bf Distribution of bias for 100$\times$ coverage.}\\ The
  distribution of bias ($s-\hat{s}$) in estimating selection
  coefficient over 1000 simulations using Gaussian Process (GP) and
  \comale\ ($H$) is shown for a range of choices for the selection
  coefficient $s$ and starting carrier frequency $\nu_0$, when
  coverage $\lambda=100$ (Panels A,B). GP and \comale\ have similar
  variance in estimates of $s$ for soft sweep, while \comale\ provides
  lower variance in hard sweep. Also see \ref{tab:biasdist}. Panels C,D
  show the variance in the estimation of $h$. 
  In all simulations, 3 replicates are evolved and sampled at generations 
  $\Tc=\{0,10,20,30,40,50\}$.}
	\label{fig:bias100}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{estimateN.pdf}
	\caption{{\bf Estimating N on simulated and real data.}  Mean
          and 95\% confidence interval of likelihoods of $N$ on
          simulated data with $N=200$ (A), $N=600$(B), and $N=1000$
          individuals, over 100 simulations. Chromosome-wise (D) and
          genome-wide (E) estimation of population size for
          \datadm. Chromosome 3R fits population size of 150, while
          genome-wide population size is 250.  (F) Despite large
          census population size
          ($10^6-10^7$~\cite{burke2014standing}), Yeast dataset
          exhibits much smaller effective population size
          ($\hN=2000$).  }
	\label{fig:estimateN}
\end{figure}

\begin{figure}[H]
	\centering
	\begin{tabular}{cc}
		(A)&(B)\\
		\includegraphics[trim=0.4in 0.in 0.6in 
		0in,clip,width=0.6\textwidth]{man-dmel-region.pdf}&
		\includegraphics[trim=0in 0.in 0in 
		.0in,clip,width=0.35\textwidth]{{topVariants.dmel.dir}.pdf}
	\end{tabular}
	\caption{{\bf Scan of composite statistic on \datadm.}
          Manhattan plot of scan for $\Hc^*$ statistic over the
          genome.  The dashed line represents cutoff for genome-wide
          FDR$\le0.05$, selecting 16 regions (A).\VB{Please remove
            this panel and caption for (B) as it is noisy and doesn't
            add much} Trajectories of the top 174 variants shown in
          panel (B).  }
	\label{fig:man-dmel-region}
\end{figure}

\begin{figure}[H]
	\centering
	\begin{tabular}{cc}
						(A)&(B)\\
	\includegraphics[trim=0.4in 0.in 0.6in 
	1.17in,clip,width=0.6\textwidth]{man-dmel-snp.pdf}&	
	\includegraphics[trim=0.1in 0.in 0in 
	1.5in,clip,width=0.35\textwidth]{{topVariants.dmel}.pdf}
	\end{tabular}
	\caption{{\bf Single locus analysis of the \datadm.}\\
          \VB{Please move this plot to supplemental, as we do not
            really say much about it.} Manhattan plot of scan for
          testing overdominant selection (A).  Significant variants
          with FDR $\le 0.01$ are denoted in red, and their
          trajectories are depicted in panel (B).}
	\label{fig:man-dmel-snp}
\end{figure}



\begin{figure}[H]
	\centering
	\begin{tabular}{cc}
		\includegraphics[trim=0.4in 0.in 0.6in 
		0.0in,clip,width=0.65\textwidth]{man-yeast-snp.pdf}&	
		\raisebox{0.2in}{
		\includegraphics[trim=0in 0.in 0in 
		0.0in,clip,width=0.35\textwidth]{{topVariants.yeast}.pdf}}
	\end{tabular}
	\caption{{\bf Single locus analysis of the \datadm.}\\ Manhattan plot 
		of scan for testing directional selection (A) and overdominant 
		selection 
		(C). dashed
		The dashed line represents cutoff for  genome-wide FDR$\le0.05$.
		Trajectories of the selected variants are depicted in panels (B) and 
		(D).}
	\label{fig:man-yeast-snp}
\end{figure}





\clearpage
\newpage