\section{Results}
\paragraph{Modeling Allele Frequency Trajectories in Finite Populations.} 
We first tested the goodness of fit of the discrete-time Markov chain
versus continuous-time Brownian motion (Gaussian approximation) in modeling 
allele frequency
trajectories in finite populations, under different sampling schemes
and starting frequencies.  For this purpose, we conducted $100$K
simulations with two time samples $\Tc=\{0,\tau\}$ where $\tau\in
\{1,10,100\}$ is the parameter controlling the density of sampling
in time.  In addition, we repeated simulations for different values of
starting frequency $\nu_0\in\{0.005,0.1\}$ (i.e., hard and soft sweep)
and selection strength $s\in\{0,0.1\}$ (i.e., neutral and
selection). Then, given initial frequency $\nu_0$, we computed
the expected distribution of the frequency of the next sample $\nu_\tau$
under two models and compared them with empirical distributions
calculated from simulated data.  \ref{fig:markov}A-F shows that
Brownian motion is inadequate when $\nu_0$ is
far from $0.5$, or when sampling times are sparse ($\tau>1$). If the
favored allele arises from standing variation in a neutral population,
it is unlikely to have frequency close to $0.5$, and the starting
frequencies are usually much smaller (see
\ref{fig:sfs}). Moreover, in typical \dmel experiments for
example, sampling is sparse. Often, the experiment is designed so that
$10\le\tau\le100$~\cite{kofler2013guide, orozco2012adaptation,
  zhou2011experimental,franssen2015patterns}.

In contrast to the Brownian motion results,  Markov chain can provide 
predictions when the the allele is under selection. In addition 
\ref{fig:markov}A-M
also shows that Markov chain predictions (Eq.~\ref{eq:mkvs}) are
highly consistent with empirical data for a wide range of simulation
parameters.

\paragraph{Detection Power.} 
We compared the performance of \comale\ against other methods for
detecting selection. For each method we calculated detection power as the 
percentage of true-positives identified with false-positive rate $\le 0.05$
(\ref{fig:powerROC}). For each
configuration (specified with values for selection coefficient $s$,
starting allele frequency $\nu_0$ and coverage $\lambda$), power of each method 
is evaluated over $2000$ distinct simulations, half of which modeled neutral 
evolution
and the rest modeled positive selection.



We compared the power of \comale\ with Gaussian process
(GP)~\cite{Terhorst2015Multi}, FIT~\cite{feder2014Identifying}, and
CMH~\cite{agresti2011categorical} statistics.  As CMH only takes read
count data, here we used $\lambda=300$ to implement infinite coverage
scenario. FIT and GP convert read counts
to allele frequencies prior to computing the test statistic.  \comale\
shows the highest power in all cases and the power stays relatively
high even for low coverage (\ref{fig:power} and
\ref{tab:power}). In particular, the difference in performance
of \comale\ with other methods is pronounced when starting frequency
is low. Starting frequency is at its minimum in selection on an allele
from \emph{de novo} mutations and is likely to be low if selection is
on an allele from standing variation (\ref{fig:sfs}).  The
advantage of \comale\ stems from the fact that favored allele with low
starting frequency might be missed by low coverage sequencing. In this
case, incorporating the signal from linked sites becomes increasingly
important. We note that methods using only two time points, such as
CMH, do relatively well for high selection values and high
coverage. However, the use of time-series data can increase detection
power in low coverage experiments or when starting frequency is
low. Moreover, time-series data provide means for estimating selection
parameters $s,h$ (see below). Finally, as \comale\ is robust to change
of coverage, our results (\ref{fig:power}B,C) suggest that taking
many samples with lower coverage is preferable to sparse sampling with
higher coverage.


\ignore{ Moreover, the observation of
  \ref{fig:markov} is revisited here, approximate continues
  models (GP and FIT) perform poorly when starting frequency is low
  (\ref{fig:power}A-C).  

\paragraph{SFS for Detection in Natural Samples.} We did not show the
SFS based statistics in \ref{fig:power} as they did not perform
better than random. In majority of controlled experimental evolution studies, 
the population is restricted set of $F$ founder lines, where $F<<N_e$
(\ref{fig:ee}B) and inbred during the experiment. This creates a severe 
bottleneck, confounding
SFS. ~\ref{fig:bottleneck} demonstrates the effect of
experimental evolution on different SFS statistics under neutral
evolution for 1000 simulations. A second problem with using SFS for
experimental evolution is that the sampling starts right after the
onset of experimentally induced selection, and the favored allele may
not reach high enough frequency to modify the site frequency spectrum
(\ref{fig:sweep}).

However, in experiments involving naturally evolving populations,
even if the span of the time-series is small, the onset of selection
might occur many generations prior to sampling. To test performance of
SFS-based statistics in natural evolution, using \texttt{msms}, we conducted 
200 (100
neutral and 100 sweep) forward simulations for different values of
$s$, $N_e=10K$ and $N=200$. The start
of sampling was chosen randomly after onset of selection in two distinct 
scenarios. Let $t_{\nu=x}(s,N_e)$ denote
the expected time (in generations) required to reach carrier frequency
$x$ in a hard sweep and $U[a,b]$ denote discrete uniform distribution
in the interval $[a,b]$. First we considered the case when start of
sampling is chosen throughout the whole sweep. i.e., $\tau_0 \sim
U\left[1,t_{\nu=1}(s,N_e)\right]$ (\ref{fig:powerSFS}A). Next, we
considered sampling start time chosen nearer to fixation of the
favored allele, i.e., $\tau_0 \sim
U\left[t_{\nu=0.9}(s,N_e),t_{\nu=1}(s,N_e)\right]$
(\ref{fig:powerSFS}B). In both scenarios, sampling was done over
$6$ time points within $50$ generations of $\tau_0$ (~\ref{fig:ee}A). We 
compared
\comale, GP, FIT with both static and dynamic SFS based statistics of
SFSelect and Tajima's D. ~\ref{fig:powerSFS}A shows that SFS based
statistics are outperformed by other methods. However,
when sampling is performed close to fixation, i.e., when the favored
allele has frequency of 0.9 or higher, SFS based statistics perform
considerably better than GP, FIT and \comale\
(\ref{fig:powerSFS}B). Moreover, dynamic SFS statistics
provide higher power than static SFS statistics, demonstrating that in the use 
of time-series SFS based statistics is advantageous.}


\paragraph{Site-identification.}
In general, localizing the favored variant, using pool-seq data is a nontrivial 
task~\cite{tobler2014massive}. We used the simple
approach of ranking each site in a region detected as being under
selection. The sites were ranked according to the likelihood ratio
scores (Eqns.~\ref{eq:mcts},~\ref{eq:hmmml}). For each setting of
$\nu_0$ and $s$, we conducted $1000$ simulations and computed the rank
of the favored mutation in each simulation. The cumulative
distribution of the rank of the favored allele in 1000 simulation for
each setting (\ref{fig:rank}) shows that \comale\ outperforms
other statistics. We also compared each method to see how often it
ranked the favored site in as the top ranked site
(\ref{tab:rank}A-B), among the top 10 ranked sites
(\ref{tab:rank}C-D), and among the top 50
(\ref{tab:rank}E-F) ranked sites. In the $\approx$1150 variants tested,
\comale\ performed consistently better than other methods in all of
these measures.

An interesting observation is revisiting the contrast between 
site-identification
and detection~\cite{long2013massive,tobler2014massive} (see \ref{app:ld} for 
more details). 
When selection 
coefficient is high, detection is easier
(\ref{fig:power}A-F), but site-identification is harder due to
the high LD between hitchhiking sites and the favored allele
(\ref{tab:rank}A-F).  Moreover, site-identification is harder in
hard sweep scenarios relative to soft sweeps. For example, when
coverage $\lambda=100$ and selection coefficient $s=0.1$, the
detection power is 75\% for hard sweep, but 100\% for soft sweep
(\ref{fig:power}B-E). In contrast, the favored site was ranked as
the top in 14\% of hard sweep cases, compared to and 95\% of soft
sweep simulations (\ref{tab:rank}A-B).  
\ignore{
\newcommand*\rot{\rotatebox{-90} }
\begin{table}[H]
	\centering
		\caption{\bf Percentage of simulations which favored allele appears in 
		top of the ranking.}
	\begin{tabular}{c||c}
		 Hard Sweep &Soft Sweep\\
		\hline
		(A) & (B)  \\
		{\input{../tables/top1.100.0.hard.tex}}
		 &{\input{../tables/top1.100.0.soft.tex} }\\
		\hline
		(C)  & (D)  \\
		{\input{../tables/top10.100.0.hard.tex}}
		& {\input{../tables/top10.100.0.soft.tex} }\\
		\hline
		(E) & (F) \\
		{\input{../tables/top50.100.0.hard.tex}}
		&{\input{../tables/top50.100.0.soft.tex} }\\
		\hline
	\end{tabular}\\
	\caption*{		Percentage of simulations in which the favored allele is 
	ranked 
	first 
		(A-B); 
		appears in top 10 (C-D); or,  appears in top 50 (E-F). In soft sweep 
		simulations (B,D,F), the ranks are consistently better than 
		hard sweep simulations (A,C,E). This can be attributed to lower LD 
		between the 
		hitchhikers (false positives) and favored allele in soft sweep 
		scenarios.}\label{tab:rank}
\end{table}
} 
\paragraph{Estimating Parameters.}
\comale\ computes the selection parameters $\hat{s}$ and $\hat{h}$ as
a byproduct of the hypothesis testing. We computed bias of selection
fitness ($s-\hat{s}$) and overdominance ($h-\hat{h}$) for of \comale\
and GP in each setting. The distribution of the error (bias) for
100$\times$ coverage is presented in \ref{fig:bias100} for
different configurations.
\ref{fig:bias30} and \ref{fig:biasinf} provide the
distribution of estimation errors for 30$\times$, and infinite
coverage, respectively.  For hard sweep, \comale\ provides estimates
of $s$ with lower variance of bias (\ref{fig:bias100}A). In soft
sweep, GP and \comale\ both provide unbiased estimates with low
variance (\ref{fig:bias100}B). \ref{fig:bias100}C-D shows
that \comale\ provides unbiased estimates of $h$ as well.

\paragraph{Running Time.}
As \comale\ does not compute exact likelihood of a region (i.e., does
not explicitly model linkage between sites), the complexity of
scanning a genome is linear in number of polymorphisms.  Calculating
score of each variant requires $\Oc(TR)$ and $\Oc(TRN^2)$ computation
for $\Mc$, and $\Hc$, respectively. However, most of the operations
are can be vectorized for all replicates to make the effective running
time for each variant.  We
conducted $1000$ simulations and measured running times for computing site 
statistics $M$, $H$, FIT, CMH and GP with different number of linked-loci.  Our
analysis reveals (\ref{fig:runTime}) that \comale\ is orders of
magnitude faster than GP, and comparable to FIT. While slower than CMH
on the time per variant, the actual running times are comparable after
vectorization and broadcasting over variants (see below).

These times can have a practical consequence. For instance, to run GP
in the single locus mode on the entire pool-seq data of the \dmel genome from a
small sample ($\approx$1.6M variant sites), it would take 1444 CPU-hours
($\approx$ 1 CPU-month). In contrast, after vectorizing and
broadcasting operations for all variants operations using
\texttt{numba} package, \comale\ took 75 minutes to perform an
scan, including precomputation, while the fastest method, CMH, took 17 minutes.

\subsection{Analysis of a \dmel Adaptation to Cold and Hot 
Temperatures}\label{sec:dmel}
We applied \comale\ to the 
\datadm~\cite{orozco2012adaptation,franssen2015patterns}, where
3 replicate samples were chosen from a population of \dmel for 59
generations under alternating 12-hour cycles of hot (28$^{\circ}$C)
and cold (18$^{\circ}$C) temperatures and sequenced.  In this dataset,
sequencing coverage is different across replicates and generations
(see S2 Fig of~\cite{Terhorst2015Multi}) which makes variant depths
highly heterogeneous (\ref{fig:depth} and \ref{fig:depthHetero}). 

We first filtered out heterochromatic,
centromeric and telomeric
regions~\cite{fiston2010drosophila}, and those variant that have collective 
coverage of more that 1500 in all 13 populations: three replicates at the base 
population, two replicates at generation 15, one replicate at generation 23, 
one replicate at generation 27, three replicates at generation 37 and three 
replicates at generation 59. After filtering, we ended up with 1,605,714 
variants.

First we estimated population size over the whole genome, by bootstrapping 
100,000  variants 100 times \ref{fig:estimateN}. 
The likelihood curves of \comale\ is sharper around the optimum compared to 
Bollback et. al~\cite{bollback2008estimation} (see Supplementary Fig. 1 
in~\cite{orozco2012adaptation}).
While chromosomes 3L and 3R 
appear to have smaller population size \ref{fig:NdmelChrom}, which can taken 
into account of 
chromosome-wise significance cutoffs, we took the genome-wide population size 
of $\widehat{N}=$250 to calculate $p$-values and FDR.
Next, we found maximum likelihood estimates of $s$, and computed the test 
statistic $H$ for each variant, given $N=250, h=0.5$. To compute $p$-values of 
$H$ statistics, we calculated 
single locus Wright-Fisher simulations for $\widehat{N}=$250, initial 
frequencies and variant depths of real data (see \ref{proc:arya}). We repeated 
forward simulations 50 times for whole genome, to collect $\approx$90M pool-seq 
trajectories 
with starting frequencies and coverage of real data. Then, $p$-value of each 
variant in the real 
data is calculated as the fraction of null statistics that are grater than or 
equal of test statistic(see~\ref{fig:null-alt}). Finally, to correct for 
multiple 
testing, we used 
Storey \& Tibshirani~\cite{storey2003statistical} method to compute 
FDR.~\ref{fig:man-dmel-snp} shows the distribution of 804 variants 
passing FDR$\le 
0.001$. We also performed GO
enrichment using \texttt{Gowinda}~\cite{kofler2012gowinda} and found 16 
enriched GO terms including XXXX, XXXX ~\ref{tab:gowinda}.

Finally, we tested if variants showing signal of overdominance, we computed $D$ 
statistic on simulated and experimental data, and computed $p$-values 
accordingly. After correcting for multiple testing, 96 variants discovered with 
FDR$\le 0.01$~\ref{fig:man-dmel-snp}. 
\subsection{Analysis of Outcrossing Yeast Populations}
