\begin{figure}[H]
 \begin{algorithm}[H]
 	\SetAlgorithmName{Generative Process}
 	\DontPrintSemicolon
 	\SetAlgoNoLine
 	\KwIn{$N,n,R,\{\lambda_{\tau_0},\ldots,\lambda_{\tau_T}\},\Tc= 
 		\{\tau_0,\ldots\tau_T\}$ }
 	\KwOut{Time-series pool-seq data for $R$ replicates of a single locus 
 		$\{\bfc^{(r)}\}$ and $\{\bfd^{(r)}\}$.}
 	\For{$r\leftarrow 1$ \KwTo $R$}
 	{
 		\For{$t\leftarrow \tau_0$ \KwTo $\tau_T$}
 		{
 			$2N\nu_t \sim \text{Binomial}({2N},{\nu_{t-1}})$\;
 			\If{$t\in \Tc$ }
 			{
 				$d^{(r)}_{t} \sim \poiss(\lambda_{\tau_i})$ \;
 				$2ny_t \sim \text{Binomial}({2n},{\nu_{t}})$\;  
 				$c^{(r)}_{t} \sim \text{Binomial}(d^{(r)}_{t},{y_{t}})$\; 
 			}
 		}
 	}
 	\caption{The Generative Process for Dynamic Pool-seq Data.} 
 \end{algorithm}
 \caption{aa}
  	\label{proc:arya}
\end{figure}	
\begin{figure}[H]
	\centering
	\includegraphics[trim=.01in 0.1in .01in 
	0.1in,clip,width=\textwidth]{sfs.pdf}
	\caption{{\bf Site Frequency Spectrum.}\\ Theoretical and
          Empirical SFS in a 50Kbp region for a neutral population of 200
          individuals when $N_e=10^6$ and $\mu=10^{-9}$. The $x$-axis 
          corresponds to site frequency, and
          the $y$-axis to the number of variants with a specific
          frequency. 
          In a neural population, majority of the variations stand in low 
          frequency.} \label{fig:sfs}
\end{figure}

\ignore{
\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{trajectoryReal.pdf}
	\caption{{\bf Trajectory of pool-sequenced variants.}\\
		Trajectory of three different variants that are increasing
		in frequency over time. Note that for read count data, the
		true allele frequency is not known. Here we draw the
		posterior distribution of the allele frequency at each time
		point using box plot. The median of each distribution is
		denoted by dots. The variance of each box is seen to be
		inversely related to the depth of the measurement. For instance, 
		generation 59 is sequenced with higher coverage than generation 37. 
		As a result, variance of observations in generation 59 is 
		considerably 
		smaller.}
	\label{fig:trajectoryReal}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=.0in 0 .0in 0, 
	clip,width=0.5\textwidth]{{statePosterior}.pdf}
	\caption{{\bf Posterior distribution of allele frequency.}\\
		Distribution of hidden allele frequency for different values
		of depth $d=\{5,50,500\}$. In all cases, the true frequency
		is $0.2$. The estimated frequency values are binomially distributed, 
		with different variances,
		around the true value in all cases with.}
	\label{fig:stateConditional}
\end{figure}}
\begin{figure}[H]
	\centering
	\begin{tabular}{c}
		\includegraphics[trim=.2in 0 .0in 0, 
		clip,width=0.8\textwidth]{{depthHetero}.pdf}
	\end{tabular}
	\caption{{\bf Coverage heterogeneity in time series data.}\\ Each panel
		shows the read depth for 3
		replicates of the \datadm\ data (see section~\ref{sec:dmel}). 
		Heterogeneity in depth of coverage is 
		seen
		between replicates, and also at different time points, in
		all 4 variants. None of these sites pass the the hard filtering with 
		minimum depth of 30. }
	\label{fig:depthHetero}
\end{figure}


\begin{figure}[H]
	\centering
	\begin{tabular}{c}
		\includegraphics[trim=.0in 0 .0in 0, 
		clip,width=0.8\textwidth]{{depth}.pdf}
	\end{tabular}
	\caption{ {\bf Distribution of depth in the real data.}\\
		Scaled PDF (A) and CDF (B) of the read depths of all ($\approx$22.1M) 
		measurements, i.e., all replicates and time points of the all 
		($\approx$1.7M) variants.
		Scaled PDF (C) and CDF (D) of the minimum depth of sites. While more 
		than half 
		most ($\approx$12.5M) of the measurements have depth of 50 or greater 
		(dashed line in (A),(B)), only a small fraction ($\approx$11K) of 
		variants (dashed line in (C),(D)) pass the filter of having minimum 
		depth of 50.}  
	\label{fig:depth}
\end{figure}

\ignore{
\begin{figure}[H]
	\centering
	\includegraphics[width=0.5\textwidth]{dominance.pdf}
	\caption{{\bf Natural selection in infinite population.}\\ Trajectory of 
		the 
		favored allele in an infinite population with $s=0.1$ for $h=0$ 
		(recessive 
		favored allele), $h=0.5$ (directional selection), $h=1$ (dominant 
		favored 
		allele) and $h=2$ (overdominant favored allele). }
	\label{fig:dominance}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[trim=0.1in 0 .08in 0.02in , 
	clip,width=0.6\textwidth]{topSNPs.pdf}
	\caption{ {\bf Variant showing strong signal of directional (Left) and 
			over-dominant, aka balancing selection (Right). }  }
	\label{fig:dir-bal}
\end{figure}





\begin{figure}[H]
	\centering 
	\includegraphics[width=\textwidth]{{msmsts}.pdf}
	\caption{{\bf Dynamic SFS-based statistics.}\\ Mean and 95\% CI
		of 100 simulations for neutral (blue trajectories) selection
		with $s=0.1$ (red trajectories). In all case, statistic computed for 
		a 50Kbp window and $N_e=10^4$, $\mu=10^{-9}$. The dashed line 
		shows 
		the parametric models derived in~\ref{app:td}.  }
	\label{fig:sfsts}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[trim=0in 0in 0.0in 0in , 
	clip,width=0.6\textwidth]{powerROC.pdf}
	\caption{{\bf Schematic for computing power of detecting
            selection.}\\ Receiver Operating Characteristics (ROC) curve
            (left) for classification of 2000 simulations (1000 selection and
          1000 neutral). The Area Under the Curve (AUC) represents
          overall performance. The diagonal black line represents
          performance of a random hypothesis which achieves Area under
          the curve (AUC) of 0.5. To avoid computing AUC for the
          regions where FPR is unacceptably high, we restrict ROC
          curve to the region where FPR$\le 0.05$ (right). In this
          case, we define power to be the (scaled) AUC of the
          restricted region.} \label{fig:powerROC}
\end{figure}











\begin{figure}[H]
	\centering 
	\includegraphics[trim=0in 0.1in 0in 0.2in , 
	clip,width=\textwidth]{bottleneck.pdf}
	\caption{{\bf Effect of bottleneck in a neutral experimental
			evolution experiment with restricted number of founder
			lines.}\\ Dynamic of SFS based statistics under neutral evolution 
			when  $F=200$ founders were selected
		from a larger population size ($N_e=10^{6}$). The statistics for
		Tajima's $D$, Fay Wu's $H$ and SFSelect were
		computed for 1000 neutral simulations and the mean and 95\%
		confidence interval plotted. Under neutral evolution, all
		the statistics are expected to vary around a fixed mean
		through time. However, under selective constraint, $D$ and
		$H$ take negative values, while SFSelect take positive
		values. In experimental evolution, bottleneck effect will
		suppress the signal of selection, especially in early
		generations. }
	\label{fig:bottleneck}
\end{figure}}


\begin{figure}[H]
	\centering
		\includegraphics[width=0.6\textwidth]{{bias.30}.pdf}
	\caption{{\bf Distribution of bias for 30$\times$ coverage.}\\ The
		distribution of bias ($s-\hat{s}$) in estimating selection
		coefficient over 1000 simulations using Gaussian Process (GP) and
		\comale\ ($H$) is shown for a range of choices for the selection
		coefficient $s$ and starting carrier frequency $\nu_0$, when
		coverage $\lambda=30$ (Panels A,B). GP and \comale\ have similar
		variance in estimates of $s$ for soft sweep, while \comale\ provides
		lower variance in hard sweep. Also see \ref{tab:biasdist}. Panels 
		C,D
		show the variance in the estimation of $h$.}
	\label{fig:bias30}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[width=0.6\textwidth]{{bias.300}.pdf}
	\caption{{\bf Distribution of bias for infinite coverage.}\\ The
		distribution of bias ($s-\hat{s}$) in estimating selection
		coefficient over 1000 simulations using Gaussian Process (GP) and
		\comale\ ($H$) is shown for a range of choices for the selection
		coefficient $s$ and starting carrier frequency $\nu_0$, when
		coverage $\lambda=\infty$ (Panels A,B). GP and \comale\ have similar
		variance in estimates of $s$ for soft sweep, while \comale\ provides
		lower variance in hard sweep. Also see \ref{tab:biasdist}. Panels 
		C,D
		show the variance in the estimation of $h$.
	}
	\label{fig:biasinf}
\end{figure}
\ignore{
\begin{figure}[H]
	\centering
	\includegraphics[width=\textwidth]{LDDecay3d}
	\caption{ {\bf Expected dynamic of LD under selection and neutral 
	evolution.}\\ Dynamic of LD ($\rho_t$) of a 1Mbp genome to the favored 
	allele 
	(at position 500K) is drawn as function of position and time for neutral 
	(top) and selection(bottom) regimes.
	For sake of illustration, we assumed that at generations 0, LD of all 
	variants with the favored allele is 0.5, initial frequency of the favored 
	allele is 0.1, recombination rate is $r=2\times10^{-8}$ (top). The 
	selection strength is 0 and 0.05 for neutral and selection regimes, 
	respectively. As expected LD decay exponentially  through space and time. 
	However, selection causes LD to increase then decrease.}	
		\label{fig:ld3d}
\end{figure}

\begin{figure}[H]
	\centering
		\includegraphics[width=0.8\textwidth]{decayFactors0}	
	\caption{{\bf Interaction between growth and decay factors of LD in 
	hard 
	sweep with	weak selection.}\\
		Expected dynamics of LD under weak selection ($s=0.025$) with hard 
		sweep 
		($\nu_0=0.005$). In addition to 
		recombination, initial frequency of the favored allele and selection 
		strength determine the dynamics of LD. The vertical dashed line denotes 
		the time in which LD start to decrease. LD between the favored 
		allele 
		and the rest of genome increase for 
		$\approx$700 generations after onset of selection, implying 
		that 
		localizing adaptive allele in short 
		term experimental evolution is a difficult task, especially 
		when the 
		frequency of the favored allele is low.} \label{fig:ldf0}
\end{figure}

\begin{figure}[H]
	\centering
	\includegraphics[width=0.8\textwidth]{decayFactors1}	
\caption{{\bf Interaction between growth and decay factors of LD in 
hard 
		sweep with	strong selection.}\\
	Expected dynamics of LD under strong selection ($s=0.1$) with hard sweep 
	($\nu_0=0.005$). In addition to 
	recombination, initial frequency of the favored allele and 
	selection 
	strength determine the dynamics of LD. The vertical dashed line 
	denotes 
	the time in which LD start to decrease. LD between the favored 
	allele 
	and the rest of genome increase for 
	$\approx$200 generations after onset of selection, implying that 
	localizing adaptive allele in short 
	term experimental evolution is a difficult task, especially when 
	the 
	frequency of the favored allele is low.} \label{fig:ldf1}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[width=0.8\textwidth]{decayFactors2}	
\caption{{\bf Interaction between growth and decay factors of LD in 
soft 
		sweep with	weak selection.}\\
	Expected dynamics of LD under weak selection ($s=0.025$) for soft sweep 
	($\nu_0=0.1$).  In addition to 
	recombination, initial frequency of the favored allele and 
	selection 
	strength determine the dynamics of LD. The vertical dashed line 
	denotes 
	the time in which LD start to decrease. LD between the favored 
	allele 
	and the rest of genome increase for 
	$\approx$300 generations after onset of selection, implying that 
	localizing adaptive allele in short 
	term experimental evolution is a difficult task.} \label{fig:ldf2}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[width=0.8\textwidth]{decayFactors3}	
\caption{{\bf Interaction between growth and decay factors of LD in 
soft 
		sweep with	strong selection.}\\
	Expected dynamics of LD under strong selection ($s=0.1$) for soft sweep 
	($\nu_0=0.1$).  In addition to 
	recombination, initial frequency of the favored allele and 
	selection 
	strength determine the dynamics of LD. The vertical dashed line 
	denotes 
	the time in which LD start to decrease. LD between the favored 
	allele 
	and the rest of genome increase for 
	$\approx$80 generations after onset of selection, implying that 
	localizing adaptive allele in short 
	term experimental evolution is a difficult task.} \label{fig:ldf3}
\end{figure}
}

\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{{rank30.0}.pdf}
	\caption{{\bf Ranking performance for 30$\times$ coverage.}\\
           Cumulative Distribution Function (CDF) of the distribution
           of the rank of the favored allele in 1000 simulations for
           \comale\ ($H$ score), Gaussian Process (GP), and Cochran 
           Mantel 
           Haenszel  
           (CMH), for different values of selection
           coefficient $s$ and initial carrier frequency.}
	\label{fig:rank300}
\end{figure}
\begin{figure}[H]
	\centering
	\includegraphics[trim=.2in 0 .2in 0, 
	clip,width=\textwidth]{{rank300.0}.pdf}
	\caption{{\bf Ranking performance for 300$\times$ coverage.}\\
           Cumulative Distribution Function (CDF) of the distribution
           of the rank of the favored allele in 1000 simulations for
           \comale\ ($H$ score), Gaussian Process (GP), and Cochran 
           Mantel 
           Haenszel  
           (CMH), for different values of selection
           coefficient $s$ and initial carrier frequency.}
	\label{fig:rankinf}
\end{figure}


\begin{figure}[H]
	\centering
	\includegraphics[trim=.0in 0in .2in 0in, 	
	clip,width=0.9\textwidth]{{null-alt-dmel}.pdf}
	\caption{{\bf .}.}
	\label{fig:null-alt}
\end{figure}

\begin{figure}[H]
\begin{tabular}{cc}
	\includegraphics[width=0.5\textwidth]{{SFS.Y.G0}.pdf}&
	\includegraphics[width=0.5\textwidth]{{SFS.Y.G180}.pdf}\\
	\includegraphics[width=0.5\textwidth]{{SFS.Y.G360}.pdf}&
	\includegraphics[width=0.5\textwidth]{{SFS.Y.G540}.pdf}
\end{tabular}
\caption{{\bf Yeast SFS.}}
\label{fig:yeast-sfs}
\end{figure}






