%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%% Supplemental Material%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\clearpage
%\newpage
\section{Window Size}\label{sec:winSize}

%\section*{Main text figure captions}
\ignore{
\section{Dynamics of Site Frequency Spectrum-based Statistics and Linkage 
	Disequilibrium under Selection }\label{app}
\subsection{An approximate logistic function  for allele frequency dynamics} 
\label{app:af}
Assume that a site is evolving under selection constraints $s,h\in
\Rbb$, where $s$ and $h$ denote selection strength and overdominance,
respectively. Let $\nu_t$ denote the frequency of the site at time
$\tau_t\in {\cal T}$. Then, $\nusp$, the frequency at time $\tau_t+1$
can be estimated using:
\beq
\hat{\nu}_{t^+}
=\nu_t+\frac{s(h+(1-2h)\nu_t)\nu_t(1-\nu_t)}{1+s\nu_t(2h+(1-2h)\nu_t))}.
  \label{eq:transition2}
\eeq
We can show that the dynamic of the favored allele can be modeled
via a logistic function, in the case of directional selection
($h=0.5$). Taking derivatives of Eq.~\ref{eq:transition2}, we have
\beq \frac{\bfd \nu_t}{\bfd t} = \frac{s\nu_t(1-\nu_t)}{2+2s\nu_t}
\eeq To, solve the differential equation, note that for small $s$,
$2+2s\nu_t \approx 2$. Substituting,
\begin{equation}
\nu_t =\frac{1}{1+\frac{1-\nu_0}{\nu_0}e^{-st/2}} = \sigma(st/2+\eta(\nu_0)) 
\label{eq:inf-pop}
\end{equation}
where $\sigma(.)$ is the logistic function and $\eta(.)$ is logit function 
(inverse of the logistic function). 

\NEW{SFS appendices removed.}


\subsection{Linkage Disequilibrium}\label{app:ld}
Nonrandom associations, Linkage Disequilibrium (LD), between 
polymorphisms are established in the 
substitution process, broken by recombination events 
and reinforced by selection. 
Although LD can not be measured in pooled sequencing data (phased 
haplotype data is required), it is still worthwhile 
to examine the behavior of LD as a result of the interaction between 
recombination and natural selection. In this part we theoretically overview 
expected LD in short EEs.

Let $\rho_0$ be the LD at time zero between the favored allele and a 
segregating site $l$ base-pairs away, then under natural selection we have
\beq
\rho_t= \alpha_t\beta_t \rho_0=e^{-rtl} \left(\frac{K_t}{K_0}\right)  
\rho_0\label{eq:ldt}
\eeq
where $K_t=2\nu_t(1-\nu_t)$ is the heterozygousity at the selected site, $r$ is 
the recombination rate/bp/gen. The 
\emph{decay factor}, $\alpha_t=e^{-rtl}$,
and \emph{growth factor}, $\beta_t$ (see Eqs. 30-31 in 
\cite{stephan2006hitchhiking}), are result of recombination and 
selection, respectively. \ref{fig:ld3d} presents the expected theoretical 
value of LD when $\rho_0=0.5$ between favored allele (site at position 500K) 
and the rest of 
genome, and $\nu_0=0.1$. For neutral evolution (top), LD decays exponentially 
through space and time, while in natural selection (bottom), LD increases and 
then decreases. Interestingly, LD increases to its maximum value, 1, for the 
nearby region (the plateau in \ref{fig:ld3d} bottom) of the favored 
allele.

In principle, LD increases after the onset of selection, until $\log(\alpha_t) 
+\log(\beta_t) 
>0$, see Eq.~\ref{eq:ldt}. 
Specifically, log of decay term is linear and, using 
Eq.~\ref{eq:inf-pop}, we write growth 
factor in term of initial frequency $\nu_0$ and selection strength $s$. 
\ref{fig:ldf0},~\ref{fig:ldf1},~\ref{fig:ldf2}, and ~\ref{fig:ldf3} 
depict interaction of 
decay and growth factors for weak and 
strong selection and soft and hard sweeps. In all the case, LD of the 
favored allele with a segregating site 50Kbp away, increases in the first 50 
generations, which give rise to increasing number of \emph{hitchhikers}. 

Increase of LD in a large (100Kbp) region is particularly advantageous to the 
task of identifying the region under selection, if the composite statistics is 
used. As a result, $\Hc$ statistic outperforms existing (single-loci) tools in 
identifying selection. In contrast, augmentation of LD, increases the 
number 
of candidates for 
the favored allele, which makes it difficult to localize the favored 
allele.
}
