%\documentclass[handout]{beamer} % \pause disabled
\documentclass[t]{beamer} % \pause enabled
\input{../macros_arya}
\usepackage{datetime}
\def\vecbold#1{{\boldsymbol#1}}

\newdateformat{monthyeardate}{%
  \monthname[\THEMONTH], \THEYEAR}

%\usepackage[linesnumbered,ruled]{algorithm2e}
\usetheme{Madrid}
\usepackage{tabularx}
\setbeamertemplate{footline}
{
  \leavevmode%
  \hbox{%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{author in head/foot}%
    \usebeamerfont{author in head/foot}\insertshortauthor
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,center]{title in head/foot}%
    \usebeamerfont{title in head/foot}\insertshorttitle
  \end{beamercolorbox}%
  \begin{beamercolorbox}[wd=.333333\paperwidth,ht=2.25ex,dp=1ex,right]{date in head/foot}%
    \usebeamerfont{date in head/foot}\insertshortdate{}\hspace*{2em}
    \insertframenumber{} / \inserttotalframenumber\hspace*{2ex} 
  \end{beamercolorbox}}%
  \vskip0pt%
}
\def\dHAF{\text{-HAF}}
\makeatother
\title{Identifying Selection in Experimental Evolution}
\subtitle{}
\author[Arya Iranmehr]
{%
  \texorpdfstring{
      \centering
      Arya Iranmehr\\
      \href{mailto:airanmehr@ucsd.edu}{airanmehr@ucsd.edu}
  }
  {Arya Iranmehr}
}
%\email{}
\institute{
Bafna Lab\\
University of California, San Diego}
\date{
\monthyeardate\today}
\subject{Optimization}
\usefonttheme{serif} % default family is serif


\begin{document}
\begin{frame}
  \titlepage
\end{frame}


\begin{frame}{Introduction}
\begin{itemize}
\item Next generation sequencing has made \emphr{whole-genome \& 
whole-population} sequencing possible.
\begin{figure}
\includegraphics[trim={0in 0in 0in 
	0in},clip,width=0.5\textwidth]{../figures/{wgwp}}
\tiny{www.1000genomes.org}
\end{figure}
\pause
\item For organisms with  ``short-generation-time", (e.g., yeast, \emph{E. 
coli}, \dmel etc.) it is  also
possible 
to collect \emphr{time-series} data of 
population.
\pause
\item Given rise of these \emphr{modern datasets} (population longitudinal 
data), new techniques required to answer classical population genetics 
questions on real data.
\end{itemize}
\end{frame}

\begin{frame}{ Goals }
Design a method which
\begin{itemize}
	\item Detect regions under selection.
	\item Localizing adaptive allele within the candidate region.
	\item Estimating selection parameters.
\end{itemize}
\end{frame}

\begin{frame}{Experimental Evolution (EE)}
\begin{itemize}
	\item EE is a long tradition in biology, which studies the 
	\emphr{phenotype in time} by reducing \emphr{environmental effects} .
		\begin{figure}
			\includegraphics[trim={0in 3.6in 8.8in 
				1.5in},clip,width=0.3\textwidth]{../figures/{ee-eg1}}
\tiny{Nature Reviews Genetics 14, 827-839 (2013)}
		\end{figure}
		\pause
	\item In a \emphr{controlled} environment, EE evolves a homogeneous 
	population.
	\pause
	\item Let phenotype of interest be the \emphr{response to a selection 
	pressure}, e.g., 
	response to
	\begin{itemize}
		\item antibiotic
		\item low oxygen conditions
		\item hot and cold temperatures 
		\item etc.
	\end{itemize}
				
\end{itemize}
\end{frame}

\begin{frame}{An experiment design for \dmel}
	\begin{figure}
		\includegraphics[trim={18in 0.0in 0.0in 
			0in},clip,width=0.8\textwidth]{../figures/ExperimentalEvolution}
	\end{figure}
\end{frame}

\begin{frame}{Whole-Genome Whole-Population Sequencing}
	\begin{itemize}
		\item Pooled-Sequencing 
		\begin{figure}
			\includegraphics[trim={.05in 7.5in 0.0in 
				0in},clip,width=0.9\textwidth]{../figures/pool-seq}\\
			\tiny{Nature Reviews Genetics 15, 749-763 (2014)}
		\end{figure}
		\pause
			\item Implication: only population allele frequency can be computed.
	\end{itemize}	
\end{frame}

\begin{frame}{Dynamic of population allele frequency }
	under different \emphr{initial conditions} and \emph{selection strengths} 
	frequency 
	change differently
	\begin{figure}
		\includegraphics[trim={10in 0.0in 0.0in 
			0in},clip,width=0.7\textwidth]{../figures/AF}
	\end{figure}
\end{frame}


\begin{frame}{ Simplified Model (I)}
	\begin{itemize}
		\item  Suppose we have sequenced a whole (diploid, size=$N$) 
		population \emphr{every 
		generation} (eg, for 6 generations) and \emphr{exact allele 
		frequency} are given.
	\pause
		\item A discrete-time discrete-state model, Markov chain, can 
		generate such a data.
		\begin{figure}
			\includegraphics[trim={.05in 0in 0.0in 
				0.5in},clip,width=0.5\textwidth]{../figures/markoveg}
		\end{figure}
	\end{itemize}
\end{frame}

\begin{frame}{ Simplified Model (II)}
	\begin{itemize}
		\item  Where $Q_{i,j}(s,h)$ is the probability of going from frequency 
		$i/(2N)$ to $j/(2N)$ when selection strength is $s$ and over dominance 
		is $h$.
		\pause
		\item Likelihood of parameter can be easily computed
		\beqq
		\Lc(s,h| \{\nu_0,\ldots,\nu_5\}) = \pr(\{\nu_0,\ldots,\nu_5\}| Q(s,h) )
		\eeqq
		\pause
		\item perform maximum likelihood to find $\hat{s}$, $\hat{h}$.
		\pause
		\item compute likelihood ratio, $M$ statistic for each SNP:
		\beqq
		M&=\frac{\text{likelihood of data as if being under 
		selection with } \hat{s},\hat{h}}{l\text{ikelihood 
		of data as if being neutral}}\\
		&=\frac{	\Lc(\hat{s},\hat{h}| \{\nu_0,\ldots,\nu_5\})}{	\Lc(0,0| 
		\{\nu_0,\ldots,\nu_5\})}
		\eeqq
	\end{itemize}
\end{frame}


\begin{frame}{ Model (complete)}
	\begin{itemize}
		\item  In reality, population is sequenced after some ($\tau$) 
		generations.\\
		\emphr{solution: use  $Q^\tau$ in computing likelihoods.}
		\pause
		\item Allele frequencies are unknown, and depth of each variant can be 
		different.\\
		\emphr{solution: extend Markov chain to an HMM by specifying 
		emission probabilities }
	\beqq%	\pr(\text{derived allele read count}|\text{depth},\nu)=\pr(c|d,\nu)\\
	d &\sim \text{Poisson}(\text{Coverage})\\	
	c &\sim \text{Bionomial}(N=d, \theta=\nu)	
	\eeqq
	\pause
	\item everything the is the same, except we need to use \emphr{forward 
	algorithm} to compute likelihood of a sequence of reads.
\pause
	\item Similarly, 
	\beqq
	H&=\frac{\text{likelihood of read-count data as if being under 
			selection }}{l\text{ikelihood 
			of read-count data as if being neutral}}
	\eeqq
	\end{itemize}
\end{frame}

\begin{frame}{Composite Likelihood for a Region (I)}
	\begin{itemize}
		\item So far we developed log-odds ratio statistics $M$ (frequency 
		data) and $H$ (read count data) \emphr{for each variant}.
		\pause
		\item  For a small region with $L$ variants we can simply take the max 
		score in the region, which is prone to \emphr{false positives}.
		\pause
		\item We know that variants are \emphr{correlated}
		\begin{figure}
			\includegraphics[trim={.05in 0in 0.0in 
				0in},clip,width=0.5\textwidth]{../figures/{ld.eg}.png}\\
			\tiny{Crit Care. 2014 Jun 20;18(3):R127}
		\end{figure}
	\end{itemize}
\end{frame}

\begin{frame}{Composite Likelihood for a Region (II)}
	\begin{itemize}
		\item Computing joint likelihoods of SNPs is \emphr{infeasible} 
		(haplotypes are 
		required) and \emphr{intractable} (requires estimating covariance).
		\pause
		\item A heuristic is to compute composite (aka, pseudo) likelihood of  
		the 
		region $L$ to reduce false-positives
		\beqq
		\Hc = \frac{1}{|L|}\sum_{\ell \in L}H_\ell
		\eeqq
	\end{itemize}
\end{frame}


\begin{frame}{ Performance in Detecting Regions under Selection}
Each point represent power  (TPR when FPR$\le$0.05) of detection in 1000 
simulations (500 neutral, 500 selection) of a 50Kbp window, for different 
coverages.
		\begin{figure}
			\includegraphics[trim={.05in .0in 0.0in 
				0in},clip,width=\textwidth]{../figures/power.pdf}
		\end{figure}
\end{frame}

\begin{frame}{ Detecting regions under selection: Observations}
\begin{enumerate}[(i)]
	\item Provides better and much robust performances to change of coverage.
	\pause
	\item It can detect well even when coverage is low, i.e., favored allele 
	frequency (1/200 in hard 
	sweep) is below accuracy of sequencing (1/30).
	\pause
	\item Run time is better or comparable with others.
	\begin{figure}
		\includegraphics[trim={.0in .0in 0.0in 
			0in},clip,width=0.5\textwidth]{../figures/runTime.pdf}
	\end{figure}
\end{enumerate}
\end{frame}


\begin{frame}{ Localizing favored allele}
Each curve depicts cumulative distribution of the rank  of favored allele among 
($\approx$ 1150) variants, in 500 simulations.

	\begin{figure}
		\includegraphics[trim={.05in .0in 0.0in 
			0in},clip,width=\textwidth]{../figures/{rank100.0}.pdf}
	\end{figure}
\end{frame}


\begin{frame}{ Estimating parameters (I)}
Our model estimates strength of selection $s$ and overdominance 
		$h$ parameter for each variant.
	\begin{figure}
			\includegraphics[trim={.0in .0in 0.0in 
				0in},clip,width=0.5\textwidth]{../figures/{dominance}.pdf}
	\end{figure}
		\begin{itemize}
			\item $h=0$ : recessive adaptive allele
			\item 				$h=0.5$ : directional selection
			\item $h=1$:	dominant adaptive allele
			\item 				$h>1$ :overdominance
		\end{itemize}
\end{frame}


\begin{frame}{ Estimating parameters (II)}
	Distribution of bias of parameters in 500 simulations.
	\begin{figure}
		\includegraphics[trim={.05in .0in 0.0in 
			0in},clip,width=0.6\textwidth]{../figures/{bias.100}.pdf}
	\end{figure}
\end{frame}

\begin{frame}{ Analysis of real data}
\begin{itemize}
	\item A population of \dmel is evolved for 59 generations, under 
	alternative hot and cold temperatures.
	\pause
	\item Coverage is different at generations and samples are not synchronized.
	\pause
	\item Genome scan for sliding window size=50Kbp,  steps=10Kbp
		\begin{figure}
			\includegraphics[trim={.05in .0in 0.0in 
				0in},clip,width=0.8\textwidth]{../figures/{manhattan.min500snp}.pdf}
		\end{figure}		
\end{itemize}
\end{frame}




\begin{frame}{ Discussion}
	\begin{itemize}
		\item An efficient method for analyzing \emphr{ full time-series 
		read-count data} is proposed.
		\pause
		\item By computing composite likelihood $\Hc$ statistic is more robust 
		to false positives.	
		\pause 
		\item When initial frequency of the favored allele is low, stronger 
		selection helps detecting selection but makes locating favored allele a 
		harder task.
		\item Next step is to apply to new dataset with a well defined 
		phenotype, e.g. response to hypoxia.
	\end{itemize}
\end{frame}

\begin{frame}{\ }
	\vspace{1	in}
	\begin{center}
		\huge{Thanks!}\\
	\end{center}
	
\end{frame}
\end{document}